

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>RNA-seq | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="GCModeller [version 1.0.0.0]Module AssemblyName: file:///G:/GCModeller/manual/bin/RNA-seq.exeRoot namespace: RNA_seq.CLI
All of the command that available in this program has been list below:



Funct">
<meta name="keywords">
<meta property="og:type" content="website">
<meta property="og:title" content="RNA-seq">
<meta property="og:url" content="http://docs.gcmodeller.org/tools/RNA-seq.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="GCModeller [version 1.0.0.0]Module AssemblyName: file:///G:/GCModeller/manual/bin/RNA-seq.exeRoot namespace: RNA_seq.CLI
All of the command that available in this program has been list below:



Funct">
<meta property="og:updated_time" content="2016-07-27T10:40:24.021Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="RNA-seq">
<meta name="twitter:description" content="GCModeller [version 1.0.0.0]Module AssemblyName: file:///G:/GCModeller/manual/bin/RNA-seq.exeRoot namespace: RNA_seq.CLI
All of the command that available in this program has been list below:



Funct">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/tools/RNA-seq.html">
      <h1 class="article-title" itemprop="name">
        RNA-seq
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/tools/RNA-seq.html" class="article-date">
  <time datetime="2016-07-27T10:40:24.000Z" itemprop="datePublished">2016-07-27</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/tools/RNA-seq.html">
      <h1 class="article-title" itemprop="name">
        RNA-seq
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#GCModeller-version-1-0-0-0"><span class="toc-number">1.</span> <span class="toc-text">GCModeller [version 1.0.0.0]</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Commands"><span class="toc-number">1.1.</span> <span class="toc-text">Commands</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Associate-GI’"><span class="toc-number">1.1.0.0.1.</span> <span class="toc-text">Help for command ‘/Associate.GI’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Clustering’"><span class="toc-number">1.1.0.0.2.</span> <span class="toc-text">Help for command ‘/Clustering’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Co-Vector’"><span class="toc-number">1.1.0.0.3.</span> <span class="toc-text">Help for command ‘/Co.Vector’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Contacts-Ref’"><span class="toc-number">1.1.0.0.4.</span> <span class="toc-text">Help for command ‘/Contacts.Ref’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Data-Frame’"><span class="toc-number">1.1.0.0.5.</span> <span class="toc-text">Help for command ‘/Data.Frame’:</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Accepted-Types"><span class="toc-number">1.1.0.1.</span> <span class="toc-text">Accepted Types</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#in"><span class="toc-number">1.1.0.1.1.</span> <span class="toc-text">/in</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-DataFrame-RPKMs’"><span class="toc-number">1.1.0.1.2.</span> <span class="toc-text">Help for command ‘/DataFrame.RPKMs’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-DEGs’"><span class="toc-number">1.1.0.1.3.</span> <span class="toc-text">Help for command ‘/DEGs’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-DEGs-UpDown’"><span class="toc-number">1.1.0.1.4.</span> <span class="toc-text">Help for command ‘/DEGs.UpDown’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-DESeq2’"><span class="toc-number">1.1.0.1.5.</span> <span class="toc-text">Help for command ‘/DESeq2’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-DOOR-Corrects’"><span class="toc-number">1.1.0.1.6.</span> <span class="toc-text">Help for command ‘/DOOR.Corrects’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Export-Megan-BIOM’"><span class="toc-number">1.1.0.1.7.</span> <span class="toc-text">Help for command ‘/Export.Megan.BIOM’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Export-SAM-Maps’"><span class="toc-number">1.1.0.1.8.</span> <span class="toc-text">Help for command ‘/Export.SAM.Maps’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Export-SSU-Refs’"><span class="toc-number">1.1.0.1.9.</span> <span class="toc-text">Help for command ‘/Export.SSU.Refs’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Export-SSU-Refs-Batch’"><span class="toc-number">1.1.0.1.10.</span> <span class="toc-text">Help for command ‘/Export.SSU.Refs.Batch’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Format-GI’"><span class="toc-number">1.1.0.1.11.</span> <span class="toc-text">Help for command ‘/Format.GI’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-fq2fa’"><span class="toc-number">1.1.0.1.12.</span> <span class="toc-text">Help for command ‘/fq2fa’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-gast’"><span class="toc-number">1.1.0.1.13.</span> <span class="toc-text">Help for command ‘/gast’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-gast-stat-names’"><span class="toc-number">1.1.0.1.14.</span> <span class="toc-text">Help for command ‘/gast.stat.names’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Group-n’"><span class="toc-number">1.1.0.1.15.</span> <span class="toc-text">Help for command ‘/Group.n’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-HT-seq’"><span class="toc-number">1.1.0.1.16.</span> <span class="toc-text">Help for command ‘/HT-seq’:</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Accepted-Types-1"><span class="toc-number">1.1.0.2.</span> <span class="toc-text">Accepted Types</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Mode"><span class="toc-number">1.1.0.2.1.</span> <span class="toc-text">/Mode</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#feature"><span class="toc-number">1.1.0.2.2.</span> <span class="toc-text">/feature</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Imports-gast-Refs-NCBI-nt’"><span class="toc-number">1.1.0.2.3.</span> <span class="toc-text">Help for command ‘/Imports.gast.Refs.NCBI_nt’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-log2-selects’"><span class="toc-number">1.1.0.2.4.</span> <span class="toc-text">Help for command ‘/log2.selects’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-PCC’"><span class="toc-number">1.1.0.2.5.</span> <span class="toc-text">Help for command ‘/PCC’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Rank-Statics’"><span class="toc-number">1.1.0.2.6.</span> <span class="toc-text">Help for command ‘/Rank.Statics’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-RPKM’"><span class="toc-number">1.1.0.2.7.</span> <span class="toc-text">Help for command ‘/RPKM’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-RPKM-Log2’"><span class="toc-number">1.1.0.2.8.</span> <span class="toc-text">Help for command ‘/RPKM.Log2’:</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Accepted-Types-2"><span class="toc-number">1.1.0.3.</span> <span class="toc-text">Accepted Types</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#cond"><span class="toc-number">1.1.0.3.1.</span> <span class="toc-text">/cond</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-sid-map’"><span class="toc-number">1.1.0.3.2.</span> <span class="toc-text">Help for command ‘/sid.map’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-SPCC’"><span class="toc-number">1.1.0.3.3.</span> <span class="toc-text">Help for command ‘/SPCC’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Stat-Changes’"><span class="toc-number">1.1.0.3.4.</span> <span class="toc-text">Help for command ‘/Stat.Changes’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Statics-Labels’"><span class="toc-number">1.1.0.3.5.</span> <span class="toc-text">Help for command ‘/Statics.Labels’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-WGCNA’"><span class="toc-number">1.1.0.3.6.</span> <span class="toc-text">Help for command ‘/WGCNA’:</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Accepted-Types-3"><span class="toc-number">1.1.0.4.</span> <span class="toc-text">Accepted Types</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#data"><span class="toc-number">1.1.0.4.1.</span> <span class="toc-text">/data</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#anno"><span class="toc-number">1.1.0.4.2.</span> <span class="toc-text">/anno</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#mods"><span class="toc-number">1.1.0.4.3.</span> <span class="toc-text">/mods</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#out"><span class="toc-number">1.1.0.4.4.</span> <span class="toc-text">/out</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#From-Deseq"><span class="toc-number">1.1.0.4.5.</span> <span class="toc-text">/From.Deseq</span></a></li></ol></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="GCModeller-version-1-0-0-0"><a href="#GCModeller-version-1-0-0-0" class="headerlink" title="GCModeller [version 1.0.0.0]"></a>GCModeller [version 1.0.0.0]</h1><p><strong>Module AssemblyName</strong>: file:///G:/GCModeller/manual/bin/RNA-seq.exe<br><strong>Root namespace</strong>: RNA_seq.CLI</p>
<p>All of the command that available in this program has been list below:</p>
<table>
<thead>
<tr>
<th>Function API</th>
<th>Info</th>
</tr>
</thead>
<tbody>
<tr>
<td>/Associate.GI</td>
<td></td>
</tr>
<tr>
<td>/Clustering</td>
<td></td>
</tr>
<tr>
<td>/Co.Vector</td>
<td></td>
</tr>
<tr>
<td>/Contacts.Ref</td>
<td></td>
</tr>
<tr>
<td>/Data.Frame</td>
<td>Generates the data input for the DESeq2 R package.</td>
</tr>
<tr>
<td>/DataFrame.RPKMs</td>
<td>Merges the RPKM csv data files.</td>
</tr>
<tr>
<td>/DEGs</td>
<td></td>
</tr>
<tr>
<td>/DEGs.UpDown</td>
<td></td>
</tr>
<tr>
<td>/DESeq2</td>
<td></td>
</tr>
<tr>
<td>/DOOR.Corrects</td>
<td></td>
</tr>
<tr>
<td>/Export.Megan.BIOM</td>
<td></td>
</tr>
<tr>
<td>/Export.SAM.Maps</td>
<td></td>
</tr>
<tr>
<td>/Export.SSU.Refs</td>
<td></td>
</tr>
<tr>
<td>/Export.SSU.Refs.Batch</td>
<td></td>
</tr>
<tr>
<td>/Format.GI</td>
<td></td>
</tr>
<tr>
<td>/fq2fa</td>
<td></td>
</tr>
<tr>
<td>/gast</td>
<td></td>
</tr>
<tr>
<td>/gast.stat.names</td>
<td></td>
</tr>
<tr>
<td>/Group.n</td>
<td></td>
</tr>
<tr>
<td>/HT-seq</td>
<td>Count raw reads for DESeq2 analysis.</td>
</tr>
<tr>
<td>/Imports.gast.Refs.NCBI_nt</td>
<td></td>
</tr>
<tr>
<td>/log2.selects</td>
<td></td>
</tr>
<tr>
<td>/PCC</td>
<td></td>
</tr>
<tr>
<td>/Rank.Statics</td>
<td></td>
</tr>
<tr>
<td>/RPKM</td>
<td></td>
</tr>
<tr>
<td>/RPKM.Log2</td>
<td></td>
</tr>
<tr>
<td>/sid.map</td>
<td></td>
</tr>
<tr>
<td>/SPCC</td>
<td></td>
</tr>
<tr>
<td>/Stat.Changes</td>
<td></td>
</tr>
<tr>
<td>/Statics.Labels</td>
<td></td>
</tr>
<tr>
<td>/WGCNA</td>
<td>Generates the cytoscape network model from WGCNA analysis.</td>
</tr>
</tbody>
</table>
<h2 id="Commands"><a href="#Commands" class="headerlink" title="Commands"></a>Commands</h2><hr>
<h5 id="Help-for-command-‘-Associate-GI’"><a href="#Help-for-command-‘-Associate-GI’" class="headerlink" title="Help for command ‘/Associate.GI’:"></a>Help for command ‘/Associate.GI’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 AssociateGI(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Associate.GI /in &lt;list.Csv.DIR&gt; /gi &lt;nt.gi.csv&gt; [/out &lt;out.DIR&gt;]</span><br><span class="line">Example:      RNA-seq /Associate.GI</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Clustering’"><a href="#Help-for-command-‘-Clustering’" class="headerlink" title="Help for command ‘/Clustering’:"></a>Help for command ‘/Clustering’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 Clustering(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Clustering /in &lt;fq&gt; /kmax &lt;int&gt; [/out &lt;out.Csv&gt;]</span><br><span class="line">Example:      RNA-seq /Clustering</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Co-Vector’"><a href="#Help-for-command-‘-Co-Vector’" class="headerlink" title="Help for command ‘/Co.Vector’:"></a>Help for command ‘/Co.Vector’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 CorrelatesVector(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Co.Vector /in &lt;co.Csv/DIR&gt; [/min 0.01 /max 0.05 /out &lt;out.csv&gt;]</span><br><span class="line">Example:      RNA-seq /Co.Vector</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Contacts-Ref’"><a href="#Help-for-command-‘-Contacts-Ref’" class="headerlink" title="Help for command ‘/Contacts.Ref’:"></a>Help for command ‘/Contacts.Ref’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 Contacts(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Contacts.Ref /in &lt;in.fasta&gt; /maps &lt;maps.sam&gt; [/out &lt;out.DIR&gt;]</span><br><span class="line">Example:      RNA-seq /Contacts.Ref</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Data-Frame’"><a href="#Help-for-command-‘-Data-Frame’" class="headerlink" title="Help for command ‘/Data.Frame’:"></a>Help for command ‘/Data.Frame’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 Df(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Generates the data input for the DESeq2 R package.</span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Data.Frame /in &lt;in.DIR&gt; /ptt &lt;genome.ptt&gt; [/out out.csv]</span><br><span class="line">Example:      RNA-seq /Data.Frame</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">/in</span><br><span class="line">Description:  A directory location which it contains the Ht-Seq raw count text files.</span><br><span class="line"></span><br><span class="line">Example:      /in &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h4 id="Accepted-Types"><a href="#Accepted-Types" class="headerlink" title="Accepted Types"></a>Accepted Types</h4><h5 id="in"><a href="#in" class="headerlink" title="/in"></a>/in</h5><h5 id="Help-for-command-‘-DataFrame-RPKMs’"><a href="#Help-for-command-‘-DataFrame-RPKMs’" class="headerlink" title="Help for command ‘/DataFrame.RPKMs’:"></a>Help for command ‘/DataFrame.RPKMs’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 MergeRPKMs(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Merges the RPKM csv data files.</span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /DataFrame.RPKMs /in &lt;in.DIR&gt; [/trim /out &lt;out.csv&gt;]</span><br><span class="line">Example:      RNA-seq /DataFrame.RPKMs</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-DEGs’"><a href="#Help-for-command-‘-DEGs’" class="headerlink" title="Help for command ‘/DEGs’:"></a>Help for command ‘/DEGs’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 DEGs(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /DEGs /in &lt;diff.csv&gt; [/out &lt;degs.csv&gt; /log_fold 2]</span><br><span class="line">Example:      RNA-seq /DEGs</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-DEGs-UpDown’"><a href="#Help-for-command-‘-DEGs-UpDown’" class="headerlink" title="Help for command ‘/DEGs.UpDown’:"></a>Help for command ‘/DEGs.UpDown’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 DEGsUpDown(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /DEGs.UpDown /in &lt;diff.csv&gt; /sample.table &lt;sampleTable.Csv&gt; [/out &lt;outDIR&gt;]</span><br><span class="line">Example:      RNA-seq /DEGs.UpDown</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-DESeq2’"><a href="#Help-for-command-‘-DESeq2’" class="headerlink" title="Help for command ‘/DESeq2’:"></a>Help for command ‘/DESeq2’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 DESeq2(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /DESeq2 /sample.table &lt;sampleTable.csv&gt; /raw &lt;raw.txt.DIR&gt; /ptt &lt;genome.ptt&gt; [/design &lt;design, default: ~condition&gt;]</span><br><span class="line">Example:      RNA-seq /DESeq2</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-DOOR-Corrects’"><a href="#Help-for-command-‘-DOOR-Corrects’" class="headerlink" title="Help for command ‘/DOOR.Corrects’:"></a>Help for command ‘/DOOR.Corrects’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 DOORCorrects(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /DOOR.Corrects /DOOR &lt;genome.opr&gt; /pcc &lt;pcc.dat&gt; [/out &lt;out.opr&gt; /pcc-cut &lt;0.45&gt;]</span><br><span class="line">Example:      RNA-seq /DOOR.Corrects</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Export-Megan-BIOM’"><a href="#Help-for-command-‘-Export-Megan-BIOM’" class="headerlink" title="Help for command ‘/Export.Megan.BIOM’:"></a>Help for command ‘/Export.Megan.BIOM’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 ExportToMegan(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Export.Megan.BIOM /in &lt;relative.table.csv&gt; [/out &lt;out.json.biom&gt;]</span><br><span class="line">Example:      RNA-seq /Export.Megan.BIOM</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Export-SAM-Maps’"><a href="#Help-for-command-‘-Export-SAM-Maps’" class="headerlink" title="Help for command ‘/Export.SAM.Maps’:"></a>Help for command ‘/Export.SAM.Maps’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 ExportSAMMaps(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Export.SAM.Maps /in &lt;in.sam&gt; [/contigs &lt;NNNN.contig.Csv&gt; /raw &lt;ref.fasta&gt; /out &lt;out.Csv&gt;]</span><br><span class="line">Example:      RNA-seq /Export.SAM.Maps</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Export-SSU-Refs’"><a href="#Help-for-command-‘-Export-SSU-Refs’" class="headerlink" title="Help for command ‘/Export.SSU.Refs’:"></a>Help for command ‘/Export.SSU.Refs’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 ExportSSURefs(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Export.SSU.Refs /in &lt;ssu.fasta&gt; [/out &lt;out.DIR&gt; /no-suffix]</span><br><span class="line">Example:      RNA-seq /Export.SSU.Refs</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Export-SSU-Refs-Batch’"><a href="#Help-for-command-‘-Export-SSU-Refs-Batch’" class="headerlink" title="Help for command ‘/Export.SSU.Refs.Batch’:"></a>Help for command ‘/Export.SSU.Refs.Batch’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 ExportSSUBatch(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Export.SSU.Refs /in &lt;ssu.fasta.DIR&gt; [/out &lt;out.DIR&gt;]</span><br><span class="line">Example:      RNA-seq /Export.SSU.Refs.Batch</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Format-GI’"><a href="#Help-for-command-‘-Format-GI’" class="headerlink" title="Help for command ‘/Format.GI’:"></a>Help for command ‘/Format.GI’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 FormatGI(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Format.GI /in &lt;txt&gt; /gi &lt;regex&gt; /format &lt;gi|&#123;gi&#125;&gt; /out &lt;out.txt&gt;</span><br><span class="line">Example:      RNA-seq /Format.GI</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-fq2fa’"><a href="#Help-for-command-‘-fq2fa’" class="headerlink" title="Help for command ‘/fq2fa’:"></a>Help for command ‘/fq2fa’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 Fq2fa(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /fq2fa /in &lt;fastaq&gt; [/out &lt;fasta&gt;]</span><br><span class="line">Example:      RNA-seq /fq2fa</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-gast’"><a href="#Help-for-command-‘-gast’" class="headerlink" title="Help for command ‘/gast’:"></a>Help for command ‘/gast’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 gastInvoke(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe </span><br><span class="line">Example:      RNA-seq /gast</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-gast-stat-names’"><a href="#Help-for-command-‘-gast-stat-names’" class="headerlink" title="Help for command ‘/gast.stat.names’:"></a>Help for command ‘/gast.stat.names’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 StateNames(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /gast.stat.names /in &lt;*.names&gt; /gast &lt;gast.out&gt; [/out &lt;out.Csv&gt;]</span><br><span class="line">Example:      RNA-seq /gast.stat.names</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Group-n’"><a href="#Help-for-command-‘-Group-n’" class="headerlink" title="Help for command ‘/Group.n’:"></a>Help for command ‘/Group.n’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 GroupN(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Group.n /in &lt;dataset.csv&gt; [/locus_map &lt;locus&gt; /out &lt;out.csv&gt;]</span><br><span class="line">Example:      RNA-seq /Group.n</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-HT-seq’"><a href="#Help-for-command-‘-HT-seq’" class="headerlink" title="Help for command ‘/HT-seq’:"></a>Help for command ‘/HT-seq’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 HTSeqCount(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Count raw reads for DESeq2 analysis.</span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Ht-seq /in &lt;in.sam&gt; /gff &lt;genome.gff&gt; [/out &lt;out.txt&gt; /mode &lt;union, intersection_strict, intersection_nonempty; default:intersection_nonempty&gt; /rpkm /feature &lt;CDS&gt;]</span><br><span class="line">Example:      RNA-seq /HT-seq</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">    [/Mode]</span><br><span class="line"> Description:  The value of this parameter specific the counter of the function will be used, the available counter values are: union, intersection_strict and intersection_nonempty</span><br><span class="line"></span><br><span class="line"> Example:      /Mode &quot;&quot;</span><br><span class="line"></span><br><span class="line">[/feature]</span><br><span class="line"> Description:  [NOTE: value is case sensitive!!!] Value of the gff features can be one of the: tRNA, CDS, exon, gene, tmRNA, rRNA, region</span><br><span class="line"></span><br><span class="line"> Example:      /feature &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h4 id="Accepted-Types-1"><a href="#Accepted-Types-1" class="headerlink" title="Accepted Types"></a>Accepted Types</h4><h5 id="Mode"><a href="#Mode" class="headerlink" title="/Mode"></a>/Mode</h5><h5 id="feature"><a href="#feature" class="headerlink" title="/feature"></a>/feature</h5><h5 id="Help-for-command-‘-Imports-gast-Refs-NCBI-nt’"><a href="#Help-for-command-‘-Imports-gast-Refs-NCBI-nt’" class="headerlink" title="Help for command ‘/Imports.gast.Refs.NCBI_nt’:"></a>Help for command ‘/Imports.gast.Refs.NCBI_nt’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 ImportsRefFromNt(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Imports.gast.Refs.NCBI_nt /in &lt;in.nt&gt; /gi2taxid &lt;dmp/txt/bin&gt; /taxonomy &lt;nodes/names.dmp_DIR&gt; [/out &lt;out.fasta&gt;]</span><br><span class="line">Example:      RNA-seq /Imports.gast.Refs.NCBI_nt</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-log2-selects’"><a href="#Help-for-command-‘-log2-selects’" class="headerlink" title="Help for command ‘/log2.selects’:"></a>Help for command ‘/log2.selects’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 Log2Selects(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /log2.selects /log2 &lt;rpkm.log2.csv&gt; /data &lt;dataset.csv&gt; [/locus_map &lt;locus&gt; /factor 1 /out &lt;out.dataset.csv&gt;]</span><br><span class="line">Example:      RNA-seq /log2.selects</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-PCC’"><a href="#Help-for-command-‘-PCC’" class="headerlink" title="Help for command ‘/PCC’:"></a>Help for command ‘/PCC’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 PCC(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /PCC /expr &lt;expr.matrix.csv&gt; [/out &lt;out.dat&gt;]</span><br><span class="line">Example:      RNA-seq /PCC</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Rank-Statics’"><a href="#Help-for-command-‘-Rank-Statics’" class="headerlink" title="Help for command ‘/Rank.Statics’:"></a>Help for command ‘/Rank.Statics’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 RankStatics(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Rank.Statics /in &lt;relative.table.csv&gt; [/out &lt;EXPORT_DIR&gt;]</span><br><span class="line">Example:      RNA-seq /Rank.Statics</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-RPKM’"><a href="#Help-for-command-‘-RPKM’" class="headerlink" title="Help for command ‘/RPKM’:"></a>Help for command ‘/RPKM’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 RPKM(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /RPKM /raw &lt;raw_count.txt&gt; /gff &lt;genome.gff&gt; [/out &lt;expr.out.csv&gt;]</span><br><span class="line">Example:      RNA-seq /RPKM</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-RPKM-Log2’"><a href="#Help-for-command-‘-RPKM-Log2’" class="headerlink" title="Help for command ‘/RPKM.Log2’:"></a>Help for command ‘/RPKM.Log2’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 Log2(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /RPKM.Log2 /in &lt;RPKM.csv&gt; /cond &lt;conditions&gt; [/out &lt;out.csv&gt;]</span><br><span class="line">Example:      RNA-seq /RPKM.Log2</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">/cond</span><br><span class="line">Description:  Syntax format as:  &lt;experiment1&gt;/&lt;experiment2&gt;|&lt;experiment3&gt;/&lt;experiment4&gt;|.....</span><br><span class="line"></span><br><span class="line">Example:      /cond &quot;colR1/xcb1|colR2/xcb2&quot;</span><br></pre></td></tr></table></figure></p>
<h4 id="Accepted-Types-2"><a href="#Accepted-Types-2" class="headerlink" title="Accepted Types"></a>Accepted Types</h4><h5 id="cond"><a href="#cond" class="headerlink" title="/cond"></a>/cond</h5><h5 id="Help-for-command-‘-sid-map’"><a href="#Help-for-command-‘-sid-map’" class="headerlink" title="Help for command ‘/sid.map’:"></a>Help for command ‘/sid.map’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 sIdMapping(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /sid.map /gff &lt;genome.gff&gt; /raw &lt;htseq-count.txt&gt; [/out &lt;out.txt&gt;]</span><br><span class="line">Example:      RNA-seq /sid.map</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-SPCC’"><a href="#Help-for-command-‘-SPCC’" class="headerlink" title="Help for command ‘/SPCC’:"></a>Help for command ‘/SPCC’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 SPCC(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /SPCC /expr &lt;expr.matrix.csv&gt; [/out &lt;out.dat&gt;]</span><br><span class="line">Example:      RNA-seq /SPCC</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Stat-Changes’"><a href="#Help-for-command-‘-Stat-Changes’" class="headerlink" title="Help for command ‘/Stat.Changes’:"></a>Help for command ‘/Stat.Changes’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 StatChanges(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Stat.Changes /deseq &lt;deseq.result.csv&gt; /sample &lt;sampletable.csv&gt; [/out &lt;out.csv&gt; /levels &lt;1000&gt; /diff &lt;0.5&gt;]</span><br><span class="line">Example:      RNA-seq /Stat.Changes</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Statics-Labels’"><a href="#Help-for-command-‘-Statics-Labels’" class="headerlink" title="Help for command ‘/Statics.Labels’:"></a>Help for command ‘/Statics.Labels’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 MergeLabels(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /Statics.Labels /in &lt;in.csv&gt; [/label &lt;Label&gt; /name &lt;Name&gt; /value &lt;value&gt; /out &lt;out.csv&gt;]</span><br><span class="line">Example:      RNA-seq /Statics.Labels</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-WGCNA’"><a href="#Help-for-command-‘-WGCNA’" class="headerlink" title="Help for command ‘/WGCNA’:"></a>Help for command ‘/WGCNA’:</h5><p><strong>Prototype</strong>: RNA_seq.CLI::Int32 FromWGCNA(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Generates the cytoscape network model from WGCNA analysis.</span><br><span class="line">Usage:        G:\GCModeller\manual\bin\RNA-seq.exe /WGCNA /data &lt;dataExpr.csv&gt; /anno &lt;annotations.csv&gt; [/out &lt;DIR.Out&gt; /mods &lt;color.list&gt; /from.DESeq /id.map &lt;GeneId&gt;]</span><br><span class="line">Example:      RNA-seq /WGCNA</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br></pre></td><td class="code"><pre><span class="line">    /data</span><br><span class="line">    Description:  A sets of RNA-seq RPKM expression data sets, the first row in the csv table should be the experiments or conditions, and first column in the table should be the id of the genes and each cell in the table should be the RPKM expression value of each gene in each condition.</span><br><span class="line">                   The data format of the table it would be like:</span><br><span class="line">                   GeneId, condi1, cond12, condi3, ....</span><br><span class="line">                   locus1, x, xx, x,</span><br><span class="line">                   locus2, y, yy, yyy,</span><br><span class="line">                   locus3, ,z, zz, zzz,</span><br><span class="line">                   ......</span><br><span class="line"></span><br><span class="line">    xyz Is the RPKM of the genes</span><br><span class="line"></span><br><span class="line">    Example:      /data &quot;&quot;</span><br><span class="line"></span><br><span class="line">/anno</span><br><span class="line">    Description:  A table of the gene name annotation, the table should be in formats of</span><br><span class="line">    Id, gene_symbol</span><br><span class="line">    locus1, geneName</span><br><span class="line">    locus2, geneName</span><br><span class="line">    ....</span><br><span class="line"></span><br><span class="line">    Example:      /anno &quot;&quot;</span><br><span class="line"></span><br><span class="line">   [/mods]</span><br><span class="line">    Description:  Each color in this parameter value is stands for a co expression module, and this parameter controls of the module output filtering, using | character as the seperator for each module color.</span><br><span class="line"></span><br><span class="line">    Example:      /mods &quot;&quot;</span><br><span class="line"></span><br><span class="line">   [/out]</span><br><span class="line">    Description:  Export directory of the WGCNA data, if this parameter value is not presents in the arguments, then the current work directory will be used.</span><br><span class="line"></span><br><span class="line">    Example:      /out &quot;&quot;</span><br><span class="line"></span><br><span class="line">   [/From.Deseq]</span><br><span class="line">    Description:  Is the /data matrix if comes from the DESeq analysis result output?</span><br><span class="line">                   If is true, then the expression value will be extract from the original matrix file and save a new file named DESeq.dataExpr0.Csv in the out directory,</span><br><span class="line">                   and last using this extracted data as the source of the WGCNA R script.</span><br><span class="line"></span><br><span class="line">    Example:      /From.Deseq &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h4 id="Accepted-Types-3"><a href="#Accepted-Types-3" class="headerlink" title="Accepted Types"></a>Accepted Types</h4><h5 id="data"><a href="#data" class="headerlink" title="/data"></a>/data</h5><h5 id="anno"><a href="#anno" class="headerlink" title="/anno"></a>/anno</h5><h5 id="mods"><a href="#mods" class="headerlink" title="/mods"></a>/mods</h5><h5 id="out"><a href="#out" class="headerlink" title="/out"></a>/out</h5><h5 id="From-Deseq"><a href="#From-Deseq" class="headerlink" title="/From.Deseq"></a>/From.Deseq</h5>
              
            </div>
            <footer class="article-footer">
              <a href="/tools/RNA-seq.html" class="article-date">
  <time datetime="2016-07-27T10:40:24.000Z" itemprop="datePublished">2016-07-27</time>
</a>
              <a data-url="http://docs.gcmodeller.org/tools/RNA-seq.html" data-id="cipe3i2ul000c20v4p7b6dh9v" class="article-share-link">Share</a>
              
              
  <ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/">undefined</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/">undefined</a></li></ul>

            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

