

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>tools | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="GCModeller [version 3.0.2456.4506]Module AssemblyName: file:///F:/bin-Debug/seqtools.exeRoot namespace: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities
All of the command that available in t">
<meta name="keywords">
<meta property="og:type" content="website">
<meta property="og:title" content="tools">
<meta property="og:url" content="http://docs.gcmodeller.org/tools/seqtools.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="GCModeller [version 3.0.2456.4506]Module AssemblyName: file:///F:/bin-Debug/seqtools.exeRoot namespace: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities
All of the command that available in t">
<meta property="og:updated_time" content="2016-06-12T15:46:00.287Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="tools">
<meta name="twitter:description" content="GCModeller [version 3.0.2456.4506]Module AssemblyName: file:///F:/bin-Debug/seqtools.exeRoot namespace: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities
All of the command that available in t">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="/css/favicon.ico">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tags">
                  
                  

                
                  Tags
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/archives">
                  
                  

                
                  Archives
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/tools/seqtools.html">
      <h1 class="article-title" itemprop="name">
        tools
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/tools/seqtools.html" class="article-date">
  <time datetime="2016-06-12T15:44:40.000Z" itemprop="datePublished">2016-06-12</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/tools/seqtools.html">
      <h1 class="article-title" itemprop="name">
        tools
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#GCModeller-version-3-0-2456-4506"><span class="toc-number">1.</span> <span class="toc-text">GCModeller [version 3.0.2456.4506]</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Commands"><span class="toc-number">1.1.</span> <span class="toc-text">Commands</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-align’"><span class="toc-number">1.1.0.0.1.</span> <span class="toc-text">Help for command ‘/align’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Clustal-Cut’"><span class="toc-number">1.1.0.0.2.</span> <span class="toc-text">Help for command ‘/Clustal.Cut’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Distinct’"><span class="toc-number">1.1.0.0.3.</span> <span class="toc-text">Help for command ‘/Distinct’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Get-Locis’"><span class="toc-number">1.1.0.0.4.</span> <span class="toc-text">Help for command ‘/Get.Locis’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Loci-describ’"><span class="toc-number">1.1.0.0.5.</span> <span class="toc-text">Help for command ‘/Loci.describ’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-logo’"><span class="toc-number">1.1.0.0.6.</span> <span class="toc-text">Help for command ‘/logo’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Merge’"><span class="toc-number">1.1.0.0.7.</span> <span class="toc-text">Help for command ‘/Merge’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Mirror-Batch’"><span class="toc-number">1.1.0.0.8.</span> <span class="toc-text">Help for command ‘/Mirror.Batch’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Mirror-Vector’"><span class="toc-number">1.1.0.0.9.</span> <span class="toc-text">Help for command ‘/Mirror.Vector’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-nw’"><span class="toc-number">1.1.0.0.10.</span> <span class="toc-text">Help for command ‘/nw’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Palindrome-BatchTask’"><span class="toc-number">1.1.0.0.11.</span> <span class="toc-text">Help for command ‘/Palindrome.BatchTask’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Palindrome-Workflow’"><span class="toc-number">1.1.0.0.12.</span> <span class="toc-text">Help for command ‘/Palindrome.Workflow’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Select-By-Locus’"><span class="toc-number">1.1.0.0.13.</span> <span class="toc-text">Help for command ‘/Select.By_Locus’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-SimpleSegment-AutoBuild’"><span class="toc-number">1.1.0.0.14.</span> <span class="toc-text">Help for command ‘/SimpleSegment.AutoBuild’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-SimpleSegment-Mirrors’"><span class="toc-number">1.1.0.0.15.</span> <span class="toc-text">Help for command ‘/SimpleSegment.Mirrors’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Split’"><span class="toc-number">1.1.0.0.16.</span> <span class="toc-text">Help for command ‘/Split’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-subset’"><span class="toc-number">1.1.0.0.17.</span> <span class="toc-text">Help for command ‘/subset’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-To-Fasta’"><span class="toc-number">1.1.0.0.18.</span> <span class="toc-text">Help for command ‘/To_Fasta’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-TrimTest’"><span class="toc-number">1.1.0.0.19.</span> <span class="toc-text">Help for command ‘/TrimTest’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-Write-Seeds’"><span class="toc-number">1.1.0.0.20.</span> <span class="toc-text">Help for command ‘/Write.Seeds’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-321’"><span class="toc-number">1.1.0.0.21.</span> <span class="toc-text">Help for command ‘-321’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–align’"><span class="toc-number">1.1.0.0.22.</span> <span class="toc-text">Help for command ‘–align’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–align-Self’"><span class="toc-number">1.1.0.0.23.</span> <span class="toc-text">Help for command ‘–align.Self’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-complement’"><span class="toc-number">1.1.0.0.24.</span> <span class="toc-text">Help for command ‘-complement’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–Drawing-ClustalW’"><span class="toc-number">1.1.0.0.25.</span> <span class="toc-text">Help for command ‘–Drawing.ClustalW’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–Hairpinks’"><span class="toc-number">1.1.0.0.26.</span> <span class="toc-text">Help for command ‘–Hairpinks’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–Hairpinks-batch-task’"><span class="toc-number">1.1.0.0.27.</span> <span class="toc-text">Help for command ‘–Hairpinks.batch.task’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–ImperfectsPalindrome-batch-Task’"><span class="toc-number">1.1.0.0.28.</span> <span class="toc-text">Help for command ‘–ImperfectsPalindrome.batch.Task’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–Mirror-From-Fasta’"><span class="toc-number">1.1.0.0.29.</span> <span class="toc-text">Help for command ‘–Mirror.From.Fasta’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–Mirror-From-NT’"><span class="toc-number">1.1.0.0.30.</span> <span class="toc-text">Help for command ‘–Mirror.From.NT’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–Palindrome-batch-Task’"><span class="toc-number">1.1.0.0.31.</span> <span class="toc-text">Help for command ‘–Palindrome.batch.Task’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–Palindrome-From-FASTA’"><span class="toc-number">1.1.0.0.32.</span> <span class="toc-text">Help for command ‘–Palindrome.From.FASTA’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–Palindrome-From-NT’"><span class="toc-number">1.1.0.0.33.</span> <span class="toc-text">Help for command ‘–Palindrome.From.NT’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–Palindrome-Imperfects’"><span class="toc-number">1.1.0.0.34.</span> <span class="toc-text">Help for command ‘–Palindrome.Imperfects’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-pattern-search’"><span class="toc-number">1.1.0.0.35.</span> <span class="toc-text">Help for command ‘-pattern_search’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–PerfectPalindrome-Filtering’"><span class="toc-number">1.1.0.0.36.</span> <span class="toc-text">Help for command ‘–PerfectPalindrome.Filtering’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘Repeats-Density’"><span class="toc-number">1.1.0.0.37.</span> <span class="toc-text">Help for command ‘Repeats.Density’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-reverse’"><span class="toc-number">1.1.0.0.38.</span> <span class="toc-text">Help for command ‘-reverse’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘rev-Repeats-Density’"><span class="toc-number">1.1.0.0.39.</span> <span class="toc-text">Help for command ‘rev-Repeats.Density’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘Search-Batch’"><span class="toc-number">1.1.0.0.40.</span> <span class="toc-text">Help for command ‘Search.Batch’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘-segment’"><span class="toc-number">1.1.0.0.41.</span> <span class="toc-text">Help for command ‘-segment’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–segments’"><span class="toc-number">1.1.0.0.42.</span> <span class="toc-text">Help for command ‘–segments’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–ToVector’"><span class="toc-number">1.1.0.0.43.</span> <span class="toc-text">Help for command ‘–ToVector’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–translates’"><span class="toc-number">1.1.0.0.44.</span> <span class="toc-text">Help for command ‘–translates’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘–Trim’"><span class="toc-number">1.1.0.0.45.</span> <span class="toc-text">Help for command ‘–Trim’:</span></a></li></ol></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="GCModeller-version-3-0-2456-4506"><a href="#GCModeller-version-3-0-2456-4506" class="headerlink" title="GCModeller [version 3.0.2456.4506]"></a>GCModeller [version 3.0.2456.4506]</h1><p><strong>Module AssemblyName</strong>: file:///F:/bin-Debug/seqtools.exe<br><strong>Root namespace</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities</p>
<p>All of the command that available in this program has been list below:</p>
<table>
<thead>
<tr>
<th>Function API</th>
<th>Info</th>
</tr>
</thead>
<tbody>
<tr>
<td>/align</td>
<td></td>
</tr>
<tr>
<td>/Clustal.Cut</td>
<td></td>
</tr>
<tr>
<td>/Distinct</td>
<td></td>
</tr>
<tr>
<td>/Get.Locis</td>
<td></td>
</tr>
<tr>
<td>/Loci.describ</td>
<td>Testing</td>
</tr>
<tr>
<td>/logo</td>
<td>* Drawing the sequence logo from the clustal alignment result.</td>
</tr>
<tr>
<td>/Merge</td>
<td>Only search for 1 level folder, dit not search receve.</td>
</tr>
<tr>
<td>/Mirror.Batch</td>
<td></td>
</tr>
<tr>
<td>/Mirror.Vector</td>
<td></td>
</tr>
<tr>
<td>/nw</td>
<td></td>
</tr>
<tr>
<td>/Palindrome.BatchTask</td>
<td></td>
</tr>
<tr>
<td>/Palindrome.Workflow</td>
<td></td>
</tr>
<tr>
<td>/Select.By_Locus</td>
<td></td>
</tr>
<tr>
<td>/SimpleSegment.AutoBuild</td>
<td></td>
</tr>
<tr>
<td>/SimpleSegment.Mirrors</td>
<td></td>
</tr>
<tr>
<td>/Split</td>
<td></td>
</tr>
<tr>
<td>/subset</td>
<td></td>
</tr>
<tr>
<td>/To_Fasta</td>
<td>Convert the sequence data in a excel annotation file into a fasta sequence file.</td>
</tr>
<tr>
<td>/TrimTest</td>
<td></td>
</tr>
<tr>
<td>/Write.Seeds</td>
<td></td>
</tr>
<tr>
<td>-321</td>
<td>Polypeptide sequence 3 letters to 1 lettes sequence.</td>
</tr>
<tr>
<td>–align</td>
<td></td>
</tr>
<tr>
<td>–align.Self</td>
<td></td>
</tr>
<tr>
<td>-complement</td>
<td></td>
</tr>
<tr>
<td>–Drawing.ClustalW</td>
<td></td>
</tr>
<tr>
<td>–Hairpinks</td>
<td></td>
</tr>
<tr>
<td>–Hairpinks.batch.task</td>
<td></td>
</tr>
<tr>
<td>–ImperfectsPalindrome.batch.Task</td>
<td></td>
</tr>
<tr>
<td>–Mirror.From.Fasta</td>
<td>Mirror Palindrome, search from a fasta file.</td>
</tr>
<tr>
<td>–Mirror.From.NT</td>
<td>Mirror Palindrome, and this function is for the debugging test</td>
</tr>
<tr>
<td>–Palindrome.batch.Task</td>
<td></td>
</tr>
<tr>
<td>–Palindrome.From.FASTA</td>
<td></td>
</tr>
<tr>
<td>–Palindrome.From.NT</td>
<td>This function is just for debugger test, /nt parameter is the nucleotide sequence data as ATGCCCC</td>
</tr>
<tr>
<td>–Palindrome.Imperfects</td>
<td></td>
</tr>
<tr>
<td>-pattern_search</td>
<td>Parsing the sequence segment from the sequence source using regular expression.</td>
</tr>
<tr>
<td>–PerfectPalindrome.Filtering</td>
<td></td>
</tr>
<tr>
<td>Repeats.Density</td>
<td></td>
</tr>
<tr>
<td>-reverse</td>
<td></td>
</tr>
<tr>
<td>rev-Repeats.Density</td>
<td></td>
</tr>
<tr>
<td>Search.Batch</td>
<td>Batch search for repeats.</td>
</tr>
<tr>
<td>-segment</td>
<td></td>
</tr>
<tr>
<td>–segments</td>
<td></td>
</tr>
<tr>
<td>–ToVector</td>
<td></td>
</tr>
<tr>
<td>–translates</td>
<td>Translates the ORF gene as protein sequence. If any error was output from the console, please using &gt; operator dump the output to a log file for the analysis.</td>
</tr>
<tr>
<td>–Trim</td>
</tr>
</tbody>
</table>
<h2 id="Commands"><a href="#Commands" class="headerlink" title="Commands"></a>Commands</h2><hr>
<h5 id="Help-for-command-‘-align’"><a href="#Help-for-command-‘-align’" class="headerlink" title="Help for command ‘/align’:"></a>Help for command ‘/align’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 Align2(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /align /query &lt;query.fasta&gt; /subject &lt;subject.fasta&gt; [/blosum &lt;matrix.txt&gt; /out &lt;out.xml&gt;]</span><br><span class="line">Example:      seqtools /align</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Clustal-Cut’"><a href="#Help-for-command-‘-Clustal-Cut’" class="headerlink" title="Help for command ‘/Clustal.Cut’:"></a>Help for command ‘/Clustal.Cut’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 CutMlAlignment(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Clustal.Cut /in &lt;in.fasta&gt; [/left 0.1 /right 0.1 /out &lt;out.fasta&gt;]</span><br><span class="line">Example:      seqtools /Clustal.Cut</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Distinct’"><a href="#Help-for-command-‘-Distinct’" class="headerlink" title="Help for command ‘/Distinct’:"></a>Help for command ‘/Distinct’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 Distinct(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Distinct /in &lt;in.fasta&gt; [/out &lt;out.fasta&gt;]</span><br><span class="line">Example:      seqtools /Distinct</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Get-Locis’"><a href="#Help-for-command-‘-Get-Locis’" class="headerlink" title="Help for command ‘/Get.Locis’:"></a>Help for command ‘/Get.Locis’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 GetSimpleSegments(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Get.Locis /in &lt;locis.csv&gt; /nt &lt;genome.nt.fasta&gt; [/out &lt;outDIR&gt;]</span><br><span class="line">Example:      seqtools /Get.Locis</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Loci-describ’"><a href="#Help-for-command-‘-Loci-describ’" class="headerlink" title="Help for command ‘/Loci.describ’:"></a>Help for command ‘/Loci.describ’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 LociDescript(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Testing</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Loci.describ /ptt &lt;genome-context.ptt&gt; [/test &lt;loci:randomize&gt; /complement /unstrand]</span><br><span class="line">Example:      seqtools /Loci.describ</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-logo’"><a href="#Help-for-command-‘-logo’" class="headerlink" title="Help for command ‘/logo’:"></a>Help for command ‘/logo’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 SequenceLogo(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  * Drawing the sequence logo from the clustal alignment result.</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /logo /in &lt;clustal.fasta&gt; [/out &lt;out.png&gt; /title &quot;&quot;]</span><br><span class="line">Example:      seqtools /logo</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"> /in</span><br><span class="line"> Description:  The file path of the clustal output fasta file.</span><br><span class="line"></span><br><span class="line"> Example:      /in &quot;&quot;</span><br><span class="line"></span><br><span class="line">[/out]</span><br><span class="line"> Description:  The output sequence logo image file path. default is the same name as the input fasta sequence file.</span><br><span class="line"></span><br><span class="line"> Example:      /out &quot;&quot;</span><br><span class="line"></span><br><span class="line">[/title]</span><br><span class="line"> Description:  The display title on the sequence logo, default is using the fasta file name.</span><br><span class="line"></span><br><span class="line"> Example:      /title &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘-Merge’"><a href="#Help-for-command-‘-Merge’" class="headerlink" title="Help for command ‘/Merge’:"></a>Help for command ‘/Merge’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 Merge(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Only search for 1 level folder, dit not search receve.</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Merge /in &lt;fasta.DIR&gt; [/out &lt;out.fasta&gt; /trim /ext &lt;*.fasta&gt; /brief]</span><br><span class="line">Example:      seqtools /Merge</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Mirror-Batch’"><a href="#Help-for-command-‘-Mirror-Batch’" class="headerlink" title="Help for command ‘/Mirror.Batch’:"></a>Help for command ‘/Mirror.Batch’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 MirrorBatch(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Mirror.Batch /nt &lt;nt.fasta&gt; [/out &lt;out.csv&gt; /mp /min &lt;3&gt; /max &lt;20&gt; /num_threads &lt;-1&gt;]</span><br><span class="line">Example:      seqtools /Mirror.Batch</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">   [/mp]</span><br><span class="line">Description:  Calculation in the multiple process mode?</span><br><span class="line"></span><br><span class="line">Example:      /mp &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘-Mirror-Vector’"><a href="#Help-for-command-‘-Mirror-Vector’" class="headerlink" title="Help for command ‘/Mirror.Vector’:"></a>Help for command ‘/Mirror.Vector’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 MirrorsVector(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Mirror.Vector /in &lt;inDIR&gt; /size &lt;genome.size&gt; [/out out.txt]</span><br><span class="line">Example:      seqtools /Mirror.Vector</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-nw’"><a href="#Help-for-command-‘-nw’" class="headerlink" title="Help for command ‘/nw’:"></a>Help for command ‘/nw’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 NW(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /nw /query &lt;query.fasta&gt; /subject &lt;subject.fasta&gt; /out &lt;out.txt&gt;</span><br><span class="line">Example:      seqtools /nw</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Palindrome-BatchTask’"><a href="#Help-for-command-‘-Palindrome-BatchTask’" class="headerlink" title="Help for command ‘/Palindrome.BatchTask’:"></a>Help for command ‘/Palindrome.BatchTask’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 PalindromeBatchTask(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Palindrome.BatchTask /in &lt;in.DIR&gt; [/num_threads 4 /min 3 /max 20 /min-appears 2 /cutoff &lt;0.6&gt; /Palindrome /max-dist &lt;1000 (bp)&gt; /partitions &lt;-1&gt; /out &lt;out.DIR&gt;]</span><br><span class="line">Example:      seqtools /Palindrome.BatchTask</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">   [/Palindrome]</span><br><span class="line">Description:  Only search for Palindrome, not includes the repeats data.</span><br><span class="line"></span><br><span class="line">Example:      /Palindrome &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘-Palindrome-Workflow’"><a href="#Help-for-command-‘-Palindrome-Workflow’" class="headerlink" title="Help for command ‘/Palindrome.Workflow’:"></a>Help for command ‘/Palindrome.Workflow’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 PalindromeWorkflow(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Palindrome.Workflow /in &lt;in.fasta&gt; [/batch /min-appears 2 /min 3 /max 20 /cutoff &lt;0.6&gt; /max-dist &lt;1000 (bp)&gt; /Palindrome /partitions &lt;-1&gt; /out &lt;out.DIR&gt;]</span><br><span class="line">Example:      seqtools /Palindrome.Workflow</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"> /in</span><br><span class="line"> Description:  This is a single sequence fasta file.</span><br><span class="line"></span><br><span class="line"> Example:      /in &quot;&quot;</span><br><span class="line"></span><br><span class="line">[/Palindrome]</span><br><span class="line"> Description:  Only search for Palindrome, not includes the repeats data.</span><br><span class="line"></span><br><span class="line"> Example:      /Palindrome &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘-Select-By-Locus’"><a href="#Help-for-command-‘-Select-By-Locus’" class="headerlink" title="Help for command ‘/Select.By_Locus’:"></a>Help for command ‘/Select.By_Locus’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 SelectByLocus(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Select.By_Locus /in &lt;locus.txt&gt; /fa &lt;fasta.inDIR&gt; [/out &lt;out.fasta&gt;]</span><br><span class="line">Example:      seqtools /Select.By_Locus</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-SimpleSegment-AutoBuild’"><a href="#Help-for-command-‘-SimpleSegment-AutoBuild’" class="headerlink" title="Help for command ‘/SimpleSegment.AutoBuild’:"></a>Help for command ‘/SimpleSegment.AutoBuild’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 ConvertsAuto(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /SimpleSegment.AutoBuild /in &lt;locis.csv&gt; [/out &lt;out.csv&gt;]</span><br><span class="line">Example:      seqtools /SimpleSegment.AutoBuild</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-SimpleSegment-Mirrors’"><a href="#Help-for-command-‘-SimpleSegment-Mirrors’" class="headerlink" title="Help for command ‘/SimpleSegment.Mirrors’:"></a>Help for command ‘/SimpleSegment.Mirrors’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 ConvertMirrors(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE </span><br><span class="line">Example:      seqtools /SimpleSegment.Mirrors</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Split’"><a href="#Help-for-command-‘-Split’" class="headerlink" title="Help for command ‘/Split’:"></a>Help for command ‘/Split’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 Split(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Split /in &lt;in.fasta&gt; [/n &lt;4096&gt; /out &lt;outDIR&gt;]</span><br><span class="line">Example:      seqtools /Split</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-subset’"><a href="#Help-for-command-‘-subset’" class="headerlink" title="Help for command ‘/subset’:"></a>Help for command ‘/subset’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 SubSet(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /subset /lstID &lt;lstID.txt&gt; /fa &lt;source.fasta&gt;</span><br><span class="line">Example:      seqtools /subset</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-To-Fasta’"><a href="#Help-for-command-‘-To-Fasta’" class="headerlink" title="Help for command ‘/To_Fasta’:"></a>Help for command ‘/To_Fasta’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 ToFasta(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Convert the sequence data in a excel annotation file into a fasta sequence file.</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /To_Fasta /in &lt;anno.csv&gt; [/out &lt;out.fasta&gt; /attrs &lt;gene;locus_tag;gi;location,...&gt; /seq &lt;Sequence&gt;]</span><br><span class="line">Example:      seqtools /To_Fasta</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-TrimTest’"><a href="#Help-for-command-‘-TrimTest’" class="headerlink" title="Help for command ‘/TrimTest’:"></a>Help for command ‘/TrimTest’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 TrimTest(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /TrimTest /in &lt;in.fasta&gt;</span><br><span class="line">Example:      seqtools /TrimTest</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-Write-Seeds’"><a href="#Help-for-command-‘-Write-Seeds’" class="headerlink" title="Help for command ‘/Write.Seeds’:"></a>Help for command ‘/Write.Seeds’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 WriteSeeds(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE /Write.Seeds /out &lt;out.dat&gt; [/prot /max &lt;20&gt;]</span><br><span class="line">Example:      seqtools /Write.Seeds</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-321’"><a href="#Help-for-command-‘-321’" class="headerlink" title="Help for command ‘-321’:"></a>Help for command ‘-321’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 PolypeptideBriefs(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Polypeptide sequence 3 letters to 1 lettes sequence.</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE -321 /in &lt;sequence.txt&gt; [/out &lt;out.fasta&gt;]</span><br><span class="line">Example:      seqtools -321</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–align’"><a href="#Help-for-command-‘–align’" class="headerlink" title="Help for command ‘–align’:"></a>Help for command ‘–align’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 Align(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --align /query &lt;query.fasta&gt; /subject &lt;subject.fasta&gt; [/out &lt;out.DIR&gt; /cost &lt;0.7&gt;]</span><br><span class="line">Example:      seqtools --align</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–align-Self’"><a href="#Help-for-command-‘–align-Self’" class="headerlink" title="Help for command ‘–align.Self’:"></a>Help for command ‘–align.Self’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 AlignSelf(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --align.Self /query &lt;query.fasta&gt; /out &lt;out.DIR&gt; [/cost 0.75]</span><br><span class="line">Example:      seqtools --align.Self</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-complement’"><a href="#Help-for-command-‘-complement’" class="headerlink" title="Help for command ‘-complement’:"></a>Help for command ‘-complement’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 Complement(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE -reverse -i &lt;input_fasta&gt; [-o &lt;output_fasta&gt;]</span><br><span class="line">Example:      seqtools -complement</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–Drawing-ClustalW’"><a href="#Help-for-command-‘–Drawing-ClustalW’" class="headerlink" title="Help for command ‘–Drawing.ClustalW’:"></a>Help for command ‘–Drawing.ClustalW’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 DrawClustalW(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --Drawing.ClustalW /in &lt;align.fasta&gt; [/out &lt;out.png&gt; /dot.Size 10]</span><br><span class="line">Example:      seqtools --Drawing.ClustalW</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–Hairpinks’"><a href="#Help-for-command-‘–Hairpinks’" class="headerlink" title="Help for command ‘–Hairpinks’:"></a>Help for command ‘–Hairpinks’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 Hairpinks(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --Hairpinks /in &lt;in.fasta&gt; [/out &lt;out.csv&gt; /min &lt;6&gt; /max &lt;7&gt; /cutoff 3 /max-dist &lt;35 (bp)&gt;]</span><br><span class="line">Example:      seqtools --Hairpinks</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–Hairpinks-batch-task’"><a href="#Help-for-command-‘–Hairpinks-batch-task’" class="headerlink" title="Help for command ‘–Hairpinks.batch.task’:"></a>Help for command ‘–Hairpinks.batch.task’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 HairpinksBatch(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --Hairpinks.batch.task /in &lt;in.fasta&gt; [/out &lt;outDIR&gt; /min &lt;6&gt; /max &lt;7&gt; /cutoff &lt;0.6&gt; /max-dist &lt;35 (bp)&gt; /num_threads &lt;-1&gt;]</span><br><span class="line">Example:      seqtools --Hairpinks.batch.task</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–ImperfectsPalindrome-batch-Task’"><a href="#Help-for-command-‘–ImperfectsPalindrome-batch-Task’" class="headerlink" title="Help for command ‘–ImperfectsPalindrome.batch.Task’:"></a>Help for command ‘–ImperfectsPalindrome.batch.Task’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 BatchSearchImperfectsPalindrome(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --ImperfectsPalindrome.batch.Task /in &lt;in.fasta&gt; /out &lt;outDir&gt; [/min &lt;3&gt; /max &lt;20&gt; /cutoff &lt;0.6&gt; /max-dist &lt;1000 (bp)&gt; /num_threads &lt;-1&gt;]</span><br><span class="line">Example:      seqtools --ImperfectsPalindrome.batch.Task</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–Mirror-From-Fasta’"><a href="#Help-for-command-‘–Mirror-From-Fasta’" class="headerlink" title="Help for command ‘–Mirror.From.Fasta’:"></a>Help for command ‘–Mirror.From.Fasta’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 SearchMirrotFasta(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Mirror Palindrome, search from a fasta file.</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --Mirror.From.Fasta /nt &lt;nt-sequence.fasta&gt; [/out &lt;out.csv&gt; /min &lt;3&gt; /max &lt;20&gt;]</span><br><span class="line">Example:      seqtools --Mirror.From.Fasta</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">/nt</span><br><span class="line">Description:  This fasta file should contains only just one sequence.</span><br><span class="line"></span><br><span class="line">Example:      /nt &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘–Mirror-From-NT’"><a href="#Help-for-command-‘–Mirror-From-NT’" class="headerlink" title="Help for command ‘–Mirror.From.NT’:"></a>Help for command ‘–Mirror.From.NT’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 SearchMirrotNT(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Mirror Palindrome, and this function is for the debugging test</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --Mirror.From.NT /nt &lt;nt-sequence&gt; /out &lt;out.csv&gt; [/min &lt;3&gt; /max &lt;20&gt;]</span><br><span class="line">Example:      seqtools --Mirror.From.NT</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–Palindrome-batch-Task’"><a href="#Help-for-command-‘–Palindrome-batch-Task’" class="headerlink" title="Help for command ‘–Palindrome.batch.Task’:"></a>Help for command ‘–Palindrome.batch.Task’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 BatchSearchPalindrome(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --Palindrome.batch.Task /in &lt;in.fasta&gt; /out &lt;outDir&gt; [/min &lt;3&gt; /max &lt;20&gt; /num_threads &lt;-1&gt;]</span><br><span class="line">Example:      seqtools --Palindrome.batch.Task</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–Palindrome-From-FASTA’"><a href="#Help-for-command-‘–Palindrome-From-FASTA’" class="headerlink" title="Help for command ‘–Palindrome.From.FASTA’:"></a>Help for command ‘–Palindrome.From.FASTA’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 SearchPalindromeFasta(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --Palindrome.From.Fasta /nt &lt;nt-sequence.fasta&gt; [/out &lt;out.csv&gt; /min &lt;3&gt; /max &lt;20&gt;]</span><br><span class="line">Example:      seqtools --Palindrome.From.FASTA</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">/nt</span><br><span class="line">Description:  Fasta sequence file, and this file should just contains only one sequence.</span><br><span class="line"></span><br><span class="line">Example:      /nt &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘–Palindrome-From-NT’"><a href="#Help-for-command-‘–Palindrome-From-NT’" class="headerlink" title="Help for command ‘–Palindrome.From.NT’:"></a>Help for command ‘–Palindrome.From.NT’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 SearchPalindromeNT(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  This function is just for debugger test, /nt parameter is the nucleotide sequence data as ATGCCCC</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --Palindrome.From.NT /nt &lt;nt-sequence&gt; /out &lt;out.csv&gt; [/min &lt;3&gt; /max &lt;20&gt;]</span><br><span class="line">Example:      seqtools --Palindrome.From.NT</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–Palindrome-Imperfects’"><a href="#Help-for-command-‘–Palindrome-Imperfects’" class="headerlink" title="Help for command ‘–Palindrome.Imperfects’:"></a>Help for command ‘–Palindrome.Imperfects’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 ImperfectPalindrome(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --Palindrome.Imperfects /in &lt;in.fasta&gt; [/out &lt;out.csv&gt; /min &lt;3&gt; /max &lt;20&gt; /cutoff &lt;0.6&gt; /max-dist &lt;1000 (bp)&gt; /partitions &lt;-1&gt;]</span><br><span class="line">Example:      seqtools --Palindrome.Imperfects</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-pattern-search’"><a href="#Help-for-command-‘-pattern-search’" class="headerlink" title="Help for command ‘-pattern_search’:"></a>Help for command ‘-pattern_search’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 PatternSearchA(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Parsing the sequence segment from the sequence source using regular expression.</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE -pattern_search -i &lt;file_name&gt; -p &lt;regex_pattern&gt;[ -o &lt;output_directory&gt; -f &lt;format:fsa/gbk&gt;]</span><br><span class="line">Example:      seqtools -pattern_search -pattern_search -i ~/xcc8004.txt -p TTA&#123;3&#125;N&#123;1,2&#125; -f fsa</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">    -i</span><br><span class="line">    Description:  The sequence input data source file, it can be a fasta or genbank file.</span><br><span class="line"></span><br><span class="line">    Example:      -i &quot;~/Desktop/xcc8004.txt&quot;</span><br><span class="line"></span><br><span class="line">-p</span><br><span class="line">    Description:  This switch specific the regular expression pattern for search the sequence segment,</span><br><span class="line">              for more detail information about the regular expression please read the user manual.</span><br><span class="line"></span><br><span class="line">    Example:      -p &quot;N&#123;1,5&#125;TA&quot;</span><br><span class="line"></span><br><span class="line">   [-o]</span><br><span class="line">    Description:  Optional, this switch value specific the output directory for the result data, default is user Desktop folder.</span><br><span class="line"></span><br><span class="line">    Example:      -o &quot;~/Documents/&quot;</span><br><span class="line"></span><br><span class="line">   [-f]</span><br><span class="line">    Description:  Optional, specific the input file format for the sequence reader, default value is FASTA sequence file.</span><br><span class="line">               fsa - The input sequence data file is a FASTA format file;</span><br><span class="line">               gbk - The input sequence data file is a NCBI genbank flat file.</span><br><span class="line"></span><br><span class="line">    Example:      -f &quot;fsa&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘–PerfectPalindrome-Filtering’"><a href="#Help-for-command-‘–PerfectPalindrome-Filtering’" class="headerlink" title="Help for command ‘–PerfectPalindrome.Filtering’:"></a>Help for command ‘–PerfectPalindrome.Filtering’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 FilterPerfectPalindrome(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --PerfectPalindrome.Filtering /in &lt;inDIR&gt; [/min &lt;8&gt; /out &lt;outDIR&gt;]</span><br><span class="line">Example:      seqtools --PerfectPalindrome.Filtering</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘Repeats-Density’"><a href="#Help-for-command-‘Repeats-Density’" class="headerlink" title="Help for command ‘Repeats.Density’:"></a>Help for command ‘Repeats.Density’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 RepeatsDensity(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE Repeats.Density /dir &lt;dir&gt; /size &lt;size&gt; /ref &lt;refName&gt; [/out &lt;out.csv&gt; /cutoff &lt;default:=0&gt;]</span><br><span class="line">Example:      seqtools Repeats.Density</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘-reverse’"><a href="#Help-for-command-‘-reverse’" class="headerlink" title="Help for command ‘-reverse’:"></a>Help for command ‘-reverse’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 Reverse(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE -reverse -i &lt;input_fasta&gt; [-o &lt;output_fasta&gt;]</span><br><span class="line">Example:      seqtools -reverse</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘rev-Repeats-Density’"><a href="#Help-for-command-‘rev-Repeats-Density’" class="headerlink" title="Help for command ‘rev-Repeats.Density’:"></a>Help for command ‘rev-Repeats.Density’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 revRepeatsDensity(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE rev-Repeats.Density /dir &lt;dir&gt; /size &lt;size&gt; /ref &lt;refName&gt; [/out &lt;out.csv&gt; /cutoff &lt;default:=0&gt;]</span><br><span class="line">Example:      seqtools rev-Repeats.Density</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘Search-Batch’"><a href="#Help-for-command-‘Search-Batch’" class="headerlink" title="Help for command ‘Search.Batch’:"></a>Help for command ‘Search.Batch’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 BatchSearch(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Batch search for repeats.</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE Search.Batch /aln &lt;alignment.fasta&gt; [/min 3 /max 20 /min-rep 2 /out &lt;./&gt;]</span><br><span class="line">Example:      seqtools Search.Batch</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">/aln</span><br><span class="line">Description:  The input fasta file should be the output of the clustal multiple alignment fasta output.</span><br><span class="line"></span><br><span class="line">Example:      /aln &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘-segment’"><a href="#Help-for-command-‘-segment’" class="headerlink" title="Help for command ‘-segment’:"></a>Help for command ‘-segment’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 GetSegment(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE -segment /fasta &lt;Fasta_Token&gt; [-loci &lt;loci&gt;] [/left &lt;left&gt; /length &lt;length&gt; /right &lt;right&gt; [/reverse]] [/ptt &lt;ptt&gt; /geneID &lt;gene_id&gt; /dist &lt;distance&gt; /downstream] -o &lt;saved&gt; [-line.break 100]</span><br><span class="line">Example:      seqtools -segment</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–segments’"><a href="#Help-for-command-‘–segments’" class="headerlink" title="Help for command ‘–segments’:"></a>Help for command ‘–segments’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 GetSegments(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --segments /regions &lt;regions.csv&gt; /fasta &lt;nt.fasta&gt; [/complement /reversed /brief-dump]</span><br><span class="line">Example:      seqtools --segments</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">    [/reversed]</span><br><span class="line"> Description:  If the sequence is on the complement strand, reversed it after complement operation?</span><br><span class="line"></span><br><span class="line"> Example:      /reversed &quot;&quot;</span><br><span class="line"></span><br><span class="line">[/complement]</span><br><span class="line"> Description:  If this Boolean switch is set on, then all of the reversed strand segment will be complemenet and reversed.</span><br><span class="line"></span><br><span class="line"> Example:      /complement &quot;&quot;</span><br><span class="line"></span><br><span class="line">[/brief-dump]</span><br><span class="line"> Description:  If this parameter is set up true, then only the locus_tag of the ORF gene will be dump to the fasta sequence.</span><br><span class="line"></span><br><span class="line"> Example:      /brief-dump &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘–ToVector’"><a href="#Help-for-command-‘–ToVector’" class="headerlink" title="Help for command ‘–ToVector’:"></a>Help for command ‘–ToVector’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 ToVector(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --ToVector /in &lt;in.DIR&gt; /min &lt;4&gt; /max &lt;8&gt; /out &lt;out.txt&gt; /size &lt;genome.size&gt;</span><br><span class="line">Example:      seqtools --ToVector</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘–translates’"><a href="#Help-for-command-‘–translates’" class="headerlink" title="Help for command ‘–translates’:"></a>Help for command ‘–translates’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 Translates(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  Translates the ORF gene as protein sequence. If any error was output from the console, please using &gt; operator dump the output to a log file for the analysis.</span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --translates /orf &lt;orf.fasta&gt; [/transl_table 1 /force]</span><br><span class="line">Example:      seqtools --translates</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"> /orf</span><br><span class="line"> Description:  ORF gene nt sequence should be completely complement and reversed as forwards strand if it is complement strand.</span><br><span class="line"></span><br><span class="line"> Example:      /orf &quot;&quot;</span><br><span class="line"></span><br><span class="line">[/force]</span><br><span class="line"> Description:  This force parameter will force the translation program ignore of the stop code and continute sequence translation.</span><br><span class="line"></span><br><span class="line"> Example:      /force &quot;&quot;</span><br><span class="line"></span><br><span class="line">[/transl_table]</span><br><span class="line"> Description:  Available index value was described at http://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html/index.cgi?chapter=tgencodes#SG25</span><br><span class="line"></span><br><span class="line"> Example:      /transl_table &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘–Trim’"><a href="#Help-for-command-‘–Trim’" class="headerlink" title="Help for command ‘–Trim’:"></a>Help for command ‘–Trim’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.AnalysisTools.SequenceTools.Utilities::Int32 Trim(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\bin-Debug\seqtools.EXE --Trim /in &lt;in.fasta&gt; [/case &lt;u/l&gt; /break &lt;-1/int&gt; /out &lt;out.fasta&gt; /brief]</span><br><span class="line">Example:      seqtools --Trim</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">    [/case]</span><br><span class="line"> Description:  Adjust the letter case of your sequence, l for lower case and u for upper case. Default value is upper case.</span><br><span class="line"></span><br><span class="line"> Example:      /case &quot;&quot;</span><br><span class="line"></span><br><span class="line">[/break]</span><br><span class="line"> Description:  Adjust the sequence break when this program write the fasta sequence, default is -1 which means no break, write all sequence in one line.</span><br><span class="line"></span><br><span class="line"> Example:      /break &quot;&quot;</span><br></pre></td></tr></table></figure></p>

              
            </div>
            <footer class="article-footer">
              <a href="/tools/seqtools.html" class="article-date">
  <time datetime="2016-06-12T15:44:40.000Z" itemprop="datePublished">2016-06-12</time>
</a>
              <a data-url="http://docs.gcmodeller.org/tools/seqtools.html" data-id="cipcrjku400006kv4qex6ise4" class="article-share-link">Share</a>
              
              
  <ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/">undefined</a></li></ul>

            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tags" class="mobile-nav-link">Tags</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

