

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>vcsm | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="virtualcell simulations host [version 2.6.0.255]Module AssemblyName: file:///F:/GCModeller/GCModeller-x64/vcsm.exeRoot namespace: LANS.SystemsBiology.GCModeller.CommandLines
All of the command that av">
<meta name="keywords">
<meta property="og:type" content="website">
<meta property="og:title" content="vcsm">
<meta property="og:url" content="http://docs.gcmodeller.org/tools/vcsm.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="virtualcell simulations host [version 2.6.0.255]Module AssemblyName: file:///F:/GCModeller/GCModeller-x64/vcsm.exeRoot namespace: LANS.SystemsBiology.GCModeller.CommandLines
All of the command that av">
<meta property="og:updated_time" content="2016-06-12T16:26:45.582Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="vcsm">
<meta name="twitter:description" content="virtualcell simulations host [version 2.6.0.255]Module AssemblyName: file:///F:/GCModeller/GCModeller-x64/vcsm.exeRoot namespace: LANS.SystemsBiology.GCModeller.CommandLines
All of the command that av">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="/css/favicon.ico">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tags">
                  
                  

                
                  Tags
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/archives">
                  
                  

                
                  Archives
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/tools/vcsm.html">
      <h1 class="article-title" itemprop="name">
        vcsm
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/tools/vcsm.html" class="article-date">
  <time datetime="2016-06-12T16:26:45.000Z" itemprop="datePublished">2016-06-13</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/tools/vcsm.html">
      <h1 class="article-title" itemprop="name">
        vcsm
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#virtualcell-simulations-host-version-2-6-0-255"><span class="toc-number">1.</span> <span class="toc-text">virtualcell simulations host [version 2.6.0.255]</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Commands"><span class="toc-number">1.1.</span> <span class="toc-text">Commands</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘Experiment-Whole-Genome-Mutation’"><span class="toc-number">1.1.0.0.1.</span> <span class="toc-text">Help for command ‘Experiment.Whole_Genome_Mutation’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘Experiment-Whole-Genome-Mutation2’"><span class="toc-number">1.1.0.0.2.</span> <span class="toc-text">Help for command ‘Experiment.Whole_Genome_Mutation2’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘load-model-csv-tabular’"><span class="toc-number">1.1.0.0.3.</span> <span class="toc-text">Help for command ‘load.model.csv_tabular’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘registry’"><span class="toc-number">1.1.0.0.4.</span> <span class="toc-text">Help for command ‘registry’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘run’"><span class="toc-number">1.1.0.0.5.</span> <span class="toc-text">Help for command ‘run’:</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#Help-for-command-‘unregistry’"><span class="toc-number">1.1.0.0.6.</span> <span class="toc-text">Help for command ‘unregistry’:</span></a></li></ol></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="virtualcell-simulations-host-version-2-6-0-255"><a href="#virtualcell-simulations-host-version-2-6-0-255" class="headerlink" title="virtualcell simulations host [version 2.6.0.255]"></a>virtualcell simulations host [version 2.6.0.255]</h1><p><strong>Module AssemblyName</strong>: file:///F:/GCModeller/GCModeller-x64/vcsm.exe<br><strong>Root namespace</strong>: LANS.SystemsBiology.GCModeller.CommandLines</p>
<p>All of the command that available in this program has been list below:</p>
<table>
<thead>
<tr>
<th>Function API</th>
<th>Info</th>
</tr>
</thead>
<tbody>
<tr>
<td>Experiment.Whole_Genome_Mutation</td>
<td>shell parameter is the shoal shell application program file location.</td>
</tr>
<tr>
<td>Experiment.Whole_Genome_Mutation2</td>
<td>shell parameter is the shoal shell application program file location. this command is required the GCML format model file.</td>
</tr>
<tr>
<td>load.model.csv_tabular</td>
<td>The csv_tabular format model file is the alternative format of the GCModeller virtual cell modle, as the GCModeller only support the GCML xml file as the modelling data source, so that you should using this command to load the csv_tabular format model file as the GCML format.</td>
</tr>
<tr>
<td>registry</td>
<td></td>
</tr>
<tr>
<td>run</td>
<td></td>
</tr>
<tr>
<td>unregistry</td>
</tr>
</tbody>
</table>
<h2 id="Commands"><a href="#Commands" class="headerlink" title="Commands"></a>Commands</h2><hr>
<h5 id="Help-for-command-‘Experiment-Whole-Genome-Mutation’"><a href="#Help-for-command-‘Experiment-Whole-Genome-Mutation’" class="headerlink" title="Help for command ‘Experiment.Whole_Genome_Mutation’:"></a>Help for command ‘Experiment.Whole_Genome_Mutation’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.GCModeller.CommandLines::Int32 WholeGenomeMutation(System.String, Double, System.String)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  shell parameter is the shoal shell application program file location.</span><br><span class="line">Usage:        F:\GCModeller\GCModeller-x64\vcsm.exe </span><br><span class="line">Example:      vcsm Experiment.Whole_Genome_Mutation</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘Experiment-Whole-Genome-Mutation2’"><a href="#Help-for-command-‘Experiment-Whole-Genome-Mutation2’" class="headerlink" title="Help for command ‘Experiment.Whole_Genome_Mutation2’:"></a>Help for command ‘Experiment.Whole_Genome_Mutation2’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.GCModeller.CommandLines::Int32 WholeGenomeMutationFromGCML(System.String, Double, System.String)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  shell parameter is the shoal shell application program file location. this command is required the GCML format model file.</span><br><span class="line">Usage:        F:\GCModeller\GCModeller-x64\vcsm.exe </span><br><span class="line">Example:      vcsm Experiment.Whole_Genome_Mutation2</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘load-model-csv-tabular’"><a href="#Help-for-command-‘load-model-csv-tabular’" class="headerlink" title="Help for command ‘load.model.csv_tabular’:"></a>Help for command ‘load.model.csv_tabular’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.GCModeller.CommandLines::LANS.SystemsBiology.GCModeller.ModellingEngine.EngineSystem.Engine.GCModeller LoadCsv(System.String, Microsoft.VisualBasic.Logging.LogFile, Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  The csv_tabular format model file is the alternative format of the GCModeller virtual cell modle, as the GCModeller only support the GCML xml file as the modelling data source, so that you should using this command to load the csv_tabular format model file as the GCML format.</span><br><span class="line">Usage:        F:\GCModeller\GCModeller-x64\vcsm.exe </span><br><span class="line">Example:      vcsm load.model.csv_tabular</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘registry’"><a href="#Help-for-command-‘registry’" class="headerlink" title="Help for command ‘registry’:"></a>Help for command ‘registry’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.GCModeller.CommandLines::Int32 RegistryModule(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\GCModeller\GCModeller-x64\vcsm.exe registry &lt;assembly_file&gt;</span><br><span class="line">Example:      vcsm registry resistry /home/xieguigang/gcmodeller/models/plas.dll</span><br></pre></td></tr></table></figure>
<h5 id="Help-for-command-‘run’"><a href="#Help-for-command-‘run’" class="headerlink" title="Help for command ‘run’:"></a>Help for command ‘run’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.GCModeller.CommandLines::Int32 Run(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\GCModeller\GCModeller-x64\vcsm.exe run -i &lt;model_file&gt; -mysql &lt;mysql_connection_string&gt; [-f &lt;gcml/csv_tabular&gt; -t &lt;time&gt; -metabolism &lt;assembly_path&gt; -expression &lt;assembly_path&gt;]</span><br><span class="line">Example:      vcsm run run -i ~/gc/ecoli.xml -t 1000 -url &quot;http://localhost:8080/client?user=username%password=password%database=database&quot;</span><br></pre></td></tr></table></figure>
<p>  Parameters information:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br></pre></td><td class="code"><pre><span class="line">    -i</span><br><span class="line">    Description:  This switch value specific the model file that the simulation engine will be load</span><br><span class="line"></span><br><span class="line">    Example:      -i &quot;~/gc/ecoli.xml&quot;</span><br><span class="line"></span><br><span class="line">-url</span><br><span class="line">    Description:  Setup the data storage service connection url string.</span><br><span class="line"></span><br><span class="line">    Example:      -url &quot;http://localhost:8080/client?user=username%password=password%database=database&quot;</span><br><span class="line"></span><br><span class="line">   [-t]</span><br><span class="line">    Description:  Optional, This switch specific that the cycle number of this simulation will run, this switch value will override the time value in the loaded model file.</span><br><span class="line"></span><br><span class="line">    Example:      -t &quot;1000&quot;</span><br><span class="line"></span><br><span class="line">   [-metabolism]</span><br><span class="line">    Description:  N/A - The engine kernel will not load the metabolism module.</span><br><span class="line"></span><br><span class="line">    Example:      -metabolism &quot;&quot;</span><br><span class="line"></span><br><span class="line">   [-expression]</span><br><span class="line">    Description:  N/A - The engine kernel will not load the gene expression regulation module.</span><br><span class="line"></span><br><span class="line">    Example:      -expression &quot;&quot;</span><br><span class="line"></span><br><span class="line">   [-interval]</span><br><span class="line">    Description:  This switch value specific the data commit to the mysql database server time interval, if your compiled model is too large you should consider set up this switch value smaller in order to avoid the unexpected memory out of range exception.</span><br><span class="line"></span><br><span class="line">    Example:      -interval &quot;&quot;</span><br><span class="line"></span><br><span class="line">   [-f]</span><br><span class="line">    Description:  This parameter specific the file format of the target input model file, default value is gcml format.</span><br><span class="line"></span><br><span class="line">    Example:      -f &quot;&quot;</span><br><span class="line"></span><br><span class="line">   [-suppress_warn]</span><br><span class="line">    Description:  T/TRUE/F/FALSE</span><br><span class="line"></span><br><span class="line">    Example:      -suppress_warn &quot;&quot;</span><br><span class="line"></span><br><span class="line">   [-suppress_error]</span><br><span class="line">    Description:  T/TRUE/F/FALSE</span><br><span class="line"></span><br><span class="line">    Example:      -suppress_error &quot;&quot;</span><br><span class="line"></span><br><span class="line">   [-suppress_periodic_message]</span><br><span class="line">    Description:  T/TRUE/F/FALSE</span><br><span class="line"></span><br><span class="line">    Example:      -suppress_periodic_message &quot;&quot;</span><br></pre></td></tr></table></figure></p>
<h5 id="Help-for-command-‘unregistry’"><a href="#Help-for-command-‘unregistry’" class="headerlink" title="Help for command ‘unregistry’:"></a>Help for command ‘unregistry’:</h5><p><strong>Prototype</strong>: LANS.SystemsBiology.GCModeller.CommandLines::Int32 UnRegistry(Microsoft.VisualBasic.CommandLine.CommandLine)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Information:  </span><br><span class="line">Usage:        F:\GCModeller\GCModeller-x64\vcsm.exe unregistry &lt;assembly_file&gt;</span><br><span class="line">Example:      vcsm unregistry unregistry ~/gcmodeller/models/plas.dll</span><br></pre></td></tr></table></figure>

              
            </div>
            <footer class="article-footer">
              <a href="/tools/vcsm.html" class="article-date">
  <time datetime="2016-06-12T16:26:45.000Z" itemprop="datePublished">2016-06-13</time>
</a>
              <a data-url="http://docs.gcmodeller.org/tools/vcsm.html" data-id="cipdqk2gj000qncv4jt51knyp" class="article-share-link">Share</a>
              
              
  <ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/">undefined</a></li><li class="article-tag-list-item"><a class="article-tag-list-link" href="/">undefined</a></li></ul>

            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tags" class="mobile-nav-link">Tags</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

