

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>AlignmentReads | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="AlignmentReads_namespace: LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements_
比对区域
 Each alignment line has 11 mandatory fields. These fields always appear in the same order">
<meta property="og:type" content="website">
<meta property="og:title" content="AlignmentReads">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.AlignmentReads.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="AlignmentReads_namespace: LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements_
比对区域
 Each alignment line has 11 mandatory fields. These fields always appear in the same order">
<meta property="og:updated_time" content="2016-06-13T10:38:12.916Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="AlignmentReads">
<meta name="twitter:description" content="AlignmentReads_namespace: LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements_
比对区域
 Each alignment line has 11 mandatory fields. These fields always appear in the same order">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="/css/favicon.ico">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tags">
                  
                  

                
                  Tags
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/archives">
                  
                  

                
                  Archives
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.AlignmentReads.html">
      <h1 class="article-title" itemprop="name">
        AlignmentReads
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.AlignmentReads.html" class="article-date">
  <time datetime="2016-06-13T10:38:12.916Z" itemprop="datePublished">2016-06-13</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.AlignmentReads.html">
      <h1 class="article-title" itemprop="name">
        AlignmentReads
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#AlignmentReads"><span class="toc-number">1.</span> <span class="toc-text">AlignmentReads</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.1.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#ctor"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">#ctor</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#RangeAt"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">RangeAt</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Properties"><span class="toc-number">1.0.2.</span> <span class="toc-text">Properties</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#CIGAR"><span class="toc-number">1.0.2.1.</span> <span class="toc-text">CIGAR</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#FLAG"><span class="toc-number">1.0.2.2.</span> <span class="toc-text">FLAG</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#MAPQ"><span class="toc-number">1.0.2.3.</span> <span class="toc-text">MAPQ</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#OptionalHash"><span class="toc-number">1.0.2.4.</span> <span class="toc-text">OptionalHash</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#PNEXT"><span class="toc-number">1.0.2.5.</span> <span class="toc-text">PNEXT</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#POS"><span class="toc-number">1.0.2.6.</span> <span class="toc-text">POS</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#QNAME"><span class="toc-number">1.0.2.7.</span> <span class="toc-text">QNAME</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#QUAL"><span class="toc-number">1.0.2.8.</span> <span class="toc-text">QUAL</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#RNAME"><span class="toc-number">1.0.2.9.</span> <span class="toc-text">RNAME</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#RNEXT"><span class="toc-number">1.0.2.10.</span> <span class="toc-text">RNEXT</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Strand"><span class="toc-number">1.0.2.11.</span> <span class="toc-text">Strand</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#TLEN"><span class="toc-number">1.0.2.12.</span> <span class="toc-text">TLEN</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="AlignmentReads"><a href="#AlignmentReads" class="headerlink" title="AlignmentReads"></a>AlignmentReads</h1><p>_namespace: <a href="N-LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.html">LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements</a>_</p>
<p>比对区域</p>
<p> Each alignment line has 11 mandatory fields. These fields always appear in the same order and must<br> be present, but their values can be ‘0’ or ‘*’ (depending on the field) if the corresponding information<br> is unavailable. </p>
<p> The following table gives an overview of the mandatory fields in the SAM format:</p>
<p> Col Field Type Regexp/Range Brief description<br> 1 QNAME String [!-?A-~]{1,255} Query template NAME<br> 2 FLAG Int [0,216-1] bitwise FLAG<br> 3 RNAME String *|[!-()+-&lt;&gt;-~][!-~]<em> Reference sequence NAME<br> 4 POS Int [0,229-1] 1-based leftmost mapping POSition<br> 5 MAPQ Int [0,28-1] MAPping Quality<br> 6 CIGAR String \</em>|([0-9]+[MIDNSHPX=])+ CIGAR string<br> 7 RNEXT String *|=|[!-()+-&lt;&gt;-~][!-~]<em> Ref. name of the mate/next segment<br> 8 PNEXT Int [0,229-1] Position of the mate/next segment<br> 9 TLEN Int [-229+1,229-1] observed Template LENgth<br> 10 SEQ String \</em>|[A-Za-z=.]+ segment SEQuence<br> 11 QUAL String [!-~]+ ASCII of Phred-scaled base QUALity+33</p>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="ctor"><a href="#ctor" class="headerlink" title="#ctor"></a>#ctor</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.AlignmentReads.<span class="meta">#ctor(System.String)</span></span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>str</td>
<td>-</td>
</tr>
</tbody>
</table>
<blockquote>
<p> 不可用的片段信息会使用0或者*来表示</p>
</blockquote>
<h4 id="RangeAt"><a href="#RangeAt" class="headerlink" title="RangeAt"></a>RangeAt</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.AlignmentReads.RangeAt(System.Int64,System.Int64)</span><br></pre></td></tr></table></figure>
<p>查看这个Read是否在该范围之内，由于只是对片段区域内的每一个碱基上面的Reads频数进行计数，所以这里只需要二者只要有重叠就可以了</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>RangeStart</td>
<td>-</td>
</tr>
<tr>
<td>RangeEnds</td>
<td>-</td>
</tr>
</tbody>
</table>
<h3 id="Properties"><a href="#Properties" class="headerlink" title="Properties"></a>Properties</h3><h4 id="CIGAR"><a href="#CIGAR" class="headerlink" title="CIGAR"></a>CIGAR</h4><p>CIGAR string</p>
<p> CIGAR: CIGAR string. The CIGAR operations are given in the following table (set `*’ if unavailable):</p>
<p> Op BAM Description<br> M 0 alignment match (can be a sequence match Or mismatch)<br> I 1 insertion to the reference<br> D 2 deletion from the reference<br> N 3 skipped region from the reference<br> S 4 soft clipping (clipped sequences present in SEQ)<br> H 5 hard clipping (clipped sequences Not present in SEQ)<br> P 6 padding (silent deletion from padded reference)<br> = 7 sequence match<br> X 8 sequence mismatch</p>
<p> H can only be present As the first And/Or last operation.<br> S may only have H operations between them And the ends Of the CIGAR String.<br> For mRNA -To -genome alignment, an N operation represents an intron. For other types Of alignments, the interpretation of N Is Not defined.<br> Sum of lengths of the M/I/S/=/X operations shall equal the length of SEQ.</p>
<h4 id="FLAG"><a href="#FLAG" class="headerlink" title="FLAG"></a>FLAG</h4><p>bitwise FLAG</p>
<p> bitwise FLAG. Each bit is explained in the following table:</p>
<p> Bit Description<br> 0x1 template having multiple segments in sequencing<br> 0x2 each segment properly aligned according to the aligner<br> 0x4 segment unmapped<br> 0x8 next segment in the template unmapped<br> 0x10 SEQ being reverse complemented<br> 0x20 SEQ of the next segment in the template being reversed<br> 0x40 the first segment in the template<br> 0x80 the last segment in the template<br> 0x100 secondary alignment<br> 0x200 Not passing quality controls<br> 0x400 PCR Or optical duplicate</p>
<p> Bit 0x4 Is the only reliable place to tell whether the segment Is unmapped. If 0x4 Is set,<br> no assumptions can be made about RNAME, POS, CIGAR, MAPQ, bits 0x2, 0x10 And 0x100 And the<br> bit 0x20 of the next segment in the template.</p>
<p> If 0x40 And 0x80 are both set, the segment Is part of a linear template, but it Is neither<br> the first nor the last segment. If both 0x40 And 0x80 are unset, the index of the segment<br> in the template Is unknown. This may happen For a non-linear template Or the index Is lost<br> in data processing.</p>
<p> Bit 0x100 marks the alignment Not to be used in certain analyses when the tools in use are aware of this bit.<br> If 0x1 Is unset, no assumptions can be made about 0x2, 0x8, 0x20, 0x40 And 0x80.</p>
<h4 id="MAPQ"><a href="#MAPQ" class="headerlink" title="MAPQ"></a>MAPQ</h4><p>MAPping Quality</p>
<p> MAPQ: MAPping Quality. It equals -10 log10 Prfmapping position is wrongg, rounded to the<br> nearest integer. A value 255 indicates that the mapping quality Is Not available.</p>
<h4 id="OptionalHash"><a href="#OptionalHash" class="headerlink" title="OptionalHash"></a>OptionalHash</h4><p>最后面的可选的字段数据</p>
<p> 可选字段（optional fields)，格式如：TAG:TYPE:VALUE，其中TAG有两个大写字母组成，每个TAG代表一类信息，<br> 每一行一个TAG只能出现一次，TYPE表示TAG对应值的类型，可以是字符串、整数、字节、数组等。</p>
<p> 要注意的几个概念，以及与之对应的模型</p>
<p> ◾reference<br> ◾read<br> ◾segment<br> ◾template（参考序列和比对上的序列共同组成的序列为template）<br> ◾alignment<br> ◾seq</p>
<h4 id="PNEXT"><a href="#PNEXT" class="headerlink" title="PNEXT"></a>PNEXT</h4><p>Position of the mate/next segment</p>
<p> PNEXT: Position of the NEXT segment in the template. Set as 0 when the information is<br> unavailable. This field equals POS of the next segment. If PNEXT Is 0, no assumptions can be<br> made on RNEXT And bit 0x20.</p>
<h4 id="POS"><a href="#POS" class="headerlink" title="POS"></a>POS</h4><p>1-based leftmost mapping POSition.(本Mapping在参考基因组之中的最左端的位置)</p>
<p> POS: 1-based leftmost mapping POSition of the first matching base. The first base in a reference<br> sequence has coordinate 1. POS Is Set As 0 For an unmapped read without coordinate. If POS Is<br> 0, no assumptions can be made about RNAME And CIGAR.</p>
<h4 id="QNAME"><a href="#QNAME" class="headerlink" title="QNAME"></a>QNAME</h4><p>Query template NAME.(Query template NAME. Reads/segments having identical QNAME are regarded to<br> come from the same template. A QNAME ‘*’ indicates the information is unavailable.)</p>
<h4 id="QUAL"><a href="#QUAL" class="headerlink" title="QUAL"></a>QUAL</h4><p>ASCII of Phred-scaled base QUALity+33</p>
<p> QUAL: ASCII of base QUALity plus 33 (same as the quality string in the Sanger FASTQ format).<br> A base quality Is the phred-scaled base Error probability which equals -10 log10 Prfbase Is wrongg.<br> This field can be a <code>*&#39; when quality is not stored. If not a</code><em>‘, SEQ must not be a `</em>‘ and the<br> length of the quality string ought to equal the length of SEQ.</p>
<h4 id="RNAME"><a href="#RNAME" class="headerlink" title="RNAME"></a>RNAME</h4><p>Reference sequence NAME</p>
<p> RNAME: Reference sequence NAME of the alignment. If @SQ header lines are present, RNAME<br> (if Not <code>*&#39;) must be present in one of the SQ-SN tag. An unmapped segment without coordinate
 has a</code><em>‘ at this field. However, an unmapped segment may also have an ordinary coordinate<br> such that it can be placed at a desired position after sorting. If RNAME Is `</em>‘, no assumptions<br> can be made about POS And CIGAR.</p>
<h4 id="RNEXT"><a href="#RNEXT" class="headerlink" title="RNEXT"></a>RNEXT</h4><p>Ref. name of the mate/next segment.</p>
<p> RNEXT: Reference sequence name of the NEXT segment in the template. For the last segment,<br> the next segment Is the first segment in the template. If @SQ header lines are present, RNEXT<br> (if Not <code>*&#39; or</code>=’) must be present in one of the SQ-SN tag. This field is set as <code>*&#39; when the
 information Is unavailable, And set as</code>=’ if RNEXT is identical RNAME. If not <code>=&#39; and the next
 segment in the template has one primary mapping (see also bit @&quot;F:LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.BitFLAGS.Bit0x100&quot;[0x100] in FLAG), this field Is
 identical to RNAME of the next segment. If the next segment has multiple primary mappings,
 no assumptions can be made about RNEXT And PNEXT. If RNEXT Is</code>*’, no assumptions can<br> be made On PNEXT And bit @”F:LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.BitFLAGS.Bit0x20”[0x20].</p>
<h4 id="Strand"><a href="#Strand" class="headerlink" title="Strand"></a>Strand</h4><p>获取当前的这条reads在基因组之上的方向</p>
<h4 id="TLEN"><a href="#TLEN" class="headerlink" title="TLEN"></a>TLEN</h4><p>observed Template LENgth</p>
<p> TLEN: signed observed Template LENgth. If all segments are mapped to the same reference, the<br> unsigned observed template length equals the number Of bases from the leftmost mapped base<br> to the rightmost mapped base. The leftmost segment has a plus sign And the rightmost has a<br> minus sign. The sign Of segments In the middle Is undefined. It Is Set As 0 For Single-segment<br> template Or when the information Is unavailable.</p>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.AlignmentReads.html" class="article-date">
  <time datetime="2016-06-13T10:38:12.916Z" itemprop="datePublished">2016-06-13</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.AlignmentReads.html" data-id="cipdv3o2u01qibgv4ehs5lkgb" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tags" class="mobile-nav-link">Tags</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

