

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>Module | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="Modulenamespace: LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject
KEGG MODULE is a collection of manually defined functional units, called KEGG modules and identified by the M numbers, used for anno">
<meta property="og:type" content="website">
<meta property="og:title" content="Module">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject.Module.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="Modulenamespace: LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject
KEGG MODULE is a collection of manually defined functional units, called KEGG modules and identified by the M numbers, used for anno">
<meta property="og:updated_time" content="2016-06-13T13:11:12.894Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Module">
<meta name="twitter:description" content="Modulenamespace: LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject
KEGG MODULE is a collection of manually defined functional units, called KEGG modules and identified by the M numbers, used for anno">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject.Module.html">
      <h1 class="article-title" itemprop="name">
        Module
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject.Module.html" class="article-date">
  <time datetime="2016-06-13T13:11:12.894Z" itemprop="datePublished">2016-06-13</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject.Module.html">
      <h1 class="article-title" itemprop="name">
        Module
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#Module"><span class="toc-number">1.</span> <span class="toc-text">Module</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.1.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#GetPathwayGenes"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">GetPathwayGenes</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="Module"><a href="#Module" class="headerlink" title="Module"></a>Module</h1><p><em>namespace: <a href="N-LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject.html">LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject</a></em></p>
<p>KEGG MODULE is a collection of manually defined functional units, called KEGG modules and identified by the M numbers,<br> used for annotation and biological interpretation of sequenced genomes. </p>
<p> There are four types of KEGG modules:<br> pathway modules – representing tight functional units In KEGG metabolic pathway maps, such As M00002 (Glycolysis, core Module involving three-carbon compounds)<br> structural complexes – often forming molecular machineries, such As M00072 (Oligosaccharyltransferase)<br> functional sets – For other types Of essential sets, such As M00360 (Aminoacyl-tRNA synthases, prokaryotes)<br> signature modules – As markers Of phenotypes, such As M00363 (EHEC pathogenicity signature, Shiga toxin)</p>
<p> The entire list Of KEGG modules can be viewed from the BRITE hierarchy file:<br> KEGG modules</p>
<p> Logical Expression</p>
<p> The M number entry Is defined by a logical expression Of K numbers (And other M numbers), allowing automatic evaluation Of whether the gene Set Is complete, i.e., the Module Is present, In a given genome. A space Or a plus sign represents an And operation, And a comma sign represents an Or operation In this expression. A plus sign Is used For a molecular complex And a minus sign designates an Optional item In the complex. </p>
<p> Each space-separated unit Is called a block, And the distinction Is made for:<br> complete modules<br> incomplete but almost complete modules With only 1 Or 2 blocks missing<br> all modules that contain any matching K numbers<br> in the KEGG MODULE database, as well as in the KEGG Mapper tool “Reconstruct Module”.<br> Module Map</p>
<p> KEGG modules are associated With graphical diagrams called Module maps. For example, M00002 represents glycolysis core Module involving three-carbon compounds And its organism specific Module can be selected from the pop-up menu Or directly specified In the form Of hsa_M00002. While KEGG pathway maps are all manually drawn, KEGG Module maps are computationally generated from the text definition describe above. There Is also a difference Of how organism-specific versions are generated. Organism specific Module entries are created only For complete modules, While organism-specific pathway map entries are created When a few matching elements exist under a predefined taxonomic constraint.<br> Ortholog Table</p>
<p> The ortholog table Is a useful tool To check completeness And consistency Of genome annotations. It shows currently annotated genes In individual genomes For a given Set Of K numbers, together With coloring Of adjacent genes (operon-Like structures) On the chromosome. Each KEGG Module contains a link To the corresponding ortholog table, such As For M00165, together With Option To Select complete Or other modules.<br> Taxonomy Mapping</p>
<p> Each KEGG module also contains a link to KEGG taxonomy mapping, showing which organisms Or organism groups have the module under the taxonomic classification of KEGG organisms. The taxonomy link from the ortholog table (designated by T) allows mapping of both complete And incomplete modules. The result Is shown in the color coding shown below.<br> complete<br> incomplete, 1 block missing<br> incomplete, 2 blocks missing<br> KEGG Modules And Reaction Modules</p>
<p> It Is interesting to note that KEGG modules sometimes correspond to reaction modules extracted from purely chemical properties as summarized in the following BRITE hierarchy file:<br> KEGG reaction modules<br> A New category of KEGG metabolic pathway maps, called overview maps, shows this correspondence as well as an overall architecture of the metabolic network. The following Is an example taken from the overview map for Degradation of aromatic compounds. </p>
<p> A single M number Or a combination of M numbers can be used for characterizing phenotypic features encoded in the genome. For example, the BTX (benzene, toluene, And xylene) degradation capacity can be seen from the following diagram where M numbers are linked to the ortholog tables indicating which organisms have complete modules.<br> benzene— M00548 →catechol<br> toluene— M00538 →benzoate— M00551 →catechol— M00569 →meta-cleavage<br> — M00568 →ortho-cleavage<br> xylene— M00537 →methyl-<br> benzoate— M00551 →methyl-<br> catechol— M00569 →meta-cleavage<br> — M00568 →ortho-cleavage<br> This example can be rewritten In terms Of the reaction modules.<br> benzene— RM006 →catechol<br> toluene— RM003 →benzoate— RM005 →catechol— RM009 →meta-cleavage<br> — RM008 →ortho-cleavage<br> xylene— RM003 →methyl-<br> benzoate— RM005 →methyl-<br> catechol— RM009 →meta-cleavage<br> — RM008 →ortho-cleavage<br> See more details In:<br> Muto, A., Kotera, M., Tokimatsu, T., Nakagawa, Z., Goto, S., And Kanehisa, M.; Modular architecture of metabolic pathways revealed by conserved sequences of reactions. J. Chem. Inf. Model. 53, 613-622 (2013). [pubmed] [pdf] </p>
<p> Kanehisa, M.; Chemical And genomic evolution of enzyme-catalyzed reaction networks. FEBS Lett. 587, 2731-2737 (2013). [pubmed] [pdf] </p>
<p> Kanehisa, M., Goto, S., Sato, Y., Kawashima, M., Furumichi, M., And Tanabe, M.; Data, information, knowledge And principle back to metabolism in KEGG. Nucleic Acids Res. 42, D199–D205 (2014). [pubmed] [pdf]</p>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="GetPathwayGenes"><a href="#GetPathwayGenes" class="headerlink" title="GetPathwayGenes"></a>GetPathwayGenes</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject.Module.GetPathwayGenes</span><br></pre></td></tr></table></figure>
<p>得到当前的模块之中的基因的编号的列表，这是个安全的函数，不会返回空值</p>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject.Module.html" class="article-date">
  <time datetime="2016-06-13T13:11:12.894Z" itemprop="datePublished">2016-06-13</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-LANS.SystemsBiology.Assembly.KEGG.DBGET.bGetObject.Module.html" data-id="cipe3i470013n20v4u07ajlmp" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

