

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>Perl | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="Perl_namespace: LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR_
dRNA-seq
 TSSAR Perl invoke module
 DESCRIPTION Transcription Start Sites Annotation Regime for dRNA-seq data, based on a Skellam distributi">
<meta property="og:type" content="website">
<meta property="og:title" content="Perl">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR.Perl.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="Perl_namespace: LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR_
dRNA-seq
 TSSAR Perl invoke module
 DESCRIPTION Transcription Start Sites Annotation Regime for dRNA-seq data, based on a Skellam distributi">
<meta property="og:updated_time" content="2016-06-13T10:38:13.061Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Perl">
<meta name="twitter:description" content="Perl_namespace: LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR_
dRNA-seq
 TSSAR Perl invoke module
 DESCRIPTION Transcription Start Sites Annotation Regime for dRNA-seq data, based on a Skellam distributi">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="/css/favicon.ico">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tags">
                  
                  

                
                  Tags
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/archives">
                  
                  

                
                  Archives
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR.Perl.html">
      <h1 class="article-title" itemprop="name">
        Perl
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR.Perl.html" class="article-date">
  <time datetime="2016-06-13T10:38:13.061Z" itemprop="datePublished">2016-06-13</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR.Perl.html">
      <h1 class="article-title" itemprop="name">
        Perl
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#Perl"><span class="toc-number">1.</span> <span class="toc-text">Perl</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.1.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#Invoke"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">Invoke</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Located"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">Located</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Properties"><span class="toc-number">1.0.2.</span> <span class="toc-text">Properties</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#R"><span class="toc-number">1.0.2.1.</span> <span class="toc-text">R</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="Perl"><a href="#Perl" class="headerlink" title="Perl"></a>Perl</h1><p>_namespace: <a href="N-LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR.html">LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR</a>_</p>
<p>dRNA-seq</p>
<p> TSSAR Perl invoke module</p>
<p> DESCRIPTION<br> <em>T</em>ranscription <em>S</em>tart <em>S</em>ites <em>A</em>nnotation <em>R</em>egime for dRNA-seq data,<br> based on a Skellam distribution with parameter estimation by<br> zero-inflated-poisson model regression analysis. The input are two<br> mapped sequencing files in SAM file formate (library[+] and library[-]),<br> the output is a *.BED file with an entry for each position which is<br> annotated as a TSS, writen to STDOUT. Addtionally, a file named Dump.bed<br> is created. It specifies regions where the applied regression model does<br> not converge. Hence, those regions are omitted from analysis.</p>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="Invoke"><a href="#Invoke" class="headerlink" title="Invoke"></a>Invoke</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR.Perl.Invoke(System.String,System.String,System.String,System.String,System.Int32,System.Int32,System.String,System.Int32,System.Boolean,System.Boolean,System.Boolean,System.Int32,System.Boolean)</span><br></pre></td></tr></table></figure>
<p><em>T</em>ranscription <em>S</em>tart <em>S</em>ites <em>A</em>nnotation <em>R</em>egime for dRNA-seq data,<br> based on a Skellam distribution with parameter estimation by<br> zero-inflated-poisson model regression analysis. The input are two<br> mapped sequencing files in SAM file formate (library[+] and library[-]),<br> the output is a *.BED file with an entry for each position which is<br> annotated as a TSS, writen to STDOUT. Addtionally, a file named Dump.bed<br> is created. It specifies regions where the applied regression model does<br> not converge. Hence, those regions are omitted from analysis.</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>prorata</td>
</tr>
</tbody>
</table>
<p> If set, the information from the SAM file how many times a read was<br> mapped to the genome is used, if present. If the read maps <em>n</em> times<br> to the genome, each position is counted only <em>1/n</em> times. Usefull in<br> combination with e.g. segemehl mapper, which can report suboptimal<br> mapping positions and/or reports all location where a read maps<br> optimally. Default is off.|<br>|libP|<br> Input library (P .. Plus; M .. Minus) in SAM format. The plus library is the one with enriched TSS<br> (for dRNA-seq this means that the plus library is the treated library, while the minus library is<br> the untreated library)<br> |<br>|libM|<br> Input library (P .. Plus; M .. Minus) in SAM format. The plus library is the one with enriched TSS<br> (for dRNA-seq this means that the plus library is the treated library, while the minus library is<br> the untreated library)<br> |<br>|fasta|<br> Either the location of reference genome sequence in fasta file<br> format OR the genome size in <em>INT</em>. The fasta file is only used to<br> parse the genome size so just one of the two must be specified.<br> |<br>|g_size|<br> Either the location of reference genome sequence in fasta file<br> format OR the genome size in <em>INT</em>. The fasta file is only used to<br> parse the genome size so just one of the two must be specified.<br> |<br>|minPeak|<br> Minimal Peak size in <em>INT</em>. Only positions where read start count in<br> the (+)library is greater or equal then <em>INT</em> are evaluated to be a<br> TSS. Positions with less reads are seen as backgroound noise and not<br> considered. Default is <em>3</em>.<br> |<br>|pval|Maximal P-value for each position to be annotated as a TSS. Default is <em>1e-04</em>.|<br>|winSize|Size of the window which slides over the genome and defines the statistical properties of the local model. Default is <em>1,000</em>.|<br>|verbose|If set, some progress reports are printed to STDERR during computation.|<br>|score|<br> If score mode is <em>p</em> the p-value is used as score in the TSS BED<br> file. If score mode is <em>d</em> the peak difference is used as score in<br> the TSS BED file. Default is <em>d</em>. Also used for clustering, which<br> advices to use ‘d’, since the p-value often becomes zero for<br> consecutive positions, thus disabling a proper merging of<br> consecutive positions to the best one.<br> |<br>|nocluster|<br> If –nocluster is set all positions annotated as TSS are reported.<br> If –cluster is set consecutive TSS positions are clustered and only<br> the ‘best’ position is reported. ‘Best’ position depends on the<br> setting of –score (see above). Either the position with the lowest<br> p-Value or the position with the highest peak difference between<br> plus and minus library is reported. Default is –cluster. The option<br> –range defines the maximal distance for two significant positions<br> to be called ‘consecutive’.<br> |<br>|range|<br> The maximal distance for two significant positions to be be<br> clustered together if option –cluster is set. Default is <em>3</em> nt. If<br> –cluster is set to –nocluster, –range is ignored.|<br>|clean|<br> If –clean is set, all temporary files which are created during the<br> computation are deleted afterwards. With –noclean they are stored.<br> Mainly for debugging purpose. Default setting is –clean.|</p>
<blockquote>
<p> 这个Perl和R脚本的执行效率太低了！</p>
</blockquote>
<h4 id="Located"><a href="#Located" class="headerlink" title="Located"></a>Located</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR.Perl.Located(System.Collections.Generic.IEnumerable&#123;LANS.SystemsBiology.Toolkits.RNA_Seq.BOW.DocumentFormat.SAM.DocumentElements.AlignmentReads&#125;,LANS.SystemsBiology.Assembly.NCBI.GenBank.TabularFormat.PTT)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>data</td>
<td>-</td>
</tr>
<tr>
<td>Ptt</td>
<td>参考基因组之中的基因的摘要信息</td>
</tr>
</tbody>
</table>
<h3 id="Properties"><a href="#Properties" class="headerlink" title="Properties"></a>Properties</h3><h4 id="R"><a href="#R" class="headerlink" title="R"></a>R</h4><p>R程序的路径</p>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR.Perl.html" class="article-date">
  <time datetime="2016-06-13T10:38:13.061Z" itemprop="datePublished">2016-06-13</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.TSSAR.Perl.html" data-id="cipdv3o5v01skbgv4p96vvd4c" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tags" class="mobile-nav-link">Tags</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

