

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>DESeq | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="DESeq_namespace: LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2_
Differential analysis of count data – the DESeq2 package (M. I. Love, W. Huber, S. Anders: Moderated estimation of fold change And">
<meta property="og:type" content="website">
<meta property="og:title" content="DESeq">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="DESeq_namespace: LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2_
Differential analysis of count data – the DESeq2 package (M. I. Love, W. Huber, S. Anders: Moderated estimation of fold change And">
<meta property="og:updated_time" content="2016-06-13T13:11:20.179Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="DESeq">
<meta name="twitter:description" content="DESeq_namespace: LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2_
Differential analysis of count data – the DESeq2 package (M. I. Love, W. Huber, S. Anders: Moderated estimation of fold change And">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.html">
      <h1 class="article-title" itemprop="name">
        DESeq
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.html" class="article-date">
  <time datetime="2016-06-13T13:11:20.179Z" itemprop="datePublished">2016-06-13</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.html">
      <h1 class="article-title" itemprop="name">
        DESeq
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#DESeq"><span class="toc-number">1.</span> <span class="toc-text">DESeq</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.1.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#DESeq2"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">DESeq2</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#FilterDifferentExpression"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">FilterDifferentExpression</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#HTSeqCount"><span class="toc-number">1.0.1.3.</span> <span class="toc-text">HTSeqCount</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Initialize"><span class="toc-number">1.0.1.4.</span> <span class="toc-text">Initialize</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#summarizeOverlaps"><span class="toc-number">1.0.1.5.</span> <span class="toc-text">summarizeOverlaps</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#mode-funtions"><span class="toc-number">1.1.</span> <span class="toc-text">mode funtions</span></a></li></ol></li></ol>
              </div>
              
                <h1 id="DESeq"><a href="#DESeq" class="headerlink" title="DESeq"></a>DESeq</h1><p>_namespace: <a href="N-LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.html">LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2</a>_</p>
<p>Differential analysis of count data – the DESeq2 package<br> (M. I. Love, W. Huber, S. Anders: Moderated estimation of fold change And dispersion For RNA-Seq data With DESeq2. bioRxiv(2014).doi : 10.1101/002832)</p>
<p> A basic task in the analysis of count data from RNA-Seq is the detection of differentially<br> expressed genes. The count data are presented As a table which reports, For Each sample, the<br> number Of sequence fragments that have been assigned To Each gene. Analogous data also arise</p>
<p> For other assay types, including comparative ChIP-Seq, HiC, shRNA screening, mass spectrometry.<br> An important analysis question Is the quantification And statistical inference Of systematic changes<br> between conditions, as compared To within-condition variability. The package DESeq2 provides<br> methods To test For differential expression by use Of negative binomial generalized linear models;<br> the estimates Of dispersion And logarithmic fold changes incorporate data-driven prior distributions.</p>
<p> This vignette explains the use of the package And demonstrates typical work flows.<br> Another vignette, “Beginner’s guide to using the DESeq2 package”, covers similar material but at a slower<br> pace, including the generation Of count tables from FASTQ files.</p>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="DESeq2"><a href="#DESeq2" class="headerlink" title="DESeq2"></a>DESeq2</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.DESeq2(System.Collections.Generic.IEnumerable&#123;System.Collections.Generic.IEnumerable&#123;System.String&#125;&#125;,System.String)</span><br></pre></td></tr></table></figure>
<p>HTSeq-Count</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>TestData</td>
</tr>
</tbody>
</table>
<p> samFile, condition;<br> samFile, condition;</p>
<p> |</p>
<h4 id="FilterDifferentExpression"><a href="#FilterDifferentExpression" class="headerlink" title="FilterDifferentExpression"></a>FilterDifferentExpression</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.FilterDifferentExpression(System.String,System.Double)</span><br></pre></td></tr></table></figure>
<p>导出差异表达的基因</p>
<h4 id="HTSeqCount"><a href="#HTSeqCount" class="headerlink" title="HTSeqCount"></a>HTSeqCount</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.HTSeqCount(System.String,System.String,System.Boolean)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>SAM</td>
<td>-</td>
</tr>
<tr>
<td>GFF</td>
<td>-</td>
</tr>
<tr>
<td>PairedEnd</td>
<td>假若是Paired-End的比对数据，则还需要首先使用samtools工具进行排序</td>
</tr>
</tbody>
</table>
<h4 id="Initialize"><a href="#Initialize" class="headerlink" title="Initialize"></a>Initialize</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.Initialize(System.String)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>R_HOME</td>
<td>-</td>
</tr>
</tbody>
</table>
<blockquote>
<p> For counting aligned reads in genes, the summarizeOverlaps function of GenomicAlignments<br> With mode=”Union” Is encouraged, resulting In a SummarizedExperiment Object(easyRNASeq Is<br> another Bioconductor package which can prepare SummarizedExperiment objects as input for DESeq2).</p>
<p> 为了计数基因的比对Reads数目， GenomicAlignments 包之中的summarizeOverlaps方法可以生成DESeq2所需要用到的数据<br> easyRNASeq包也可以用来生成这种数据</p>
</blockquote>
<h4 id="summarizeOverlaps"><a href="#summarizeOverlaps" class="headerlink" title="summarizeOverlaps"></a>summarizeOverlaps</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.summarizeOverlaps(System.Collections.Generic.IEnumerable&#123;System.String&#125;,System.String)</span><br></pre></td></tr></table></figure>
<p>Counting reads with summarizeOverlaps, Perform overlap queries between reads and genomic features</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>annoGFF</td>
<td>GTF基因组注释文件的文件路径</td>
</tr>
<tr>
<td>bamList</td>
<td>Mapping所得到的*.bam二进制文件的文件路径的列表</td>
</tr>
</tbody>
</table>
<blockquote>
<p> 作者推荐使用这个函数方法的模式</p>
<h2 id="mode-funtions"><a href="#mode-funtions" class="headerlink" title="mode funtions"></a>mode funtions</h2><p> Union(features, reads, ignore.strand=FALSE, inter.feature=TRUE)</p>
</blockquote>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.html" class="article-date">
  <time datetime="2016-06-13T13:11:20.179Z" itemprop="datePublished">2016-06-13</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-LANS.SystemsBiology.Toolkits.RNA_Seq.RTools.DESeq2.DESeq.html" data-id="cipe3ilsw02j520v4ao378gyq" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

