

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>Palindrome | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="Palindromenamespace: LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically
=== Palindromic hexamers === For a given sequence, any palindrome of 6 nt (e.g., AAATTT) Is given a">
<meta property="og:type" content="website">
<meta property="og:title" content="Palindrome">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="Palindromenamespace: LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically
=== Palindromic hexamers === For a given sequence, any palindrome of 6 nt (e.g., AAATTT) Is given a">
<meta property="og:updated_time" content="2016-06-19T03:14:45.897Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Palindrome">
<meta name="twitter:description" content="Palindromenamespace: LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically
=== Palindromic hexamers === For a given sequence, any palindrome of 6 nt (e.g., AAATTT) Is given a">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.html">
      <h1 class="article-title" itemprop="name">
        Palindrome
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.html" class="article-date">
  <time datetime="2016-06-19T03:14:45.897Z" itemprop="datePublished">2016-06-19</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.html">
      <h1 class="article-title" itemprop="name">
        Palindrome
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#Palindrome"><span class="toc-number">1.</span> <span class="toc-text">Palindrome</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.1.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#CreateMirrors"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">CreateMirrors</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#HaveMirror"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">HaveMirror</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#HavePalindrome"><span class="toc-number">1.0.1.3.</span> <span class="toc-text">HavePalindrome</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#SearchMirror"><span class="toc-number">1.0.1.4.</span> <span class="toc-text">SearchMirror</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="Palindrome"><a href="#Palindrome" class="headerlink" title="Palindrome"></a>Palindrome</h1><p><em>namespace: <a href="N-LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.html">LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically</a></em></p>
<p>=== Palindromic hexamers ===<br> For a given sequence, any palindrome of 6 nt (e.g., AAATTT) Is given a value of 1, while<br> all bases Not included inpalindromic hexamers are given a value of 0 (van et al. 2003).<br> – van Noort V, Worning P, Ussery DW, Rosche WA, Sinden RR Strand misalignments lead To<br> quasipalindrome correction (2003) 19:365-9<br> @”M:LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.SearchMirror(LANS.SystemsBiology.SequenceModel.I_PolymerSequenceModel,System.Int32,System.Int32)” (镜像回文序列)</p>
<p> === Inverted Repeats ===<br> Local Inverted repeats are found by taking a 100 bp sequence window, And looking For the<br> best match Of a 30 bp piece withinthat window, On the opposite strand, In the opposite<br> direction (Jensen et al. 1999).<br> Values can range from 0 (no match at all)To 1 (one Or more perfect match within the window).<br> – L. J. Jensen And C. Friis And D.W. Ussery Three views of complete chromosomes (1999) 150773-777<br> @”M:LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.SimilarityMatches.Repeats.InvokeSearchReversed(System.String,System.Int32,System.Int32,System.Double)” (反向重复)</p>
<p> === Quasi-palindromes ===<br> “Quasi-palindromes” are short inverted repeats, which are found by taking a 30 bp piece of sequence,<br> And looking for matcheswith at least 6 out of 7 nt matching, on the opposite strand, in the opposite<br> direction (van et al. 2003). Values canrange from 0 (no match at all) to 1 (one Or more perfect<br> match within the window).<br> – van Noort V, Worning P, Ussery DW, Rosche WA, Sinden RR Strand misalignments lead<br> To quasipalindrome correction (2003) 19:365-9<br> @”T:LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Imperfect” (非完全回文)</p>
<p> === Perfect-palindromes ===<br> “Perfect-palindromes” are short inverted repeats, which are found by taking a 30 bp piece of sequence,<br> And looking forperfect matches of 7 nt Or longer, on the opposite strand, in the opposite direction (van et al. 2003).<br> Values can rangefrom 0 (no match at all) to 1 (one Or more perfect match within the window).<br> – van Noort V, Worning P, Ussery DW, Rosche WA, Sinden RR Strand misalignments lead To<br> quasipalindrome correction (2003) 19:365-9<br> @”M:LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.SearchPalindrome(LANS.SystemsBiology.SequenceModel.I_PolymerSequenceModel,System.Int32,System.Int32)” (简单回文)</p>
<p> === Simple Repeats ===<br> A “simple repeat” Is a region which contains a simple oligonucleotide repeat, Like microsattelites.<br> Simple repeats are foundby looking for tandem repeats of length R within a 2R-bp window.<br> By using the values 12, 14, 15, 16, And 18 for R, allsimple repeats of lengths 1 through 9 are calculated,<br> of length of at least 24 bp (Jensen et al. 1999). Values can range from 0(no match at all) to 1<br> (one Or more perfect match within the window).<br> – L. J. Jensen And C. Friis And D.W. Ussery Three views of complete chromosomes (1999) 150773-777<br> @”M:LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.RepeatsSearchAPI.SearchRepeats(LANS.SystemsBiology.SequenceModel.I_PolymerSequenceModel,System.Int32,System.Int32,System.Int32)” (简单重复序列)</p>
<p> === GC Skew ===<br> For many genomes there Is a strand bias, such that one strand tends To have more G’s,<br> whilst the other strand has more C’s.This GC-skew bias can be measured the number of G’s<br> minus the number of C’s over a fixed length (e.g. 10,000 bp) of DNA(Jensen et al. 1999).<br> The values can range from +1 (all G’s on the examined sequence, with all C’s on the other strand),<br> to -1(the reverse case - all C’s on the examined sequence, and all G’s on the other strand).<br> There is a correlation with GC-skewand the replication leading and lagging strands.<br> – L. J. Jensen And C. Friis And D.W. Ussery Three views of complete chromosomes (1999) 150773-777</p>
<p> === Percent AT ===<br> The percent AT Is a running average Of the AT content, over a given window size. Typically For a bacterial<br> genomes Of about5 Mbp, the window size Is 10,000 bp. The Percent AT can range from 0 (no AT content) To 1 (100% AT).<br> The Percent AT iscorrelated With other DNA structural features, such that AT rich regions are often more readily<br> melted, tend To be lessflexible And more rigid, although they can also be readily compacted chromatin proteins (Pedersen et al. 2000).<br> – A.G. Pedersen And L.J. Jensen And H.H. St\aerfeldt And S. Brunak And D.W.<br> Ussery A DNA structural atlas of extitE. coli (2000) 299907-930</p>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="CreateMirrors"><a href="#CreateMirrors" class="headerlink" title="CreateMirrors"></a>CreateMirrors</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.CreateMirrors(System.String,System.String)</span><br></pre></td></tr></table></figure>
<p>这个函数求解的是绝对相等的</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Segment</td>
<td>-</td>
</tr>
<tr>
<td>Sequence</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="HaveMirror"><a href="#HaveMirror" class="headerlink" title="HaveMirror"></a>HaveMirror</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.HaveMirror(System.String,System.String)</span><br></pre></td></tr></table></figure>
<p>Have mirror repeats?</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Segment</td>
<td>-</td>
</tr>
<tr>
<td>Sequence</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="HavePalindrome"><a href="#HavePalindrome" class="headerlink" title="HavePalindrome"></a>HavePalindrome</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.HavePalindrome(System.String,System.String)</span><br></pre></td></tr></table></figure>
<p>Have Palindrome repeats?</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Segment</td>
<td>-</td>
</tr>
<tr>
<td>Sequence</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="SearchMirror"><a href="#SearchMirror" class="headerlink" title="SearchMirror"></a>SearchMirror</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.SearchMirror(LANS.SystemsBiology.SequenceModel.I_PolymerSequenceModel,System.Int32,System.Int32)</span><br></pre></td></tr></table></figure>
<p>搜索序列上面的镜像回文片段</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sequence</td>
<td>-</td>
</tr>
<tr>
<td>Min</td>
<td>-</td>
</tr>
<tr>
<td>Max</td>
<td>-</td>
</tr>
</tbody>
</table>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.html" class="article-date">
  <time datetime="2016-06-19T03:14:45.897Z" itemprop="datePublished">2016-06-19</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-LANS.SystemsBiology.AnalysisTools.SequenceTools.SequencePatterns.Topologically.Palindrome.html" data-id="cipe3i42s010p20v4czrambzg" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

