

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>Exports | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="Exportsnamespace: SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices
Reactome Data Model

Introduction
Life on the c ellular level is a network of molecular interactions. Molecules are synthesize">
<meta property="og:type" content="website">
<meta property="og:title" content="Exports">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices.Exports.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="Exportsnamespace: SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices
Reactome Data Model

Introduction
Life on the c ellular level is a network of molecular interactions. Molecules are synthesize">
<meta property="og:updated_time" content="2016-07-07T09:41:56.361Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Exports">
<meta name="twitter:description" content="Exportsnamespace: SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices
Reactome Data Model

Introduction
Life on the c ellular level is a network of molecular interactions. Molecules are synthesize">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices.Exports.html">
      <h1 class="article-title" itemprop="name">
        Exports
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices.Exports.html" class="article-date">
  <time datetime="2016-07-07T09:41:56.361Z" itemprop="datePublished">2016-07-07</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices.Exports.html">
      <h1 class="article-title" itemprop="name">
        Exports
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#Exports"><span class="toc-number">1.</span> <span class="toc-text">Exports</span></a></li></ol>
              </div>
              
                <h1 id="Exports"><a href="#Exports" class="headerlink" title="Exports"></a>Exports</h1><p><em>namespace: <a href="N-SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices.html">SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices</a></em></p>
<p>Reactome Data Model</p>
<ol>
<li><p>Introduction</p>
<p>Life on the c ellular level is a network of molecular interactions. Molecules are synthesized and degraded,<br>undergo a bewildering array of temporary and permanent modifications, are transported from one location to<br>another, and form complexes with other molecules. Reactome represents all of this complexity as reactions in<br>which input physical entities are converted to output entities. These reactions can occur spontaneously or be<br>facilitated by physical entities acting as catalysts, and their progress can be modulated by regulatory effects of<br>other physical entities. Reactions are linked together by shared physical entities: a product from one reaction<br>may be a substrate in another reaction and may catalyze yet a thir d. It is often convenient, if sometimes<br>arbitrary, to group such sets of interlinked reactions into pathways.<br>The functions of macromolecular entities such as proteins are often determined not only by their primary<br>sequenc es, but by chemical modific ations they have undergone. In Reactome, unmodified and modif ied forms<br>of a protein are distinct physical entities and the modification proc ess is treated as an ex plicit reaction. A<br>macromolecule’s function may depend on whether the molecule is free or complexed with specif ic other<br>molecules. Reactome treats complexes as physical entities distinct from their components, and the<br>multimerization events that build up complexes are modeled explicitly as reactions.<br>Cellular compartments play a key role in biological proc esses. The segregation of molecules into different<br>compartments often regulates the reactions in which those entities can participate, or can be responsible for<br>driving a reaction forward. In Reactome, a molecule in one compartment is distinct from that molecule in<br>another compartment. Thus, extracellular and cytosolic glucose are different Reactome entities and, e.g., the<br>movement of glucose across the plasma membrane is a reaction that converts the extracellular glucose entity<br>into the cytosolic one.<br>Many biochemical entities and proc esses appear redundant: there are two or more chemically distinct entities<br>that can act more or less interchangeably. It is often useful to treat functionally equivalent protein isoforms,<br>splice variants, and paralogues as a single entity, implying that any individual entity from the given set could<br>fulfill the same role in a given situation. The Reactome data model allows this type of generalization, but does<br>so explicitly in a way that allows us to trac e specific functions back to the individual molecules covered by the<br>generalization.<br>The goal of the Reactome knowledgebase is to represent human biological processes, but many of these<br>processes have not been directly studied in humans. Rather, a human event has been inferred from<br>experiments on material from a model organism. In such cases, the model organism reaction is annotated in<br>Reactome, the inferred human reaction is annotated as a separate event, and the inferential link between the<br>two reactions is explicitly noted.<br>Reactome uses a frame- based knowledge representation. The data model consists of classes (frames) that<br>describe the different concepts (e.g., reaction, simple entity). Knowledge is captured as instanc es of these<br>classes (e.g., “glucose transport across the plasma membrane”, “cytosolic ATP”). Classes have attributes<br>(slots) which hold properties of the instances (e.g., the identities of the molecules that participate as inputs and<br>outputs in a reaction).</p>
</li>
<li><p>Key data classes</p>
<p>2.1 PhysicalEntity<br>PhysicalEntities include individual molecules, multi-molecular complex es, and sets of molecules or complexes<br>grouped together on the basis of shared characteristics. Molecules are further c lassif ied as genome encoded<br>(DNA, RNA, and proteins) or not (all others). Attributes of a PhysicalEntity instanc e capture the chemical<br>structure of an entity, including any covalent modifications in the case of a macromolecule, and its subc ellular<br>localization.<br>PhysicalEntity instanc es that represent, e.g., the same chemical in different compartments, or different<br>post-translationally modified forms of a single protein, share numerous invariant f eatures such as names,<br>molecular structure and links to external databases like UniProt or ChEBI. To enable storage of this shared<br>information in a single place, and to create an explic it link among all the variant forms of what c an also be seen<br>as a single chemical entity, Reactome creates instances of the separate Refere nceEntity class.<br>A ReferenceEntity instance captures the invariant features of a molecule. A PhysicalEntity instance is then the<br>combination of a Referenc eEntity attribute (e.g., Glycogen phosphorylase UniProt:P06737) and attributes<br>giving specific conditional information (e.g., localization to the cytosol and phosphorylation on serine residue<br>14).<br>The PhysicalEntity class has subclasses to distinguish between different kinds of entity and to ensure data<br>integrity while enabling different handling rules for different categories:<br>EntityWithAccessionedSequence - proteins and nucleic acids with known sequences.<br>GenomeEncodedEntity - a species-specif ic protein or nucleic acid whose sequence is unknown, such as an<br>enzyme that has been characterized functionally but not yet purified and sequenced, e.g. cytosolic triokinase<br>SimpleEntity - other fully characterized molecules, e.g. nucleoplasmic ATP or cytosolic glutathione<br>Complex - a complex of two or more PhysicalEntities, e.g. Trimerization of the FASL:FAS receptor complex<br>EntitySet - a set of PhysicalEntities (molecules or c omplexes) which function interchangeably in a given<br>situation, e.g. Notch 3 heterodimer binds with a Notch ligand in the extracellular spac e. This notation allows<br>collective properties of multiple individual entities to be described explicitly.</p>
<p>2.2. CatalystActivity<br>PhysicalEntities are paired with molecular functions taken from the Gene Ontology molecular function<br>controlled vocabulary to describe instanc es of biological cataly sis. An optional ActiveUnit attribute indicates<br>the specif ic domain of a protein or subunit of a c omplex that mediates the catalysis. If a PhysicalEntity has<br>multiple catalytic activities, a separate CatalystActivity is created for each. This strategy allo ws the association<br>of specific activities with specific variant forms of a protein or complex, and also enables easy retrieval of all<br>activities of a protein, or all proteins capable of mediating a specific molecular function.</p>
<p>2.3. Event<br>Events – the conversion of input entities to output entities in one or more steps – are the building blocks used<br>in Reactome to represent all biological processes. Two subclasses of Event are<br>recognized, ReactionlikeEventand Pathway. A ReactionlikeEvent is an event that converts inputs into outputs .<br>A Pathway is any grouping of related Events. An event may be a member of more than one Pathway.<br>The ReactionlikeEvent class is further divided<br>into Reaction, BlackBoxEvent, Polymerisation andDepolymerisation. The Reaction class holds bona fide<br>reactions with balanced inputs and outputs. The BlackBoxEvent class is used for ‘unbalanced’ reactions like<br>protein synthesis or degradation, as well as ‘shortcut’ reactions for more complex proc esses that essentially<br>convert inputs into outputs, e.g. the series of cyclical reactions involved in fatty ac id biosynthesis. The<br>De-/Polymerisation classes can hold reactions that describe the mechanics of a de -/polymerisation reaction,<br>which is inherent ly ‘unbalanced’ due to the nature of a Polymer (that remains the ‘same’ entity even<br>after adding or subtracting a unit).</p>
<p>Full specification of the Reactome data model<br>A full specification of all Reactome classes, slots and a listing of all instances of eac h class is accessible from<br>theSchema page on the top menu bar. There is also a Data model glossary on the Reactome wiki page,<br>giving more details on the usage of the various classes and slots.</p>
</li>
</ol>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices.Exports.html" class="article-date">
  <time datetime="2016-07-07T09:41:56.361Z" itemprop="datePublished">2016-07-07</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Data.Reactome.LocalMySQL.ExportServices.Exports.html" data-id="ciqc7dt5501dkigv4x9p5s0g7" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

