

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>GenomeSignatures | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="GenomeSignatures_namespace: SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative_
在本模块之中，所有的计算过程都是基于@”T:SMRUCC.genomics.SequenceModel.NucleotideModels.NucleicAcid”核酸对象的
Methods__counts1SMRUCC.genomi">
<meta property="og:type" content="website">
<meta property="og:title" content="GenomeSignatures">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="GenomeSignatures_namespace: SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative_
在本模块之中，所有的计算过程都是基于@”T:SMRUCC.genomics.SequenceModel.NucleotideModels.NucleicAcid”核酸对象的
Methods__counts1SMRUCC.genomi">
<meta property="og:updated_time" content="2016-07-07T09:40:59.474Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="GenomeSignatures">
<meta name="twitter:description" content="GenomeSignatures_namespace: SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative_
在本模块之中，所有的计算过程都是基于@”T:SMRUCC.genomics.SequenceModel.NucleotideModels.NucleicAcid”核酸对象的
Methods__counts1SMRUCC.genomi">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.html">
      <h1 class="article-title" itemprop="name">
        GenomeSignatures
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.html" class="article-date">
  <time datetime="2016-07-07T09:40:59.474Z" itemprop="datePublished">2016-07-07</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.html">
      <h1 class="article-title" itemprop="name">
        GenomeSignatures
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#GenomeSignatures"><span class="toc-number">1.</span> <span class="toc-text">GenomeSignatures</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.1.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#counts"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">__counts</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#counts-p"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">__counts_p</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#CodonSignature"><span class="toc-number">1.0.1.3.</span> <span class="toc-text">CodonSignature</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#DinucleotideBIAS"><span class="toc-number">1.0.1.4.</span> <span class="toc-text">DinucleotideBIAS</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#DinucleotideBIAS-p"><span class="toc-number">1.0.1.5.</span> <span class="toc-text">DinucleotideBIAS_p</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="GenomeSignatures"><a href="#GenomeSignatures" class="headerlink" title="GenomeSignatures"></a>GenomeSignatures</h1><p>_namespace: <a href="N-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.html">SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative</a>_</p>
<p>在本模块之中，所有的计算过程都是基于@”T:SMRUCC.genomics.SequenceModel.NucleotideModels.NucleicAcid”核酸对象的</p>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="counts"><a href="#counts" class="headerlink" title="__counts"></a>__counts</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.__counts(SMRUCC.genomics.SequenceModel.NucleotideModels.NucleicAcid,SMRUCC.genomics.SequenceModel.NucleotideModels.DNA[])</span><br></pre></td></tr></table></figure>
<p>计算某些连续的碱基片段在序列之中的出现频率</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sequence</td>
<td>-</td>
</tr>
<tr>
<td>ns</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="counts-p"><a href="#counts-p" class="headerlink" title="__counts_p"></a>__counts_p</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.__counts_p(SMRUCC.genomics.SequenceModel.NucleotideModels.NucleicAcid,SMRUCC.genomics.SequenceModel.NucleotideModels.DNA[])</span><br></pre></td></tr></table></figure>
<p>计算某些连续的碱基片段在序列之中的出现频率(并行版本)</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sequence</td>
<td>-</td>
</tr>
<tr>
<td>ns</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="CodonSignature"><a href="#CodonSignature" class="headerlink" title="CodonSignature"></a>CodonSignature</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.CodonSignature(SMRUCC.genomics.SequenceModel.NucleotideModels.NucleicAcid,SMRUCC.genomics.SequenceModel.NucleotideModels.Translation.Codon)</span><br></pre></td></tr></table></figure>
<p>CODON SIGNATURE</p>
<p> For a given collection of genes, let fX(1); fY(2); fZ(3) denote frequencies of the indicated nucleotide at codon sites 1, 2, and 3, respectively,<br> and let f(XYZ) indicate codon frequency. The embedded dinucleotide frequencies are denoted fXY(1, 2); fYZ(2, 3); and fXZ(1, 3). Dinucleotide<br> contrasts are assessed through the odds ratio pXY = f(XY)/f(X)f(Y).<br> In the context of codons, we define</p>
 <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">pXY(1, 2) = fXY(1, 2)/fX(1)fY(2)</span><br><span class="line">pYZ(2, 3) = fYZ(2, 3)/fY(2)fZ(3)</span><br><span class="line">pXZ(1, 3) = fXZ(1, 3)/fX(1)fZ(3)</span><br></pre></td></tr></table></figure>
<p> We refer to the profiles {pXY(1, 2)}; {pXZ(1, 3)}; {pYZ(2, 3)}, and also {pZW(3, 4)}, where 4(=1) is the first position of the next codon, as the<br> codon signature to be distinguished from the global genome signature</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sequence</td>
<td>-</td>
</tr>
<tr>
<td>Codon</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="DinucleotideBIAS"><a href="#DinucleotideBIAS" class="headerlink" title="DinucleotideBIAS"></a>DinucleotideBIAS</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.DinucleotideBIAS(SMRUCC.genomics.SequenceModel.NucleotideModels.NucleicAcid,SMRUCC.genomics.SequenceModel.NucleotideModels.DNA,SMRUCC.genomics.SequenceModel.NucleotideModels.DNA)</span><br></pre></td></tr></table></figure>
<p>Dinucleotide relative abundance values (dinucleotide bias) are assessed through the odds ratio <code>p(XY) = f(XY)/f(X)f(Y)</code>,<br> where <code>fX</code> denotes the frequency of the nucleotide <code>X</code> and <code>fXY</code> is the frequency of the dinucleotide <code>XY</code> in the<br> sequence under study.</p>
<h4 id="DinucleotideBIAS-p"><a href="#DinucleotideBIAS-p" class="headerlink" title="DinucleotideBIAS_p"></a>DinucleotideBIAS_p</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.DinucleotideBIAS_p(SMRUCC.genomics.SequenceModel.NucleotideModels.NucleicAcid,SMRUCC.genomics.SequenceModel.NucleotideModels.DNA,SMRUCC.genomics.SequenceModel.NucleotideModels.DNA)</span><br></pre></td></tr></table></figure>
<p>Dinucleotide relative abundance values (dinucleotide bias) are assessed through the<br> odds ratio <code>p(XY) = f(XY)/f(X)f(Y)</code>, where fX denotes the frequency of<br> the nucleotide X and fXY is the frequency of the dinucleotide XY in the<br> sequence under study.(并行版本)</p>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.html" class="article-date">
  <time datetime="2016-07-07T09:40:59.474Z" itemprop="datePublished">2016-07-07</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.GenomeSignatures.html" data-id="ciqc7cxed00r3igv4z3my11jo" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

