

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>gwANI | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="gwANI_namespace: SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI_
pANIto Given a multi-FASTA alignment, output the genome wide average nucleotide identity (gwANI) For Each sample against">
<meta property="og:type" content="website">
<meta property="og:title" content="gwANI">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI.gwANI.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="gwANI_namespace: SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI_
pANIto Given a multi-FASTA alignment, output the genome wide average nucleotide identity (gwANI) For Each sample against">
<meta property="og:updated_time" content="2016-07-07T09:40:59.779Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="gwANI">
<meta name="twitter:description" content="gwANI_namespace: SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI_
pANIto Given a multi-FASTA alignment, output the genome wide average nucleotide identity (gwANI) For Each sample against">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI.gwANI.html">
      <h1 class="article-title" itemprop="name">
        gwANI
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI.gwANI.html" class="article-date">
  <time datetime="2016-07-07T09:40:59.779Z" itemprop="datePublished">2016-07-07</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI.gwANI.html">
      <h1 class="article-title" itemprop="name">
        gwANI
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#gwANI"><span class="toc-number">1.</span> <span class="toc-text">gwANI</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#pANIto"><span class="toc-number">1.0.1.</span> <span class="toc-text">pANIto</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#Usage"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">Usage</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Input-format"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">Input format</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Output"><span class="toc-number">1.0.1.3.</span> <span class="toc-text">Output</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Etymology"><span class="toc-number">1.0.1.4.</span> <span class="toc-text">Etymology</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.2.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#fast-calculate-gwani"><span class="toc-number">1.0.2.1.</span> <span class="toc-text">__fast_calculate_gwani</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#calculate-and-output-gwani"><span class="toc-number">1.0.2.2.</span> <span class="toc-text">calculate_and_output_gwani</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#fast-calculate-gwani-1"><span class="toc-number">1.0.2.3.</span> <span class="toc-text">fast_calculate_gwani</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Properties"><span class="toc-number">1.0.3.</span> <span class="toc-text">Properties</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#out"><span class="toc-number">1.0.3.1.</span> <span class="toc-text">out</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="gwANI"><a href="#gwANI" class="headerlink" title="gwANI"></a>gwANI</h1><p>_namespace: <a href="N-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI.html">SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI</a>_</p>
<h3 id="pANIto"><a href="#pANIto" class="headerlink" title="pANIto"></a>pANIto</h3><p> Given a multi-FASTA alignment, output the genome wide average nucleotide identity (gwANI)<br> For Each sample against all other samples. A matrix containing the percentages Is outputted.<br> This software loads the whole file into memory.</p>
<h4 id="Usage"><a href="#Usage" class="headerlink" title="Usage"></a>Usage</h4> <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">$ panito</span><br><span class="line">Usage: panito [-hV] &lt;file&gt;</span><br><span class="line">This program calculates the genome wide ANI for a multiFASTA alignment.</span><br><span class="line">-h this help message</span><br><span class="line">-V print version and exit</span><br><span class="line">&lt;file&gt; input alignment file which can optionally be gzipped</span><br></pre></td></tr></table></figure>
<h4 id="Input-format"><a href="#Input-format" class="headerlink" title="Input format"></a>Input format</h4><p> The input file must be a multi-FASTA file, where all sequences are the same length:</p>
 <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">&gt;sample1</span><br><span class="line">AAAAAAAAAA</span><br><span class="line">&gt;sample2</span><br><span class="line">AAAAAAAAAC</span><br><span class="line">&gt;sample3</span><br><span class="line">AAAAAAAACC</span><br></pre></td></tr></table></figure>
<h4 id="Output"><a href="#Output" class="headerlink" title="Output"></a>Output</h4> <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">sample1 sample2 sample3</span><br><span class="line">sample1100.00000090.0000080.000000</span><br><span class="line">sample2-100.00000090.000000</span><br><span class="line">sample3--100.000000</span><br></pre></td></tr></table></figure>
<h4 id="Etymology"><a href="#Etymology" class="headerlink" title="Etymology"></a>Etymology</h4><p> pANIto has ‘ani’ in the middle. In Spanish it means babylon.</p>
<blockquote>
 <figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><span class="line">&gt;  </span><br><span class="line">&gt;  Public Sub print_version()</span><br><span class="line">&gt;      Call Console.Write("&#123;0&#125; &#123;1&#125;" &amp; vbLf, DefineConstants.PROGRAM_NAME, PACKAGE_VERSION)</span><br><span class="line">&gt;  End Sub</span><br><span class="line">&gt; </span><br><span class="line">&gt;  Public Sub Main(argc As Integer, args As String())</span><br><span class="line">&gt;      Dim multi_fasta_filename As String() = &#123;""&#125;</span><br><span class="line">&gt;      Dim output_filename As String() = &#123;""&#125;</span><br><span class="line">&gt; </span><br><span class="line">&gt;      Dim c As Integer</span><br><span class="line">&gt;      Dim index As Integer</span><br><span class="line">&gt;      Dim output_multi_fasta_file As Integer = 0</span><br><span class="line">&gt; </span><br><span class="line">&gt;      While (InlineAssignHelper(c, getopt(argc, args, "ho:V"))) &lt;&gt; -1</span><br><span class="line">&gt;          Select Case c</span><br><span class="line">&gt;              Case "V"c</span><br><span class="line">&gt;                  GlobalMembersMain.print_version()</span><br><span class="line">&gt;             </span><br><span class="line">&gt;              Case "o"c</span><br><span class="line">&gt;                  output_filename = optarg.Substring(0, FILENAME_MAX)</span><br><span class="line">&gt;                   </span><br><span class="line">&gt;              Case "h"c</span><br><span class="line">&gt;                  GlobalMembersMain.print_usage()</span><br><span class="line">&gt;                  </span><br><span class="line">&gt;              Case Else</span><br><span class="line">&gt;                  output_multi_fasta_file = 1</span><br><span class="line">&gt;                    </span><br><span class="line">&gt;          End Select</span><br><span class="line">&gt;      End While</span><br><span class="line">&gt; </span><br><span class="line">&gt;      If optind &lt; argc Then</span><br><span class="line">&gt;          multi_fasta_filename = Convert.ToString(args(optind)).Substring(0, FILENAME_MAX)</span><br><span class="line">&gt;          gwANI.calculate_and_output_gwani(multi_fasta_filename);</span><br><span class="line">&gt;          gwANI.fast_calculate_gwani(multi_fasta_filename)</span><br><span class="line">&gt;      Else</span><br><span class="line">&gt;          Call print_usage()</span><br><span class="line">&gt;      End If</span><br><span class="line">&gt;  End Sub</span><br><span class="line">&gt;</span><br></pre></td></tr></table></figure>
</blockquote>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="fast-calculate-gwani"><a href="#fast-calculate-gwani" class="headerlink" title="__fast_calculate_gwani"></a>__fast_calculate_gwani</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI.gwANI.__fast_calculate_gwani(System.String@)</span><br></pre></td></tr></table></figure>
<p>执行入口点</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>filename</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="calculate-and-output-gwani"><a href="#calculate-and-output-gwani" class="headerlink" title="calculate_and_output_gwani"></a>calculate_and_output_gwani</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI.gwANI.calculate_and_output_gwani(System.String@,System.IO.TextWriter)</span><br></pre></td></tr></table></figure>
<p>执行入口点</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>filename</td>
<td>-</td>
</tr>
<tr>
<td>out</td>
<td>默认是打印在终端之上</td>
</tr>
</tbody>
</table>
<h4 id="fast-calculate-gwani-1"><a href="#fast-calculate-gwani-1" class="headerlink" title="fast_calculate_gwani"></a>fast_calculate_gwani</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI.gwANI.fast_calculate_gwani(System.String@,System.IO.TextWriter)</span><br></pre></td></tr></table></figure>
<p>执行入口点</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>filename</td>
<td>-</td>
</tr>
<tr>
<td>out</td>
<td>默认是打印在终端之上</td>
</tr>
</tbody>
</table>
<h3 id="Properties"><a href="#Properties" class="headerlink" title="Properties"></a>Properties</h3><h4 id="out"><a href="#out" class="headerlink" title="out"></a>out</h4><p>The result output stream</p>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI.gwANI.html" class="article-date">
  <time datetime="2016-07-07T09:40:59.779Z" itemprop="datePublished">2016-07-07</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Analysis.SequenceTools.DNA_Comparative.gwANI.gwANI.html" data-id="ciqc7cxf700riigv4ezyu8g11" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

