

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>Feature | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="Featurenamespace: SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat
A feature is here an interval (i.e., a range of positions) on a chromosome or a union of such intervals. (Feature是基因组序列片段之上的一个具备有生">
<meta property="og:type" content="website">
<meta property="og:title" content="Feature">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.Feature.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="Featurenamespace: SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat
A feature is here an interval (i.e., a range of positions) on a chromosome or a union of such intervals. (Feature是基因组序列片段之上的一个具备有生">
<meta property="og:updated_time" content="2016-07-27T11:12:34.782Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Feature">
<meta name="twitter:description" content="Featurenamespace: SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat
A feature is here an interval (i.e., a range of positions) on a chromosome or a union of such intervals. (Feature是基因组序列片段之上的一个具备有生">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.Feature.html">
      <h1 class="article-title" itemprop="name">
        Feature
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.Feature.html" class="article-date">
  <time datetime="2016-07-27T11:12:34.782Z" itemprop="datePublished">2016-07-27</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.Feature.html">
      <h1 class="article-title" itemprop="name">
        Feature
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#Feature"><span class="toc-number">1.</span> <span class="toc-text">Feature</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Properties"><span class="toc-number">1.0.1.</span> <span class="toc-text">Properties</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#attributes"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">attributes</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#comments"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">comments</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Ends"><span class="toc-number">1.0.1.3.</span> <span class="toc-text">Ends</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Feature-1"><span class="toc-number">1.0.1.4.</span> <span class="toc-text">Feature</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#frame"><span class="toc-number">1.0.1.5.</span> <span class="toc-text">frame</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#ID"><span class="toc-number">1.0.1.6.</span> <span class="toc-text">ID</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#ProteinId"><span class="toc-number">1.0.1.7.</span> <span class="toc-text">ProteinId</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#score"><span class="toc-number">1.0.1.8.</span> <span class="toc-text">score</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#seqname"><span class="toc-number">1.0.1.9.</span> <span class="toc-text">seqname</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#source"><span class="toc-number">1.0.1.10.</span> <span class="toc-text">source</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#start"><span class="toc-number">1.0.1.11.</span> <span class="toc-text">start</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Strand"><span class="toc-number">1.0.1.12.</span> <span class="toc-text">Strand</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="Feature"><a href="#Feature" class="headerlink" title="Feature"></a>Feature</h1><p><em>namespace: <a href="N-SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.html">SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat</a></em></p>
<p>A feature is here an interval (i.e., a range of positions) on a chromosome or a union of such intervals.<br> (Feature是基因组序列片段之上的一个具备有生物学功能意义的区域，故而这个对象继承自核酸位点对象)</p>
<p> In the case of RNA-Seq, the features are typically genes, where each gene is considered here as the union of all its exons.<br> One may also consider each exon as a feature, e.g., in order to check for alternative splicing. </p>
<p> For comparative ChIP-Seq, the features might be binding region from a pre-determined list.</p>
<h3 id="Properties"><a href="#Properties" class="headerlink" title="Properties"></a>Properties</h3><h4 id="attributes"><a href="#attributes" class="headerlink" title="attributes"></a>attributes</h4><p>From version 2 onwards, the attribute field must have an tag value structure following the syntax used within objects in<br> a .ace file, flattened onto one line by semicolon separators. Tags must be standard identifiers ([A-Za-z][A-Za-z0-9_]*).<br> Free text values must be quoted with double quotes. Note: all non-printing characters in such free text value strings<br> (e.g. newlines, tabs, control characters, etc) must be explicitly represented by their C (UNIX) style backslash-escaped<br> representation (e.g. newlines as ‘\n’, tabs as ‘\t’). As in ACEDB, multiple values can follow a specific tag. The aim is<br> to establish consistent use of particular tags, corresponding to an underlying implied ACEDB model if you want to think<br> that way (but acedb is not required). Examples of these would be:</p>
<p> seq1 BLASTX similarity 101 235 87.1 + 0 Target “HBA_HUMAN” 11 55 ; E_value 0.0003<br> dJ102G20 GD_mRNA coding_exon 7105 7201 . - 2 Sequence “dJ102G20.C1.1”</p>
<p> The semantics Of tags In attribute field tag-values pairs has intentionally Not been formalized. Two useful guidelines are<br> To use DDBJ/EMBL/GenBank feature ‘qualifiers’ (see DDBJ/EMBL/GenBank feature table documentation), or the features that<br> ACEDB generates when it dumps GFF. Version 1 note In version 1 the attribute field was called the group field, with the<br> following specification: An optional string-valued field that can be used as a name to group together a set of records.<br> Typical uses might be to group the introns and exons in one gene prediction (or experimentally verified gene structure),<br> or to group multiple regions of match to another sequence, such as an EST or a protein.<br> (请注意，所有的key都已经被转换为小写的形式了)</p>
<h4 id="comments"><a href="#comments" class="headerlink" title="comments"></a>comments</h4><p>Comments are allowed, starting with “#” as in Perl, awk etc. Everything following # until the end of the line is ignored.<br> Effectively this can be used in two ways. Either it must be at the beginning of the line (after any whitespace), to make<br> the whole line a comment, or the comment could come after all the required fields on the line.</p>
<h4 id="Ends"><a href="#Ends" class="headerlink" title="Ends"></a>Ends</h4><p>Integers. <start> must be less than or equal to <end>. Sequence numbering starts at 1, so these numbers<br> should be between 1 and the length of the relevant sequence, inclusive. </end></start></p>
<p> (Version 2 change: version 2 condones values of <start> and <end> that extend outside the reference sequence.<br> This is often more natural when dumping from acedb, rather than clipping. It means that some software using the<br> files may need to clip for itself.)</end></start></p>
<h4 id="Feature-1"><a href="#Feature-1" class="headerlink" title="Feature"></a>Feature</h4><p>The feature type name. We hope to suggest a standard set of features, to facilitate import/export, comparison etc..<br> Of course, people are free to define new ones as needed. For example, Genie splice detectors account for a region<br> of DNA, and multiple detectors may be available for the same site, as shown above. We would like to enforce a<br> standard nomenclature for common GFF features. This does not forbid the use of other features, rather, just that<br> if the feature is obviously described in the standard list, that the standard label should be used. For this standard<br> table we propose to fall back on the international public standards for genomic database feature annotation,<br> specifically, the DDBJ/EMBL/GenBank feature table documentation).</p>
<h4 id="frame"><a href="#frame" class="headerlink" title="frame"></a>frame</h4><p>One of ‘0’, ‘1’, ‘2’ or ‘.’. ‘0’ indicates that the specified region is in frame, i.e. that its first base corresponds to<br> the first base of a codon. ‘1’ indicates that there is one extra base, i.e. that the second base of the region corresponds<br> to the first base of a codon, and ‘2’ means that the third base of the region is the first base of a codon. </p>
<p> If the strand is ‘-‘, then the first base of the region is value of <end>, because the corresponding coding region will run<br> from <end> to <start> on the reverse strand. As with <strand>, if the frame is not relevant then set <frame> to ‘.’.<br> It has been pointed out that “phase” might be a better descriptor than “frame” for this field. </strand></start></end></end></p>
<p> Version 2 change: This field is left empty ‘.’ for RNA and protein features.</p>
<h4 id="ID"><a href="#ID" class="headerlink" title="ID"></a>ID</h4><p>请注意，这个属性不是基因号</p>
<h4 id="ProteinId"><a href="#ProteinId" class="headerlink" title="ProteinId"></a>ProteinId</h4><h4 id="score"><a href="#score" class="headerlink" title="score"></a>score</h4><p>A floating point value. When there is no score (i.e. for a sensor that just records the possible presence of a signal,<br> as for the EMBL features above) you should use ‘.’. </p>
<p> (Version 2 change: in version 1 of GFF you had to write 0 in such circumstances.)</p>
<h4 id="seqname"><a href="#seqname" class="headerlink" title="seqname"></a>seqname</h4><p>The name of the sequence. Having an explicit sequence name allows a feature file to be prepared for a data set<br> of multiple sequences. Normally the seqname will be the identifier of the sequence in an accompanying fasta<br> format file. An alternative is that <seqname> is the identifier for a sequence in a public database, such as<br> an EMBL/Genbank/DDBJ accession number. Which is the case, and which file or database to use, should be explained<br> in accompanying information.</seqname></p>
<h4 id="source"><a href="#source" class="headerlink" title="source"></a>source</h4><p>The source of this feature. This field will normally be used to indicate the program making the prediction,<br> or if it comes from public database annotation, or is experimentally verified, etc.</p>
<h4 id="start"><a href="#start" class="headerlink" title="start"></a>start</h4><p>Integers. <start> must be less than or equal to <end>. Sequence numbering starts at 1, so these numbers<br> should be between 1 and the length of the relevant sequence, inclusive. </end></start></p>
<p> (Version 2 change: version 2 condones values of <start> and <end> that extend outside the reference sequence.<br> This is often more natural when dumping from acedb, rather than clipping. It means that some software using the<br> files may need to clip for itself.)</end></start></p>
<h4 id="Strand"><a href="#Strand" class="headerlink" title="Strand"></a>Strand</h4><p>One of ‘+’, ‘-‘ or ‘.’. ‘.’ should be used when strand is not relevant, e.g. for dinucleotide repeats. </p>
<p> Version 2 change: This field is left empty ‘.’ for RNA and protein features.</p>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.Feature.html" class="article-date">
  <time datetime="2016-07-27T11:12:34.782Z" itemprop="datePublished">2016-07-27</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.Feature.html" data-id="ciqc7drc900yzigv46a0l5x9c" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

