

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>GSW`1 | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="GSW`1namespace: SMRUCC.genomics.Analysis.SequenceTools
Generic Smith-Waterman computing kernel.(Smith-Waterman泛型化通用计算核心)
Methods#ctor1SMRUCC.genomics.Analysis.SequenceTools.GSW`1.#ctor(`0[],`0[],SMRUC">
<meta property="og:type" content="website">
<meta property="og:title" content="GSW`1">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Analysis.SequenceTools.GSW`1.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="GSW`1namespace: SMRUCC.genomics.Analysis.SequenceTools
Generic Smith-Waterman computing kernel.(Smith-Waterman泛型化通用计算核心)
Methods#ctor1SMRUCC.genomics.Analysis.SequenceTools.GSW`1.#ctor(`0[],`0[],SMRUC">
<meta property="og:updated_time" content="2016-07-27T11:12:33.238Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="GSW`1">
<meta name="twitter:description" content="GSW`1namespace: SMRUCC.genomics.Analysis.SequenceTools
Generic Smith-Waterman computing kernel.(Smith-Waterman泛型化通用计算核心)
Methods#ctor1SMRUCC.genomics.Analysis.SequenceTools.GSW`1.#ctor(`0[],`0[],SMRUC">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.GSW`1.html">
      <h1 class="article-title" itemprop="name">
        GSW`1
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.GSW`1.html" class="article-date">
  <time datetime="2016-07-27T11:12:33.238Z" itemprop="datePublished">2016-07-27</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.GSW`1.html">
      <h1 class="article-title" itemprop="name">
        GSW`1
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#GSW-1"><span class="toc-number">1.</span> <span class="toc-text">GSW`1</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.1.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#ctor"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">#ctor</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#buildMatrix"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">__buildMatrix</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#similarity"><span class="toc-number">1.0.1.3.</span> <span class="toc-text">__similarity</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#GetDPMAT"><span class="toc-number">1.0.1.4.</span> <span class="toc-text">GetDPMAT</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#GetTraceback"><span class="toc-number">1.0.1.5.</span> <span class="toc-text">GetTraceback</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#printAlignments"><span class="toc-number">1.0.1.6.</span> <span class="toc-text">printAlignments</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#printDPMatrix"><span class="toc-number">1.0.1.7.</span> <span class="toc-text">printDPMatrix</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#traceback"><span class="toc-number">1.0.1.8.</span> <span class="toc-text">traceback</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Properties"><span class="toc-number">1.0.2.</span> <span class="toc-text">Properties</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#AlignmentScore"><span class="toc-number">1.0.2.1.</span> <span class="toc-text">AlignmentScore</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#DR-LEFT"><span class="toc-number">1.0.2.2.</span> <span class="toc-text">DR_LEFT</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#MATCH-SCORE"><span class="toc-number">1.0.2.3.</span> <span class="toc-text">MATCH_SCORE</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Matches"><span class="toc-number">1.0.2.4.</span> <span class="toc-text">Matches</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#MaxScore"><span class="toc-number">1.0.2.5.</span> <span class="toc-text">MaxScore</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#NORM-FACTOR"><span class="toc-number">1.0.2.6.</span> <span class="toc-text">NORM_FACTOR</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#prevCells"><span class="toc-number">1.0.2.7.</span> <span class="toc-text">prevCells</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#query"><span class="toc-number">1.0.2.8.</span> <span class="toc-text">query</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#queryLength"><span class="toc-number">1.0.2.9.</span> <span class="toc-text">queryLength</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#score"><span class="toc-number">1.0.2.10.</span> <span class="toc-text">score</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#similarity-1"><span class="toc-number">1.0.2.11.</span> <span class="toc-text">similarity</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#subject"><span class="toc-number">1.0.2.12.</span> <span class="toc-text">subject</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#subjectLength"><span class="toc-number">1.0.2.13.</span> <span class="toc-text">subjectLength</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="GSW-1"><a href="#GSW-1" class="headerlink" title="GSW`1"></a>GSW`1</h1><p><em>namespace: <a href="N-SMRUCC.genomics.Analysis.SequenceTools.html">SMRUCC.genomics.Analysis.SequenceTools</a></em></p>
<p>Generic Smith-Waterman computing kernel.(Smith-Waterman泛型化通用计算核心)</p>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="ctor"><a href="#ctor" class="headerlink" title="#ctor"></a>#ctor</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.GSW`<span class="number">1.</span><span class="meta">#ctor(`0[],`0[],SMRUCC.genomics.Analysis.SequenceTools.ISimilarity&#123;`0&#125;,Microsoft.VisualBasic.LevenshteinDistance.ToChar&#123;`0&#125;)</span></span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>query</td>
<td>-</td>
</tr>
<tr>
<td>subject</td>
<td>-</td>
</tr>
<tr>
<td>similarity</td>
<td>Blosum matrix or motif similarity</td>
</tr>
<tr>
<td>asChar</td>
<td>Display alignment</td>
</tr>
</tbody>
</table>
<h4 id="buildMatrix"><a href="#buildMatrix" class="headerlink" title="__buildMatrix"></a>__buildMatrix</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.GSW`<span class="number">1.</span>__buildMatrix</span><br></pre></td></tr></table></figure>
<p>Build the score matrix using dynamic programming.<br> Note: The indel scores must be negative. Otherwise, the<br> part handling the first row and column has to be<br> modified.</p>
<h4 id="similarity"><a href="#similarity" class="headerlink" title="__similarity"></a>__similarity</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.GSW`<span class="number">1.</span>__similarity(System.Int32,System.Int32)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>i</td>
<td>Position of the character in str1</td>
</tr>
<tr>
<td>j</td>
<td>Position of the character in str2</td>
</tr>
</tbody>
</table>
<p><em>returns:  Cost of substitution of the character in str1 by the one in str2 </em></p>
<h4 id="GetDPMAT"><a href="#GetDPMAT" class="headerlink" title="GetDPMAT"></a>GetDPMAT</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.GSW`<span class="number">1.</span>GetDPMAT</span><br></pre></td></tr></table></figure>
<p>Gets the dynmaic programming matrix</p>
<h4 id="GetTraceback"><a href="#GetTraceback" class="headerlink" title="GetTraceback"></a>GetTraceback</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.GSW`<span class="number">1.</span>GetTraceback</span><br></pre></td></tr></table></figure>
<p>See this overview on implementing Smith-Waterman, including trace back. At each location in the matrix you should check 4 things:</p>
<p> If the location value equals the gap penalty plus the location above, up Is a valid move.<br> If the location value equals the gap penalty plus the location left, left Is a valid move.<br> If the location value equals the match value plus the location up And left, diagonal Is a valid move.<br> If the location value Is 0, you’re done.</p>
<p> The first And second options correlate To inserting a gap In one Of the strings, And the third correlates To aligning two characters.<br> If multiple paths work, Then you have multiple possible alignments. </p>
<p> As the article states, the decision at that point depends largely On context (you have several options).</p>
<h4 id="printAlignments"><a href="#printAlignments" class="headerlink" title="printAlignments"></a>printAlignments</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.GSW`<span class="number">1.</span>printAlignments</span><br></pre></td></tr></table></figure>
<p>Output the local alignments with the maximum score.</p>
<h4 id="printDPMatrix"><a href="#printDPMatrix" class="headerlink" title="printDPMatrix"></a>printDPMatrix</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.GSW`<span class="number">1.</span>printDPMatrix</span><br></pre></td></tr></table></figure>
<p>print the dynmaic programming matrix</p>
<h4 id="traceback"><a href="#traceback" class="headerlink" title="traceback"></a>traceback</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.SequenceTools.GSW`<span class="number">1.</span>traceback(System.Int32,System.Int32)</span><br></pre></td></tr></table></figure>
<p>given the bottom right corner point trace back the top left conrner.<br> at entry: i, j hold bottom right (end of Aligment coords)<br> at return: hold top left (start of Alignment coords)</p>
<h3 id="Properties"><a href="#Properties" class="headerlink" title="Properties"></a>Properties</h3><h4 id="AlignmentScore"><a href="#AlignmentScore" class="headerlink" title="AlignmentScore"></a>AlignmentScore</h4><p>Get the alignment score between the two input strings.</p>
<h4 id="DR-LEFT"><a href="#DR-LEFT" class="headerlink" title="DR_LEFT"></a>DR_LEFT</h4><p>Constants of directions.<br> Multiple directions are stored by bits.<br> The zero direction is the starting point.</p>
<h4 id="MATCH-SCORE"><a href="#MATCH-SCORE" class="headerlink" title="MATCH_SCORE"></a>MATCH_SCORE</h4><p>The similarity function constants.<br> They are amplified by the normalization factor to be integers.</p>
<h4 id="Matches"><a href="#Matches" class="headerlink" title="Matches"></a>Matches</h4><p>Return a set of Matches idenfied in Dynamic programming matrix.<br> A match is a pair of subsequences whose score is higher than the<br> preset scoreThreshold</p>
<h4 id="MaxScore"><a href="#MaxScore" class="headerlink" title="MaxScore"></a>MaxScore</h4><p>Get the maximum value in the score matrix.</p>
<h4 id="NORM-FACTOR"><a href="#NORM-FACTOR" class="headerlink" title="NORM_FACTOR"></a>NORM_FACTOR</h4><p>The normalization factor.<br> To get the true score, divide the integer score used in computation<br> by the normalization factor.</p>
<h4 id="prevCells"><a href="#prevCells" class="headerlink" title="prevCells"></a>prevCells</h4><p>The directions pointing to the cells that<br> give the maximum score at the current cell.<br> The first index is the column index.<br> The second index is the row index.</p>
<h4 id="query"><a href="#query" class="headerlink" title="query"></a>query</h4><p>The first input string</p>
<h4 id="queryLength"><a href="#queryLength" class="headerlink" title="queryLength"></a>queryLength</h4><p>The lengths of the input strings</p>
<h4 id="score"><a href="#score" class="headerlink" title="score"></a>score</h4><p>The score matrix.<br> The true scores should be divided by the normalization factor.</p>
<h4 id="similarity-1"><a href="#similarity-1" class="headerlink" title="similarity"></a>similarity</h4><p>Compute the similarity score of substitution: use a substitution matrix if the cost model<br> The position of the first character is 1.<br> A position of 0 represents a gap.</p>
<h4 id="subject"><a href="#subject" class="headerlink" title="subject"></a>subject</h4><p>The second input String</p>
<h4 id="subjectLength"><a href="#subjectLength" class="headerlink" title="subjectLength"></a>subjectLength</h4><p>The lengths of the input strings</p>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-SMRUCC.genomics.Analysis.SequenceTools.GSW`1.html" class="article-date">
  <time datetime="2016-07-27T11:12:33.238Z" itemprop="datePublished">2016-07-27</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Analysis.SequenceTools.GSW`1.html" data-id="cir4v17el006o5wv4iygpbh33" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

