

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>SAM | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="SAMnamespace: SMRUCC.genomics.SequenceModel.SAM
The Sequence Alignment/Map (SAM) format is a generic nucleotide alignment format that describes the alignment of query sequences or sequencing reads to">
<meta property="og:type" content="website">
<meta property="og:title" content="SAM">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.SequenceModel.SAM.SAM.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="SAMnamespace: SMRUCC.genomics.SequenceModel.SAM
The Sequence Alignment/Map (SAM) format is a generic nucleotide alignment format that describes the alignment of query sequences or sequencing reads to">
<meta property="og:updated_time" content="2016-07-27T11:12:46.615Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="SAM">
<meta name="twitter:description" content="SAMnamespace: SMRUCC.genomics.SequenceModel.SAM
The Sequence Alignment/Map (SAM) format is a generic nucleotide alignment format that describes the alignment of query sequences or sequencing reads to">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-SMRUCC.genomics.SequenceModel.SAM.SAM.html">
      <h1 class="article-title" itemprop="name">
        SAM
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-SMRUCC.genomics.SequenceModel.SAM.SAM.html" class="article-date">
  <time datetime="2016-07-27T11:12:46.615Z" itemprop="datePublished">2016-07-27</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-SMRUCC.genomics.SequenceModel.SAM.SAM.html">
      <h1 class="article-title" itemprop="name">
        SAM
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#SAM"><span class="toc-number">1.</span> <span class="toc-text">SAM</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.1.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#Assembling"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">Assembling</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Load"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">Load</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#TrimUnmappedReads"><span class="toc-number">1.0.1.3.</span> <span class="toc-text">TrimUnmappedReads</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Properties"><span class="toc-number">1.0.2.</span> <span class="toc-text">Properties</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#AlignmentsReads"><span class="toc-number">1.0.2.1.</span> <span class="toc-text">AlignmentsReads</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Head"><span class="toc-number">1.0.2.2.</span> <span class="toc-text">Head</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="SAM"><a href="#SAM" class="headerlink" title="SAM"></a>SAM</h1><p><em>namespace: <a href="N-SMRUCC.genomics.SequenceModel.SAM.html">SMRUCC.genomics.SequenceModel.SAM</a></em></p>
<p>The Sequence Alignment/Map (SAM) format is a generic nucleotide alignment format that describes the alignment of query sequences or sequencing reads to a reference sequence or assembly.<br> Importantly:</p>
<p> It is flexible enough to store all the alignment information generated by various alignment programs;<br> It is simple enough to be easily generated by alignment programs or converted from existing alignment formats;<br> It is compact in file size;<br> It allows most of the operations on the alignment to work on a stream without loading the whole alignment into memory;<br> It allows the file to be indexed by genomic position to efficiently retrieve all reads aligning to a locus.</p>
<p> SAM is a tab-delimited text format. SAM is a bit slow to parse; so there is a binary equivalent to SAM, called BAM.</p>
<p> SAM allows optional fields to be stored. In SAM, each alignment must contain a fixed number of mandatory fields that describe<br> the key information about the alignment (such as coordinate detailed alignment and sequences) and may contain a variable<br> number of optional fields which are less important or aligner specific.</p>
<p> SAM is able to store clipped alignments, spliced alignments, multi-part alignments, padded alignments and alignments in colour space.<br> The extended CIGAR string is the key to describing these types of alignments.</p>
<blockquote>
<p> SAM stands for Sequence Alignment/Map format. It is a TAB-delimited text format consisting of a<br> header section, which is optional, and an alignment section. If present, the header must be prior to<br> the alignments. Header lines start with `@’, while alignment lines do not. Each alignment line has 11<br> mandatory fields for essential alignment information such as mapping position, and variable number of<br> optional fields for flexible or aligner specific information.</p>
<p> SAM格式的文件是一种序列比对文件，使用TAB符号进行分隔，文件的格式为一个可选的标题头部区域，标题头部使用@符号起始而比对区域则不需要<br> 每一行序列比对的数据有11个域用于储存比对信息，诸如：mapping的位置之类</p>
</blockquote>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="Assembling"><a href="#Assembling" class="headerlink" title="Assembling"></a>Assembling</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.SequenceModel.SAM.SAM.Assembling(System.Collections.Generic.Dictionary&#123;System.Int32,Microsoft.VisualBasic.List&#123;SMRUCC.genomics.SequenceModel.SAM.AlignmentReads&#125;&#125;,System.Boolean)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Alignment</td>
<td>请注意先按照方向排序</td>
</tr>
<tr>
<td>Reversed</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="Load"><a href="#Load" class="headerlink" title="Load"></a>Load</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.SequenceModel.SAM.SAM.Load(System.String,System.Text.Encoding)</span><br></pre></td></tr></table></figure>
<p>从一个文本文件之中加载一个SAM格式的文件数据</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Path</td>
<td>-</td>
</tr>
<tr>
<td>encoding</td>
<td>Default value is utf8 encoding.</td>
</tr>
</tbody>
</table>
<h4 id="TrimUnmappedReads"><a href="#TrimUnmappedReads" class="headerlink" title="TrimUnmappedReads"></a>TrimUnmappedReads</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.SequenceModel.SAM.SAM.TrimUnmappedReads</span><br></pre></td></tr></table></figure>
<p>移除没有被mapping到基因组上面的reads</p>
<h3 id="Properties"><a href="#Properties" class="headerlink" title="Properties"></a>Properties</h3><h4 id="AlignmentsReads"><a href="#AlignmentsReads" class="headerlink" title="AlignmentsReads"></a>AlignmentsReads</h4><p>The object of this value is the details alignment data.</p>
<h4 id="Head"><a href="#Head" class="headerlink" title="Head"></a>Head</h4><p>If present, the header must be prior to the alignments. Header lines start With `@’, while alignment lines do not.<br> (文件的可选头部区域必须要在比对数据区域的前面并且每一行以@符号开始)</p>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-SMRUCC.genomics.SequenceModel.SAM.SAM.html" class="article-date">
  <time datetime="2016-07-27T11:12:46.615Z" itemprop="datePublished">2016-07-27</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.SequenceModel.SAM.SAM.html" data-id="ciqc7dxqz027rigv41moyh971" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

