

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>MotifFootPrintsGenerates | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="MotifFootPrintsGenerates_namespace: SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints_
这里所定义的所有对象都是和数据解析无关了，都是用于进行数据存储的对象类型
Methods__checkMoitfCoRegulations1SMRUCC.genomics.Inter">
<meta property="og:type" content="website">
<meta property="og:title" content="MotifFootPrintsGenerates">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="MotifFootPrintsGenerates_namespace: SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints_
这里所定义的所有对象都是和数据解析无关了，都是用于进行数据存储的对象类型
Methods__checkMoitfCoRegulations1SMRUCC.genomics.Inter">
<meta property="og:updated_time" content="2016-07-27T11:12:44.680Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="MotifFootPrintsGenerates">
<meta name="twitter:description" content="MotifFootPrintsGenerates_namespace: SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints_
这里所定义的所有对象都是和数据解析无关了，都是用于进行数据存储的对象类型
Methods__checkMoitfCoRegulations1SMRUCC.genomics.Inter">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.html">
      <h1 class="article-title" itemprop="name">
        MotifFootPrintsGenerates
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.html" class="article-date">
  <time datetime="2016-07-27T11:12:44.680Z" itemprop="datePublished">2016-07-27</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.html">
      <h1 class="article-title" itemprop="name">
        MotifFootPrintsGenerates
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#MotifFootPrintsGenerates"><span class="toc-number">1.</span> <span class="toc-text">MotifFootPrintsGenerates</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.1.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#checkMoitfCoRegulations"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">__checkMoitfCoRegulations</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#reGenerate"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">__reGenerate</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#CreateMotifInformations"><span class="toc-number">1.0.1.3.</span> <span class="toc-text">CreateMotifInformations</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#DataFilteringByPathwayCoExpression"><span class="toc-number">1.0.1.4.</span> <span class="toc-text">DataFilteringByPathwayCoExpression</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Diff"><span class="toc-number">1.0.1.5.</span> <span class="toc-text">Diff</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#ExportRegulators"><span class="toc-number">1.0.1.6.</span> <span class="toc-text">ExportRegulators</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#FootPrintMatches"><span class="toc-number">1.0.1.7.</span> <span class="toc-text">FootPrintMatches</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#FootprintMatchesTEXT"><span class="toc-number">1.0.1.8.</span> <span class="toc-text">FootprintMatchesTEXT</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#GenerateNetwork"><span class="toc-number">1.0.1.9.</span> <span class="toc-text">GenerateNetwork</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#GroupMotifs"><span class="toc-number">1.0.1.10.</span> <span class="toc-text">GroupMotifs</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Matches"><span class="toc-number">1.0.1.11.</span> <span class="toc-text">Matches</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#MatchRegulator"><span class="toc-number">1.0.1.12.</span> <span class="toc-text">MatchRegulator</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#PathwayFunctionAssociation"><span class="toc-number">1.0.1.13.</span> <span class="toc-text">PathwayFunctionAssociation</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#PccAssumption"><span class="toc-number">1.0.1.14.</span> <span class="toc-text">PccAssumption</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="MotifFootPrintsGenerates"><a href="#MotifFootPrintsGenerates" class="headerlink" title="MotifFootPrintsGenerates"></a>MotifFootPrintsGenerates</h1><p>_namespace: <a href="N-SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.html">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints</a>_</p>
<p>这里所定义的所有对象都是和数据解析无关了，都是用于进行数据存储的对象类型</p>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="checkMoitfCoRegulations"><a href="#checkMoitfCoRegulations" class="headerlink" title="__checkMoitfCoRegulations"></a>__checkMoitfCoRegulations</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.__checkMoitfCoRegulations(SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.PredictedRegulationFootprint[])</span><br></pre></td></tr></table></figure>
<p>假若目标Regulator可以被<strong>groupedMotif</strong>里面的多余60%的对象所具备，则认为该调控因子对目标motif的调控关系是可能存在的</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>groupedMotif</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="reGenerate"><a href="#reGenerate" class="headerlink" title="__reGenerate"></a>__reGenerate</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.__reGenerate(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.PredictedRegulationFootprint&#125;,SMRUCC.genomics.SequenceModel.NucleotideModels.SegmentReader)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>grouped</td>
<td>已经按照@”P:SMRUCC.genomics.Model.Network.VirtualFootprint.DocumentFormat.VirtualFootprints.Starts”属性进行排序</td>
</tr>
</tbody>
</table>
<h4 id="CreateMotifInformations"><a href="#CreateMotifInformations" class="headerlink" title="CreateMotifInformations"></a>CreateMotifInformations</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.CreateMotifInformations(System.String,SMRUCC.genomics.Data.Regprecise.TranscriptionFactors)</span><br></pre></td></tr></table></figure>
<p>将Regprecise数据库之中的Moitf数据解析出来进行MEME分析之后，将MEME_OUT文件夹里面的数据整理出来</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>MotifMEME_OUT</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="DataFilteringByPathwayCoExpression"><a href="#DataFilteringByPathwayCoExpression" class="headerlink" title="DataFilteringByPathwayCoExpression"></a>DataFilteringByPathwayCoExpression</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.DataFilteringByPathwayCoExpression(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.PredictedRegulationFootprint&#125;,System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Assembly.KEGG.Archives.Csv.Pathway&#125;,System.Boolean)</span><br></pre></td></tr></table></figure>
<p>通过同一个代谢途径之中的所有的基因可能共表达的生物学知识，在缺乏转录组数据的条件之下进行数据的筛选，算法的要点：<br> 假若同一个代谢途径之中的基因，有65%或者以上的基因具有同一个motif位点，则该motif位点可能是正确的位点</p>
<p> 本方法的缺点是，仅能够大致的筛选出具有代谢途径信息的基因</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>data</td>
<td>-</td>
</tr>
<tr>
<td>pathwayInfo</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="Diff"><a href="#Diff" class="headerlink" title="Diff"></a>Diff</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.Diff(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Model.Network.VirtualFootprint.DocumentFormat.VirtualFootprints&#125;,System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Model.Network.VirtualFootprint.DocumentFormat.VirtualFootprints&#125;,System.String,System.String,System.String)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>data1</td>
<td>-</td>
</tr>
<tr>
<td>data2</td>
<td>-</td>
</tr>
<tr>
<td>export</td>
<td>导出的文件夹</td>
</tr>
</tbody>
</table>
<h4 id="ExportRegulators"><a href="#ExportRegulators" class="headerlink" title="ExportRegulators"></a>ExportRegulators</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.ExportRegulators(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.PredictedRegulationFootprint&#125;)</span><br></pre></td></tr></table></figure>
<p>从调控数据之中导出真正被使用到的调控因子</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Regulations</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="FootPrintMatches"><a href="#FootPrintMatches" class="headerlink" title="FootPrintMatches"></a>FootPrintMatches</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.FootPrintMatches(System.String,System.String)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>MEME_OUT</td>
<td>MEME html文件夹集合</td>
</tr>
<tr>
<td>MAST_OUT</td>
<td>Mast html文件夹集合</td>
</tr>
</tbody>
</table>
<h4 id="FootprintMatchesTEXT"><a href="#FootprintMatchesTEXT" class="headerlink" title="FootprintMatchesTEXT"></a>FootprintMatchesTEXT</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.FootprintMatchesTEXT(System.String,System.String,SMRUCC.genomics.SequenceModel.NucleotideModels.SegmentReader,SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.PTT,System.Int32,System.Boolean)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>MEME_Text</td>
<td>-</td>
</tr>
<tr>
<td>MAST_html</td>
<td>-</td>
</tr>
<tr>
<td>GenomeSequence</td>
<td>-</td>
</tr>
<tr>
<td>GeneBriefInformation</td>
<td>-</td>
</tr>
<tr>
<td>ATGDistance</td>
<td>-</td>
</tr>
<tr>
<td>FilterPromoter</td>
<td>由于某一个位点是落在某一个基因的内部或者下游区的，所以使用这个参数来过滤掉这些不在启动子区的位点，</td>
</tr>
</tbody>
</table>
<p> 本参数为真，则执行过滤操作，返回的记过之中仅包含有启动子区的位点，不为真，则返回所有类型的位点。<br> 默认不进行过滤操作|</p>
<h4 id="GenerateNetwork"><a href="#GenerateNetwork" class="headerlink" title="GenerateNetwork"></a>GenerateNetwork</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.GenerateNetwork(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.PredictedRegulationFootprint&#125;,System.String)</span><br></pre></td></tr></table></figure>
<p>Export a cytoscape network file from the predicted footprint data.(从所预测的footprint调控数据之中导出一个cytoscape网络的定义文件)</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Regulations</td>
<td>-</td>
</tr>
<tr>
<td>saveto</td>
<td>所导出的文件夹</td>
</tr>
</tbody>
</table>
<h4 id="GroupMotifs"><a href="#GroupMotifs" class="headerlink" title="GroupMotifs"></a>GroupMotifs</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.GroupMotifs(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.PredictedRegulationFootprint&#125;,SMRUCC.genomics.SequenceModel.NucleotideModels.NucleicAcid,System.Int32)</span><br></pre></td></tr></table></figure>
<p>按照家族将可能重复的motif归为一组数据</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>data</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="Matches"><a href="#Matches" class="headerlink" title="Matches"></a>Matches</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.Matches(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Interops.NBCR.MEME_Suite.DocumentFormat.MEME.HTML.MEMEOutput&#125;,System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifDb.MotifFamily&#125;,System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Data.Regprecise.RegpreciseMPBBH&#125;,SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.PTTDbLoader,System.String,System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Assembly.KEGG.Archives.Csv.Pathway&#125;,SMRUCC.genomics.Analysis.RNA_Seq.PccMatrix,System.Boolean,System.Double,System.Int32)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>data</td>
<td>-</td>
</tr>
<tr>
<td>MotifDatabase</td>
<td>-</td>
</tr>
<tr>
<td>RegulatorMatches</td>
<td>-</td>
</tr>
<tr>
<td>GenomeBrief</td>
<td>-</td>
</tr>
<tr>
<td>Door</td>
<td>-</td>
</tr>
<tr>
<td>KEGG_Pathways</td>
<td>-</td>
</tr>
<tr>
<td>PccMatrix</td>
<td>PCC或者SPCC的混合矩阵</td>
</tr>
<tr>
<td>ignoreDirection</td>
<td>-</td>
</tr>
<tr>
<td>pccCutoff</td>
<td>-</td>
</tr>
<tr>
<td>ATGDistance</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="MatchRegulator"><a href="#MatchRegulator" class="headerlink" title="MatchRegulator"></a>MatchRegulator</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.MatchRegulator(SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.PredictedRegulationFootprint,System.Collections.Generic.Dictionary&#123;System.String,SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifDb.Motif&#125;,System.Collections.Generic.KeyValuePair&#123;System.String,System.String&#125;[])</span><br></pre></td></tr></table></figure>
<p>通过@”P:SMRUCC.genomics.Interops.NBCR.MEME_Suite.DocumentFormat.MEME.HTML.MEMEOutput.ObjectId”</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>item</td>
<td>-</td>
</tr>
<tr>
<td>MotifDb</td>
<td>-</td>
</tr>
<tr>
<td>maps</td>
<td>-</td>
</tr>
</tbody>
</table>
<p>_returns: {RegpreciseRegulators(), bh<em>Matches()}</em></p>
<h4 id="PathwayFunctionAssociation"><a href="#PathwayFunctionAssociation" class="headerlink" title="PathwayFunctionAssociation"></a>PathwayFunctionAssociation</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.PathwayFunctionAssociation(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Model.Network.VirtualFootprint.DocumentFormat.VirtualFootprints&#125;,System.String,System.String)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>data</td>
<td>-</td>
</tr>
<tr>
<td>KEGGPathway</td>
<td>数据存放的文件夹</td>
</tr>
<tr>
<td>EXPORT</td>
<td>结果数据的导出文件名，当参数不为空的时候，会导出一个Csv文件</td>
</tr>
</tbody>
</table>
<h4 id="PccAssumption"><a href="#PccAssumption" class="headerlink" title="PccAssumption"></a>PccAssumption</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.PccAssumption(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.PredictedRegulationFootprint&#125;,System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Data.Regprecise.RegpreciseMPBBH&#125;)</span><br></pre></td></tr></table></figure>
<p>可以使用本方法来假设Pcc数据，以方便后面的模拟计算的实验</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>data</td>
<td>-</td>
</tr>
<tr>
<td>RegulatorMatches</td>
<td>-</td>
</tr>
</tbody>
</table>
<blockquote>
<p> 对于所有基因调控关系对，都假设为0.85，根据Regprecise数据库之中所记录的Effect来猜测可能的符号</p>
</blockquote>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.html" class="article-date">
  <time datetime="2016-07-27T11:12:44.680Z" itemprop="datePublished">2016-07-27</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Interops.NBCR.MEME_Suite.Analysis.GenomeMotifFootPrints.MotifFootPrintsGenerates.html" data-id="ciqc7dw9c0200igv4dnjz591g" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

