

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>IdentifyUTRs | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="IdentifyUTRs_namespace: SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs_
For each gene, identify its 5’UTR and 3’UTR based on the expression data.
Methods__analysis1SMRUCC.genomics.Analy">
<meta property="og:type" content="website">
<meta property="og:title" content="IdentifyUTRs">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="IdentifyUTRs_namespace: SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs_
For each gene, identify its 5’UTR and 3’UTR based on the expression data.
Methods__analysis1SMRUCC.genomics.Analy">
<meta property="og:updated_time" content="2016-07-07T09:40:57.083Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="IdentifyUTRs">
<meta name="twitter:description" content="IdentifyUTRs_namespace: SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs_
For each gene, identify its 5’UTR and 3’UTR based on the expression data.
Methods__analysis1SMRUCC.genomics.Analy">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.html">
      <h1 class="article-title" itemprop="name">
        IdentifyUTRs
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.html" class="article-date">
  <time datetime="2016-07-07T09:40:57.083Z" itemprop="datePublished">2016-07-07</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.html">
      <h1 class="article-title" itemprop="name">
        IdentifyUTRs
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#IdentifyUTRs"><span class="toc-number">1.</span> <span class="toc-text">IdentifyUTRs</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Methods"><span class="toc-number">1.0.1.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#analysis"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">__analysis</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#dataPartitionings"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">__dataPartitionings</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#genomeAssumption"><span class="toc-number">1.0.1.3.</span> <span class="toc-text">__genomeAssumption</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#getPoissonPDF"><span class="toc-number">1.0.1.4.</span> <span class="toc-text">__getPoissonPDF</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#setBoundary"><span class="toc-number">1.0.1.5.</span> <span class="toc-text">__setBoundary</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#GenomicsContext"><span class="toc-number">1.0.1.6.</span> <span class="toc-text">GenomicsContext</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#identifyUTRs"><span class="toc-number">1.0.1.7.</span> <span class="toc-text">identifyUTRs</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#siRNAPredictes"><span class="toc-number">1.0.1.8.</span> <span class="toc-text">siRNAPredictes</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#TestSites"><span class="toc-number">1.0.1.9.</span> <span class="toc-text">TestSites</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="IdentifyUTRs"><a href="#IdentifyUTRs" class="headerlink" title="IdentifyUTRs"></a>IdentifyUTRs</h1><p>_namespace: <a href="N-SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.html">SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs</a>_</p>
<p>For each gene, identify its 5’UTR and 3’UTR based on the expression data.</p>
<h3 id="Methods"><a href="#Methods" class="headerlink" title="Methods"></a>Methods</h3><h4 id="analysis"><a href="#analysis" class="headerlink" title="__analysis"></a>__analysis</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.__analysis(System.Int32,System.Boolean,System.Int32,System.Int64,SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.DocumentFormat.Transcript[],System.Collections.Generic.SortedDictionary&#123;System.String,SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.DocumentFormat.Transcript&#125;@,SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.Replicate@)</span><br></pre></td></tr></table></figure>
<p>For each IG region</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>x</td>
<td>-</td>
</tr>
<tr>
<td>unstranded</td>
<td>-</td>
</tr>
<tr>
<td>replicate</td>
<td>-</td>
</tr>
<tr>
<td>genome</td>
<td>真实的基因组上下文或者假基因种子，请注意ORF一定要填满ATG和TGA位点的值</td>
</tr>
</tbody>
</table>
<h4 id="dataPartitionings"><a href="#dataPartitionings" class="headerlink" title="__dataPartitionings"></a>__dataPartitionings</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.__dataPartitionings(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.ReadsCount&#125;,System.Int32,System.Int32,System.Int64,System.Int32,System.Boolean,System.Boolean)</span><br></pre></td></tr></table></figure>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Transcripts</td>
<td>-</td>
</tr>
<tr>
<td>sharedReads</td>
<td>-</td>
</tr>
<tr>
<td>genomeSize</td>
<td>-</td>
</tr>
<tr>
<td>readsLen</td>
<td>-</td>
</tr>
<tr>
<td>unstrand</td>
<td>-</td>
</tr>
<tr>
<td>siRNAPredicts</td>
<td>筛选的模式会反转</td>
</tr>
</tbody>
</table>
<h4 id="genomeAssumption"><a href="#genomeAssumption" class="headerlink" title="__genomeAssumption"></a>__genomeAssumption</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.__genomeAssumption(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.DocumentFormat.Transcript&#125;,System.Int64)</span><br></pre></td></tr></table></figure>
<p>从转录组数据之中生成假基因</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Transcripts</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="getPoissonPDF"><a href="#getPoissonPDF" class="headerlink" title="__getPoissonPDF"></a>__getPoissonPDF</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.__getPoissonPDF(System.Int32,System.Double)</span><br></pre></td></tr></table></figure>
<p>Returns the PDF value at x for the specified Poisson distribution.</p>
<h4 id="setBoundary"><a href="#setBoundary" class="headerlink" title="__setBoundary"></a>__setBoundary</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.__setBoundary(SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.DocumentFormat.Transcript,System.Collections.Generic.SortedDictionary&#123;System.String,Microsoft.VisualBasic.ComponentModel.Value&#123;System.Int32&#125;&#125;,System.Collections.Generic.SortedDictionary&#123;System.String,Microsoft.VisualBasic.ComponentModel.Value&#123;System.Int32&#125;&#125;)</span><br></pre></td></tr></table></figure>
<p>设置左右端起始和终止的位点，非ATG和TGA</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Transcript</td>
<td>-</td>
</tr>
<tr>
<td>gStart</td>
<td>-</td>
</tr>
<tr>
<td>gStop</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="GenomicsContext"><a href="#GenomicsContext" class="headerlink" title="GenomicsContext"></a>GenomicsContext</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.GenomicsContext(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.DocumentFormat.Transcript&#125;,SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.PTT,System.Int32)</span><br></pre></td></tr></table></figure>
<p>这个函数则会将TSSs和TTS组装在一个构成完整的基因结构的信息</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sites</td>
<td>includes TSSs/TTS</td>
</tr>
<tr>
<td>PTT</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="identifyUTRs"><a href="#identifyUTRs" class="headerlink" title="identifyUTRs"></a>identifyUTRs</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.identifyUTRs(SMRUCC.genomics.Assembly.NCBI.GenBank.TabularFormat.PTT,System.Boolean,System.String,System.Int32,System.Double,System.String)</span><br></pre></td></tr></table></figure>
<p>For each gene, identify its 5’UTR and 3’UTR based on the expression data.(使用现有的基因组上下文数据)</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Transcripts</td>
<td>File path of the RNA-seq @”T:SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.DocumentFormat.Transcript” csv document</td>
</tr>
<tr>
<td>minExpression</td>
<td>0-1之间的一个数</td>
</tr>
</tbody>
</table>
<h4 id="siRNAPredictes"><a href="#siRNAPredictes" class="headerlink" title="siRNAPredictes"></a>siRNAPredictes</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.siRNAPredictes(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.ReadsCount&#125;,System.Int64,System.Boolean,System.Int32,System.Int32,System.Int32)</span><br></pre></td></tr></table></figure>
<p>和TSS不同的是，小RNA分子的表达量一般较低，所以在这里对原始数据是反向筛选的</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>readsCount</td>
<td>-</td>
</tr>
<tr>
<td>genomeSize</td>
<td>-</td>
</tr>
<tr>
<td>unstranded</td>
<td>-</td>
</tr>
<tr>
<td>sharedReads</td>
<td>-</td>
</tr>
<tr>
<td>minIGD</td>
<td>-</td>
</tr>
</tbody>
</table>
<h4 id="TestSites"><a href="#TestSites" class="headerlink" title="TestSites"></a>TestSites</h4><figure class="highlight csharp"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.TestSites(System.Collections.Generic.IEnumerable&#123;SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.ReadsCount&#125;,System.Int64,System.Boolean,System.Int32,System.Int32)</span><br></pre></td></tr></table></figure>
<p>Testing if the site can be identified as a TSSs.(使用种子来鉴定)</p>
<table>
<thead>
<tr>
<th>Parameter Name</th>
<th>Remarks</th>
</tr>
</thead>
<tbody>
<tr>
<td>Transcripts</td>
<td>-</td>
</tr>
</tbody>
</table>
<p>_returns: 程序会尝试延伸，假若不能够继续延伸，则认为是转录边界<br> 由于序列片段之间会存在重叠的情况，所以在计算之前需要先分区，将序列分割为非重叠的状态，即序列片段之间的最小距离要满足一个用户自定义的基因间隔区的最小距离</p>
<p> _</p>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.html" class="article-date">
  <time datetime="2016-07-07T09:40:57.083Z" itemprop="datePublished">2016-07-07</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Analysis.RNA_Seq.TSSsTools.Transcriptome.UTRs.IdentifyUTRs.html" data-id="ciqc7cxcp00qbigv4lhmrjqqz" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

