

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>CLIBuilder | GCModeller docs library</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="baidu-site-verification" content="1ZsVVOmjoT" />
    <meta name="description" content="CLIBuildernamespace: SMRUCC.genomics.Interops.ClustalOrg
Clustal Omega - 1.2.0 (AndreaGiacomo)
 If you Like Clustal - Omega please cite: Sievers F, Wilm A, Dineen D, Gibson TJ, Karplus K, Li W, Lopez">
<meta property="og:type" content="website">
<meta property="og:title" content="CLIBuilder">
<meta property="og:url" content="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Interops.ClustalOrg.CLIBuilder.html">
<meta property="og:site_name" content="GCModeller docs library">
<meta property="og:description" content="CLIBuildernamespace: SMRUCC.genomics.Interops.ClustalOrg
Clustal Omega - 1.2.0 (AndreaGiacomo)
 If you Like Clustal - Omega please cite: Sievers F, Wilm A, Dineen D, Gibson TJ, Karplus K, Li W, Lopez">
<meta property="og:updated_time" content="2016-07-07T09:40:20.695Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="CLIBuilder">
<meta name="twitter:description" content="CLIBuildernamespace: SMRUCC.genomics.Interops.ClustalOrg
Clustal Omega - 1.2.0 (AndreaGiacomo)
 If you Like Clustal - Omega please cite: Sievers F, Wilm A, Dineen D, Gibson TJ, Karplus K, Li W, Lopez">
    
      <link rel="alternative" href="/atom.xml" title="GCModeller docs library" type="application/atom+xml">
    
    
      <link rel="icon" href="http://gcmodeller.org/DNA.png">
    
    <link rel="stylesheet" href="/css/style.css">

    
<!-- Baidu Analytics -->
<script type="text/javascript">
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?fe2c7555911b8560db5e56121c5d5960";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>
<!-- End Baidu Analytics -->


    <script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>
</head>
<body>
<div id="container"> 

    
    <div id="wrap">
	   

<header id="header" class="hdpage">
    
    <div id="header-outer" class="outer">
        
        <div id="header-inner" class="inner">
          <nav id="main-nav">
            <a id="main-nav-toggle" class="nav-icon"></a>
            
              <a class="main-nav-link" href="/">
                  
                  

                
                  Home
                </a>
            
              <a class="main-nav-link" href="/API">
                  
                  

                
                  API
                </a>
            
              <a class="main-nav-link" href="/tools">
                  
                  

                
                  Tools
                </a>
            
              <a class="main-nav-link" href="/about">
                  
                  

                
                  About
                </a>
            
              <a class="main-nav-link" href="https://github.com/SMRUCC/">
                  
                  

                
                  Github
                </a>
            
          </nav>
          <nav id="sub-nav">
            
              <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
            
            <a id="nav-search-btn" class="nav-icon" title="Search"></a>
          </nav>
          <div id="search-form-wrap">
            <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" results="0" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://docs.gcmodeller.org"></form>
          </div>
        </div>
    </div>
</header>
		<div class="outer">
		
    <article id="page-undefined" class="article article-type-page" itemscope itemprop="blogPost">
      <div class="article-meta">
        
          <header class="article-header">
            
  
    <a href="/API/T-SMRUCC.genomics.Interops.ClustalOrg.CLIBuilder.html">
      <h1 class="article-title" itemprop="name">
        CLIBuilder
      </h1>
    </a>
  



          </header>
        

        <!--<a href="/API/T-SMRUCC.genomics.Interops.ClustalOrg.CLIBuilder.html" class="article-date">
  <time datetime="2016-07-07T09:40:20.695Z" itemprop="datePublished">2016-07-07</time>
</a>-->
        <!--  -->
      </div>
      <div class="article-inner">
            
            
              <!--<header class="article-header">
                
  
    <a href="/API/T-SMRUCC.genomics.Interops.ClustalOrg.CLIBuilder.html">
      <h1 class="article-title" itemprop="name">
        CLIBuilder
      </h1>
    </a>
  



              </header>-->
            
            <div class="article-entry" itemprop="articleBody">
              <div class="page-toc">
                <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#CLIBuilder"><span class="toc-number">1.</span> <span class="toc-text">CLIBuilder</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Properties"><span class="toc-number">1.0.1.</span> <span class="toc-text">Properties</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#Auto"><span class="toc-number">1.0.1.1.</span> <span class="toc-text">Auto</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#ClusteringOut"><span class="toc-number">1.0.1.2.</span> <span class="toc-text">ClusteringOut</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#ClusterSize"><span class="toc-number">1.0.1.3.</span> <span class="toc-text">ClusterSize</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Dealign"><span class="toc-number">1.0.1.4.</span> <span class="toc-text">Dealign</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#DistmatIn"><span class="toc-number">1.0.1.5.</span> <span class="toc-text">DistmatIn</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#DistmatOut"><span class="toc-number">1.0.1.6.</span> <span class="toc-text">DistmatOut</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Force"><span class="toc-number">1.0.1.7.</span> <span class="toc-text">Force</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Full"><span class="toc-number">1.0.1.8.</span> <span class="toc-text">Full</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#FullIter"><span class="toc-number">1.0.1.9.</span> <span class="toc-text">FullIter</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#GuidetreeIn"><span class="toc-number">1.0.1.10.</span> <span class="toc-text">GuidetreeIn</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#GuidetreeOut"><span class="toc-number">1.0.1.11.</span> <span class="toc-text">GuidetreeOut</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Help"><span class="toc-number">1.0.1.12.</span> <span class="toc-text">Help</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#HMMIn"><span class="toc-number">1.0.1.13.</span> <span class="toc-text">HMMIn</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#InFile"><span class="toc-number">1.0.1.14.</span> <span class="toc-text">InFile</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#InFmt"><span class="toc-number">1.0.1.15.</span> <span class="toc-text">InFmt</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#IsProfile"><span class="toc-number">1.0.1.16.</span> <span class="toc-text">IsProfile</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Iterations"><span class="toc-number">1.0.1.17.</span> <span class="toc-text">Iterations</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Log"><span class="toc-number">1.0.1.18.</span> <span class="toc-text">Log</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#LongVersion"><span class="toc-number">1.0.1.19.</span> <span class="toc-text">LongVersion</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#MaxGuidetreeIterations"><span class="toc-number">1.0.1.20.</span> <span class="toc-text">MaxGuidetreeIterations</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#MaxHMMIterations"><span class="toc-number">1.0.1.21.</span> <span class="toc-text">MaxHMMIterations</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#MaxNumSeq"><span class="toc-number">1.0.1.22.</span> <span class="toc-text">MaxNumSeq</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#MaxSeqLen"><span class="toc-number">1.0.1.23.</span> <span class="toc-text">MaxSeqLen</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Out"><span class="toc-number">1.0.1.24.</span> <span class="toc-text">Out</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#OutFmt"><span class="toc-number">1.0.1.25.</span> <span class="toc-text">OutFmt</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#OutputOrder"><span class="toc-number">1.0.1.26.</span> <span class="toc-text">OutputOrder</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#p1"><span class="toc-number">1.0.1.27.</span> <span class="toc-text">p1</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#p2"><span class="toc-number">1.0.1.28.</span> <span class="toc-text">p2</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#PercentId"><span class="toc-number">1.0.1.29.</span> <span class="toc-text">PercentId</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#ResidueNumber"><span class="toc-number">1.0.1.30.</span> <span class="toc-text">ResidueNumber</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#SeqType"><span class="toc-number">1.0.1.31.</span> <span class="toc-text">SeqType</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Threads"><span class="toc-number">1.0.1.32.</span> <span class="toc-text">Threads</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#UseKimura"><span class="toc-number">1.0.1.33.</span> <span class="toc-text">UseKimura</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Verbose"><span class="toc-number">1.0.1.34.</span> <span class="toc-text">Verbose</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Version"><span class="toc-number">1.0.1.35.</span> <span class="toc-text">Version</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Wrap"><span class="toc-number">1.0.1.36.</span> <span class="toc-text">Wrap</span></a></li></ol></li></ol></li></ol></li></ol>
              </div>
              
                <h1 id="CLIBuilder"><a href="#CLIBuilder" class="headerlink" title="CLIBuilder"></a>CLIBuilder</h1><p><em>namespace: <a href="N-SMRUCC.genomics.Interops.ClustalOrg.html">SMRUCC.genomics.Interops.ClustalOrg</a></em></p>
<p>Clustal Omega - 1.2.0 (AndreaGiacomo)</p>
<p> If you Like Clustal - Omega please cite:<br> Sievers F, Wilm A, Dineen D, Gibson TJ, Karplus K, Li W, Lopez R, McWilliam H, Remmert M, S├╢ding J, Thompson JD, Higgins DG.<br> Fast, scalable generation of high-quality protein multiple sequence alignments using Clustal Omega.<br> Mol Syst Biol. 2011 Oct 11;7:539. doi: 10.1038/msb.2011.75. PMID: 21988835.<br> If you don’t like Clustal-Omega, please let us know why (and cite us anyway).</p>
<p> Check http : //www.clustal.org for more information And updates.</p>
<p> Usage: clustalo [-hv] [-i {<file>,-}] [–hmm-In=<file>]… [–dealign] [–profile1=<file>] [–profile2=<file>]<br> [–Is-profile] [-t {Protein, RNA, DNA}] [–infmt={a2m=fa[sta],clu[stal],msf,phy[lip],selex,st[ockholm],vie[nna]}]<br> [–distmat-In=<file>] [–distmat-out=<file>] [–guidetree-In=<file>] [–guidetree-out=<file>] [–full] [–full-iter]<br> [–cluster-size=<n>] [–clustering-out=<file>] [–use-kimura] [–percent-id] [-o {file,-}]<br> [–outfmt={a2m=fa[sta],clu[stal],msf,phy[lip],selex,st[ockholm],vie[nna]}] [–residuenumber] [–wrap=<n>]<br> [–output-order={input-order,tree-order}] [–iterations=<n>] [–max-guidetree-iterations=<n>] [–max-hmm-iterations=<n>]<br> [–maxnumseq=<n>] [–maxseqlen=<l>] [–auto] [–threads=<n>] [-l <file>] [–version] [–Long-version] [–force]<br> [–MAC-RAM=<n>]</n></file></n></l></n></n></n></n></n></file></n></file></file></file></file></file></file></file></file></p>
<p> A typical invocation would be: clustalo -i my-in-seqs.fa -o my-out-seqs.fa -v</p>
<h3 id="Properties"><a href="#Properties" class="headerlink" title="Properties"></a>Properties</h3><h4 id="Auto"><a href="#Auto" class="headerlink" title="Auto"></a>Auto</h4><p>–auto<br> Set options automatically (might overwrite some of your options)</p>
<h4 id="ClusteringOut"><a href="#ClusteringOut" class="headerlink" title="ClusteringOut"></a>ClusteringOut</h4><p>–clustering-out=<file><br> Clustering output file</file></p>
<h4 id="ClusterSize"><a href="#ClusterSize" class="headerlink" title="ClusterSize"></a>ClusterSize</h4><p>–cluster-size=<n><br> soft maximum of sequences in sub-clusters</n></p>
<h4 id="Dealign"><a href="#Dealign" class="headerlink" title="Dealign"></a>Dealign</h4><p>–dealign<br> Dealign input sequences</p>
<h4 id="DistmatIn"><a href="#DistmatIn" class="headerlink" title="DistmatIn"></a>DistmatIn</h4><p>–distmat-in=<file><br> Pairwise distance matrix input file (skips distance computation)</file></p>
<h4 id="DistmatOut"><a href="#DistmatOut" class="headerlink" title="DistmatOut"></a>DistmatOut</h4><p>–distmat-out=<file><br> Pairwise distance matrix output file</file></p>
<h4 id="Force"><a href="#Force" class="headerlink" title="Force"></a>Force</h4><p>–force<br> Force file overwriting</p>
<h4 id="Full"><a href="#Full" class="headerlink" title="Full"></a>Full</h4><p>–full<br> Use full distance matrix for guide-tree calculation (might be slow; mBed Is default)</p>
<h4 id="FullIter"><a href="#FullIter" class="headerlink" title="FullIter"></a>FullIter</h4><p>–full-iter<br> Use full distance matrix for guide-tree calculation during iteration (might be slowish; mBed Is default)</p>
<h4 id="GuidetreeIn"><a href="#GuidetreeIn" class="headerlink" title="GuidetreeIn"></a>GuidetreeIn</h4><p>–guidetree-in=<file><br> Guide tree input file (skips distance computation And guide-tree clustering step)</file></p>
<h4 id="GuidetreeOut"><a href="#GuidetreeOut" class="headerlink" title="GuidetreeOut"></a>GuidetreeOut</h4><p>–guidetree-out=<file><br> Guide tree output file</file></p>
<h4 id="Help"><a href="#Help" class="headerlink" title="Help"></a>Help</h4><p>-h, –help<br> Print this help And exit</p>
<h4 id="HMMIn"><a href="#HMMIn" class="headerlink" title="HMMIn"></a>HMMIn</h4><p>–hmm-in=<file><br> HMM input files</file></p>
<h4 id="InFile"><a href="#InFile" class="headerlink" title="InFile"></a>InFile</h4><p>-i, –in, –infile={<file>,-}<br> Multiple sequence input file (- for stdin)</file></p>
<h4 id="InFmt"><a href="#InFmt" class="headerlink" title="InFmt"></a>InFmt</h4><p>–infmt={a2m=fa[sta],clu[stal],msf,phy[lip],selex,st[ockholm],vie[nna]}<br> Forced sequence input file format (default: auto)</p>
<h4 id="IsProfile"><a href="#IsProfile" class="headerlink" title="IsProfile"></a>IsProfile</h4><p>–Is-profile<br> disable check if profile, force profile (default no)</p>
<h4 id="Iterations"><a href="#Iterations" class="headerlink" title="Iterations"></a>Iterations</h4><p>–iterations, –iter=<n><br> Number of (combined guide-tree/HMM) iterations</n></p>
<h4 id="Log"><a href="#Log" class="headerlink" title="Log"></a>Log</h4><p>-l, –log=<file><br> Log all non-essential output to this file</file></p>
<h4 id="LongVersion"><a href="#LongVersion" class="headerlink" title="LongVersion"></a>LongVersion</h4><p>–long-version<br> Print long version information And exit</p>
<h4 id="MaxGuidetreeIterations"><a href="#MaxGuidetreeIterations" class="headerlink" title="MaxGuidetreeIterations"></a>MaxGuidetreeIterations</h4><p>–max-guidetree-iterations=<n><br> Maximum number guidetree iterations</n></p>
<h4 id="MaxHMMIterations"><a href="#MaxHMMIterations" class="headerlink" title="MaxHMMIterations"></a>MaxHMMIterations</h4><p>–max-hmm-iterations=<n><br> Maximum number of HMM iterations</n></p>
<h4 id="MaxNumSeq"><a href="#MaxNumSeq" class="headerlink" title="MaxNumSeq"></a>MaxNumSeq</h4><p>–maxnumseq=<n><br> Maximum allowed number of sequences</n></p>
<h4 id="MaxSeqLen"><a href="#MaxSeqLen" class="headerlink" title="MaxSeqLen"></a>MaxSeqLen</h4><p>–maxseqlen=<l><br> Maximum allowed sequence length</l></p>
<h4 id="Out"><a href="#Out" class="headerlink" title="Out"></a>Out</h4><p>-o, –out, –outfile={file,-}<br> Multiple sequence alignment output file (default stdout)</p>
<h4 id="OutFmt"><a href="#OutFmt" class="headerlink" title="OutFmt"></a>OutFmt</h4><p>–outfmt={a2m=fa[sta],clu[stal],msf,phy[lip],selex,st[ockholm],vie[nna]}<br> MSA output file format (default: fasta)</p>
<h4 id="OutputOrder"><a href="#OutputOrder" class="headerlink" title="OutputOrder"></a>OutputOrder</h4><p>–output-order={input-order,tree-order}<br> MSA output orderlike in input/guide-tree</p>
<h4 id="p1"><a href="#p1" class="headerlink" title="p1"></a>p1</h4><p>–profile1, –p1=<file><br> Pre-aligned multiple sequence file (aligned columns will be kept fix)</file></p>
<h4 id="p2"><a href="#p2" class="headerlink" title="p2"></a>p2</h4><p>–profile2, –p2=<file><br> Pre-aligned multiple sequence file (aligned columns will be kept fix)</file></p>
<h4 id="PercentId"><a href="#PercentId" class="headerlink" title="PercentId"></a>PercentId</h4><p>–percent-id<br> convert distances into percent identities (default no)</p>
<h4 id="ResidueNumber"><a href="#ResidueNumber" class="headerlink" title="ResidueNumber"></a>ResidueNumber</h4><p>–residuenumber, –resno<br> in Clustal format print residue numbers (default no)</p>
<h4 id="SeqType"><a href="#SeqType" class="headerlink" title="SeqType"></a>SeqType</h4><p>-t, –seqtype={Protein, RNA, DNA}<br> Force a sequence type (default: auto)</p>
<h4 id="Threads"><a href="#Threads" class="headerlink" title="Threads"></a>Threads</h4><p>–threads=<n><br> Number of processors to use</n></p>
<h4 id="UseKimura"><a href="#UseKimura" class="headerlink" title="UseKimura"></a>UseKimura</h4><p>–use-kimura<br> use Kimura distance correction for aligned sequences (default no)</p>
<h4 id="Verbose"><a href="#Verbose" class="headerlink" title="Verbose"></a>Verbose</h4><p>-v, –verbose<br> Verbose output (increases if given multiple times)</p>
<h4 id="Version"><a href="#Version" class="headerlink" title="Version"></a>Version</h4><p>–version<br> Print version information And exit</p>
<h4 id="Wrap"><a href="#Wrap" class="headerlink" title="Wrap"></a>Wrap</h4><p>–wrap=<n><br> number of residues before line-wrap in output</n></p>

              
            </div>
            <footer class="article-footer">
              <a href="/API/T-SMRUCC.genomics.Interops.ClustalOrg.CLIBuilder.html" class="article-date">
  <time datetime="2016-07-07T09:40:20.695Z" itemprop="datePublished">2016-07-07</time>
</a>
              <a data-url="http://docs.gcmodeller.org/API/T-SMRUCC.genomics.Interops.ClustalOrg.CLIBuilder.html" data-id="ciqc7dw8601zmigv4uvqmraay" class="article-share-link">Share</a>
              
              
            </footer>
      </div>
      
        
      
    </article>
    <script type="text/javascript">
      window.onscroll = function(){ 
        $(function(){
          var $toc = $('.page-toc'),
              $article = $('.article-entry')
              toc_top = $toc.offset().top
              tocsub_width = $toc.children().width()
              article_height = $article.children().height()
              scroll_top = document.body.scrollTop
              screen_h = $(window).height()
              footer_top = $('.article-footer').offset().top - $(window).height()
              if(toc_top&&scroll_top&&scroll_top>toc_top&&scroll_top<footer_top){
                $toc.children().css({
                  "position": "fixed",
                  "height": (screen_h -20)+'px',
                  "overflow": "auto"
                })
                $article.css('margin-right',tocsub_width+'px')
              }else{
                $toc.children().css({
                  "position": "relative",
                  "height":'inherit',
                  "overflow": "auto"
                })
                $article.css('margin-right',0+'px')
              }
        })
      } 
    </script>








		</div>
		

<footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
        <div class="left">
            &copy; 2016 GCModeller docs library<br>
            Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and Theme by <a href="https://github.com/jaywcjlove/hexoThemeKacper">hexoThemeKacper </a>
        </div>
        <div class="right">
            
        </div>
    </div>
  </div>
</footer>
<script src="//cdn.bootcss.com/jquery/2.0.3/jquery.min.js"></script>


<script type="text/javascript">
$(function(){
  var $img = $('.article-entry img'),
      $link = $img.css({
        "display":"inline-block"
      }).parent()

  if($link.parent().is("a")){
      $img.each(function(idx,item) {
      $(this).parent().attr('href',$(this).parent().parent().attr('href'))
    })
  }
})
</script>


<script src="/js/script.js"></script>

  	</div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/API" class="mobile-nav-link">API</a>
  
    <a href="/tools" class="mobile-nav-link">Tools</a>
  
    <a href="/about" class="mobile-nav-link">About</a>
  
    <a href="https://github.com/SMRUCC/" class="mobile-nav-link">Github</a>
  
</nav>
    
</div>
</body>
</html>

